<html>

<head>
    <meta charset="utf-8">
    <link rel="stylesheet" type="text/css" href="../../shared/SkylineStyles.css">
    <script src="../../shared/skyline.js" type="text/javascript"></script>
</head>

<body onload="skylineOnload();">
    <h1 class="document-title">Skyline Processing Grouped Study Data</h1>
    <p class="subtitle">
        Label-Free Differentiation for Method Refinement
    </p>
    <p>
        Targeted proteomics as an approach has become popular largely because of its advantages in measuring differences in protein and peptide
        expression across multiple groups or states of organisms. Skyline was originally funded by the NCI Clinical Proteomics Technology Assessment
        for Cancer (CPTAC) program in its efforts to show targeted proteomics could be effectively applied to detect and achieve clinical verification
        of candidates for disease biomarkers.
    </p>
    <p>
        In this tutorial, you will use Skyline to analyze SRM data from an experimental case-control study of heart failure using a salt-sensitive rat
        model. This study sought to identify differences in protein expression in plasma between diseased and healthy groups utilizing a list of
        candidate proteins associated with heart failure derived from the literature.
    </p>
    <p>
        This experiment monitored 137 peptides from 49 proteins with a set of 42 LC-MS/MS injections. Once familiar with the data processing and
        visualization techniques described in this tutorial, you should find it relatively straightforward to manage and assess data from larger
        studies with even more targets across more experimental groups. Without this tutorial, however, one can easily get overwhelmed by the details
        of working with comparatively smaller data sets. The knowledge we will gain here may be applied to larger datasets and other acquisition
        methods (PRM, DIA, and DDA) supported by Skyline.
    </p>
    <h1>Getting Started</h1>
    <p>
        To start this tutorial, download the following ZIP file: 
    </p>
    <p>
        <a href="https://skyline.ms/tutorials/GroupedStudies1.zip">https://skyline.ms/tutorials/GroupedStudies1.zip</a>
    </p>
    <p>
        Extract the files in it to a folder on your computer, like: 
    </p>
    <p>
        C:\Users\brendanx\Documents 
    </p>
    <p>
        This will create a new folder: 
    </p>
    <p>
        C:\Users\brendanx\Documents\GroupedStudies1
    </p>
    <p>
        If you inspect the contents of this new folder, you will find that it contains a subfolder:
    </p>
    <ul>
        <li>Heart Failure – files from a heart failure study in a salt sensitive rat model, provided by the labs of Michael J. Toth and Michael J.
            MacCoss.</li>
    </ul>
    <p>
        Follow the tutorial below to learn how to process data for a study of this nature, and to gain a better understanding of a study’s
        protein/peptide targets as well as its overall data quality.
    </p>
    <h1>Differential analysis after method refinement</h1>
    <p>
        In the “Heart Failure” subfolder of the “GroupedStudies1” folder you just created, the data were collected at the MacCoss lab while
        investigating the potential and limitations of the approach described in the <a
            href="https://skyline.ms/tutorial_method_refine_zh.url">Targeted Method
            Refinement</a> tutorial. The data were never published for a variety of reasons, some of which will become apparent when you use Skyline
        to understand the collected data. However, many of the ideas that came from this trial of targeted method refinement and other experiments
        like it were published separate from this dataset.<sup>1</sup>
    </p>
    <p>
        In the original study, 109 proteins were selected from the literature based on their proposed involvement in heart failure. These proteins were
        added to a Skyline document. Some were added by importing FASTA format text for the protein sequences and allowing Skyline to perform <i>in
            silico</i> tryptic digestion. Parameters were set in Skyline to include all peptides between 6 and 30 amino acids long with no missed
        cleavages. For other proteins, the <i>in silico</i> processing was performed outside Skyline. The resulting peptide lists from these
        proteins were pasted directly into a Skyline document (see <a
            href="https://skyline.ms/tutorial_method_edit_zh.url">Targeted Method Editing</a>
        tutorial). This created a document with 2,165 target peptides. For each doubly-charged precursor, singly-charged product ions corresponding to
        y<sub>3</sub> through y<sub>(n-1)</sub> were considered. This initial, exhaustive method covered 12,194 transitions and was measured with an
        unscheduled SRM method on a sample of pooled rat plasma. This initial analysis was only performed once and required 151 separate mass
        spectrometer injections in 2009. Today, we would likely use DIA in this phase, since all 2,165 peptides could be tested with a single run on a
        mass spectrometer.
    </p>
    <p>
        The 151 raw data files were imported into Skyline and used to determine which of the target peptides showed any promise of being detected in
        the plasma matrix without any further sample preparation. Of the original 2,165 target peptides, only 135 (from 49 proteins) had a peak in the
        full-gradient chromatograms with sufficient co-elution of the targeted y-ions (2 with 3 ions, 5 with 4 ions, 27 with 5 ions and 103 with 6
        ions). These remaining 780 transitions were scheduled for quantification in a single run using 5-minute-wide retention time windows around the
        detected peaks in a 90-minute gradient. This scheduled method was then run on plasma digests from 14 salt sensitive rats (7 healthy and 7 that
        had experienced heart failure induced by a high salt diet) in triplicate for a total of 42 injections. These injections, which will be
        analyzed throughout the current tutorial, were intended to determine which of the 135 detectable peptides demonstrated any potential of
        differentiating between the two groups, making them candidates for further study as potential biomarkers.
    </p>
    <p>
        The overall goal of this study was to determine whether useful scientific insight could come from a targeted investigation, starting without
        very much prior knowledge of the targeted proteins or peptides and without aid of stable isotope labeled reference peptides.
    </p>
    <p>
        <img src="s-detection-and-differentiation-phases.png" />
    </p>
    <p>
        <b>Figure 1</b>: A schematic showing proposed “Detection” and “Differentiation” phases of targeted method refinement in a heart
        failure study on a salt sensitive rat model.
    </p>
    <p>
        To begin processing the data collected for the Differentiation phase of this method refinement study:
    </p>
    <ul>
        <li>Open the “Rat_plasma.sky” file in the “GroupedStudies1\Heart Failure\raw” subfolder.</li>
    </ul>
    <p>
        By inspecting the indicators at the bottom right corner of the Skyline window, you will see that the file you opened contains 49 proteins, 137
        peptides and 789 transitions.
    </p>
    <p>
        <img src="s-1.png" />
    </p>
    <p>
        This is not quite what is promised in the figure above (from a manuscript draft that was never published), but the document also contains one
        extra peptide list (at the bottom, named “S”), which contains 3 global normalization peptides. These are intended to reduce the impact of
        systematic variance in measurement that impacts all peptides in the sample. You will learn more about this later. Also, this file must have
        lost 1 protein with 1 peptide from what was described above.
    </p>
    <p>
        You will notice that some of the peptides have matching library spectra. You can see this in the <b>Targets</b> view. The peptide
        icons with spectrum lines in the lower right corner (<img src="../../shared/peptide-with-spectrum-lines-icon.png" />) have matching spectra and those without these lines (<img
            src="../../shared/peptide-without-spectrum-lines-icon.png" />) do not. The spectra come from two different public spectral libraries: one from NIST and one from the Global
        Proteome Machine (GPM). You can explore their coverage by doing the following:
    </p>
    <ul>
        <li>On the <b>View</b> menu, click <b>Document Grid</b>.</li>
        <li>On the <b>Reports</b> menu in the upper-left corner of the <b>Document Grid</b> form, click
            <b>Precursors</b>.</li>
        <li>Scroll to the right in the <b>Document Grid</b> to find the “Library Name” column, click it and click <b>Sort
                Descending</b>.</li>
    </ul>
    <p>
        You can now scroll down the grid to see the number of peptides with a library spectrum (80), from the Rat (NIST) library (49) and from the Rat
        (GPM) library (31). Note that the row number of the selected cell is shown in the toolbar above the grid.
    </p>
    <p>
        <img src="s-2.png" />
    </p>
    <p>
        For the current example, this information gives you a sense of the sparse prior knowledge on which the experiment was based. Modern knowledge
        of the proteins might be much higher, and spectrum prediction tools like <a
            href="https://skyline.ms/wiki/home/software/Skyline/page.view?name=asms_2020_maclean_video">Prosit</a>, now integrated in Skyline, would
        allow complete coverage of the targeted peptides.
    </p>
    <h2>Setting up to process multi-replicate data without reference standards</h2>
    <p>
        To begin processing the SRM data acquired for the 14 subjects in technical triplicate, do the following:
    </p>
    <ul>
        <li>Close the <b>Document Grid</b> form.</li>
        <li>On the <b>File</b> menu, choose <b>Import</b> and click <b>Results</b>.</li>
        <li>Click the <b>OK</b> button in the <b>Import Results</b> form to import single injection replicates.</li>
        <li>Click on the first raw data file named “D_102_REP1.raw” in the <b>Import Results Files</b> form.</li>
        <li>Hold down the Shift key and click the last raw data file named “H_162_REP3.raw” to select all the raw data files from the
            “GroupedStudies1\Heart Failure\raw” folder.</li>
        <li>Click the <b>Open</b> button.</li>
    </ul>
    <p class="keep-next">
        The files should begin loading, and Skyline will display progress in a window like the one shown below:
    </p>
    <p>
        <img src="s-3.png" />
    </p>
    <p>
        To continue preparing for data processing while Skyline imports the SRM chromatograms, do the following:
    </p>
    <ul>
        <li>Click the <b>Hide</b> button.</li>
        <li>Close the <b>Library Match</b> view.</li>
        <li>Click and drag the title of the <b>Targets</b> view until you see a floating blue rectangle tracking the mouse cursor.</li>
        <li>Release the left mouse button to leave the <b>Targets</b> view floating above the Skyline window.</li>
        <li>On the <b>View</b> menu, choose <b>Retention Times</b> and click <b>Replicate Comparison</b> (F8).</li>
        <li>Click and drag the title of the <b>Retention Times</b> view until you see the docking indicators.</li>
        <li>Drag the cursor into the bottom docking indicator to see a blue rectangle fill the bottom edge of the Skyline window, as shown below.</li>
    </ul>
    <p>
        <img src="s-4.png" />
    </p>
    <ul>
        <li>Release the left mouse button to dock the <b>Retention Times</b> view.</li>
        <li>On the <b>View</b> menu, choose <b>Peak Areas</b> and click <b>Replicate Comparison</b> (F7).</li>
        <li>Drag the <b>Peak Areas</b> view until the mouse cursor is inside a docking indicator, and a blue rectangle indicates it will
            share the bottom edge with the <b>Peak Areas</b> view.</li>
    </ul>
    <p>
        <img src="s-5.png" />
    </p>
    <ul>
        <li>Release the left mouse button to dock the <b>Peak Areas</b> view.</li>
        <li>Drag the <b>Targets</b> view until the mouse cursor is inside a docking indicator, and a blue rectangle indicates it will dock
            to the left edge of the Skyline window.</li>
    </ul>
    <p>
        <img src="s-6.png" />
    </p>
    <ul>
        <li>Close any other windows left floating outside the main Skyline window.</li>
        <li>On the <b>View</b> menu, choose <b>Arrange Graphs</b> and click <b>Grouped</b>.</li>
        <li>In the <b>Group panes</b> field enter “3” on the <b>Arrange Graphs Grouped</b> form.</li>
        <li>Click the <b>Distribute graphs among groups</b> option.</li>
        <li>In the <b>Sort order</b> dropdown list choose “Document”.</li>
    </ul>
    <p class="keep-next">
        The form should look like this:
    </p>
    <p>
        <img src="s-7.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
        <li>Maximize the Skyline main window.</li>
        <li>Select the first peptide in the document.</li>
    </ul>
    <p class="keep-next">
        Your Skyline window should now look something like this:
    </p>
    <p>
        <img src="s-8.png" />
    </p>
    <p>
        <b>Note</b>: This image was taken on a 24” monitor at 1920x1080. If you are viewing this tutorial digitally, you may want to zoom in
        to 200% or more to view this image. After this, the tutorial will return to using images better suited to the 8½ x 11” page layout. While many
        Skyline workshops have been taught with the 1024x768 screen resolution, you will likely enjoy using Skyline more with a larger monitor.
    </p>
    <p>
        Looking at the <b>Retention Times</b> view for this first peptide, K.GILAADESVGSMAK.R [28, 41], you can see that the integration is
        not very consistent. Most of the chosen peaks elute around 19 minutes. However, nearly one third (12) elute closer to 22 minutes.
    </p>
    <p>
        <img src="s-9.png" />
    </p>
    <p>
        The integration for this peptide is quite poor. While the peak at around 19 minutes appears to be the better peak, it is hard to have much
        confidence that it is caused by a single peptide. Furthermore, it is unlikely that any integration adjustment will result in a single peptide
        that can be measured consistently across all runs. Therefore, instead of wasting more time here making additional adjustments, you should:
    </p>
    <ul>
        <li>Delete the first peptide.</li>
    </ul>
    <h2>Scheduled SRM and peak truncation</h2>
    <p>
        Just by looking at the retention time plot for the second peptide, you can see that it is much more consistently integrated.
    </p>
    <p>
        <img src="s-10.png" />
    </p>
    <p>
        However, you can still see that the peak apexes for all transitions (shown by the horizontal lines in the bars) are not very consistent. If you
        look at a few of the chromatogram graphs, you can also see that the peaks are jagged and of medium- to low-intensity.
    </p>
    <p>
        If you do not see the order shown above, with all the diseased replicates (D_) on the left and all the healthy replicates (H_) on the right, do
        the following:
    </p>
    <ul>
        <li>Right-click the <b>Retention Times</b> view, choose <b>Order</b> and click <b>Document</b>.</li>
    </ul>
    <p>
        Before continuing, make one final adjustment to the plots you are viewing.
    </p>
    <ul>
        <li>Right-click on the <b>Peak Areas</b> view, choose <b>Normalize To</b> and click <b>Total</b>.</li>
    </ul>
    <p class="keep-next">
        This changes the <b>Peak Areas</b> plot from:
    </p>
    <p>
        <img src="s-11.png" />
    </p>
    <p class="keep-next">
        To:
    </p>
    <p>
        <img src="s-12.png" />
    </p>
    <p>
        This makes it possible to see differences in relative ion abundance across multiple runs in a single glance and adds to the information you are
        already getting from the <b>Retention Times</b> view. You might notice that D_103_REP3 and D_108_REP2 are quite different, and that
        H_162_REP3 is somewhat different from all other replicates. Note that only the D_108_REP2 sample looked suspicious when only using the
        <b>Retention Times</b> view.
    </p>
    <p>
        Do the following to inspect and correct these peaks:
    </p>
    <ul>
        <li>Click the D_103_REP3 bar in the <b>Peak Areas</b> view.</li>
    </ul>
    <p class="keep-next">
        This will activate the chromatogram graph for the peak, which looks like:
    </p>
    <p>
        <img src="s-13.png" />
    </p>
    <p>
        While plots for the correctly integrated peaks look more like this:
    </p>
    <p>
        <img src="s-14.png" />
    </p>
    <p>
        Since the scheduled acquisition window did not capture the entire elution profile for the peptide, Skyline has picked the wrong peak. However,
        you can still see the correct peak ending around 30.2 minutes. This is called a “truncated peak”. You can correct the integration for this
        peptide to use the truncated peak by doing the following:
    </p>
    <ul>
        <li>Click-and-drag beneath the x-axis at about 30.4 minutes, just after the truncated peak.</li>
        <li>Drag left until the integration boundary stops moving, and release the left mouse button.</li>
        <li>Click the bars for D_108_REP2 in the <b>Retention Times</b> view to activate its
            chromatogram graph.</li>
        <li>Click-and-drag beneath the x-axis of the graph from about 30.5 minutes to the far left.</li>
        <li>Click the bars for D_162_REP3 in the <b>Retention Times</b> view to activate its
            chromatogram graph.</li>
    </ul>
    <p>
        The peak for D_162_REP3 is also truncated. However, Skyline has already chosen it correctly, as it has in several other cases. You might be
        able to deduce peak truncations by closely inspecting the <b>Retention Times</b> plot. Pay close attention to the lengths of the
        bars, as well as the proximity of the horizontal lines to the bar edges.
    </p>
    <p>
        Peak truncation is a considerable problem for label-free data such as this. With isotope labeled reference peptides, where every point in the
        peak provides a valid ratio between light and heavy peptide precursors, you will lose precision with a truncated peak but the ratio between
        light and heavy should still be valid. With label-free data, conversely, you cannot count a truncated peak as a valid measurement of the
        peptide. You must either calculate your differential statistics with missing data or omit the peptide from consideration entirely. Skyline
        keeps track of truncated peaks in order to make that possible for downstream tools.
    </p>
    <p>
        Skyline defines a truncated peak as any peak where one of its boundaries is at a terminal point of the chromatogram and the intensity at that
        terminal end is more than 1% of the peak height higher than the intensity at the other integration boundary.
    </p>
    <p>
        To see other automatically integrated peaks which Skyline has identified as truncated, perform the following steps:
    </p>
    <ul>
        <li>On the <b>Edit</b> menu, click <b>Find</b> (Ctrl-F).</li>
        <li>Clear the <b>Find what</b> field.</li>
        <li>Choose <b>Down</b> for <b>Direction</b>.</li>
        <li>Click the <b>Show Advanced</b> button, advanced options are not already showing.</li>
        <li>Check <b>Truncated peaks</b> (and uncheck anything else that may be checked).</li>
    </ul>
    <p class="keep-next">
        The <b>Find</b> form should look like this:
    </p>
    <p>
        <img src="s-15.png" />
    </p>
    <ul>
        <li>Click the <b>Find All</b> button.</li>
        <li>Click the <b>Close</b> button.</li>
    </ul>
    <p>
        Skyline shows a long list of truncated precursors and transitions, starting with the three peaks you just inspected. You can double-click on
        any line to see the corresponding chromatogram graph activated.
    </p>
    <p>
        <img src="s-16.png" />
    </p>
    <p>
        You can also use the <b>Document Grid</b> to create a list of all precursors with any peak truncation by doing the following:
    </p>
    <ul>
        <li>On the <b>View</b> menu, click <b>Document Grid</b>.</li>
        <li>On the <b>Reports</b> menu in the <b>Document Grid</b> view, click <b>Precursors</b>.</li>
        <li>On the <b>Reports</b> menu in the <b>Document Grid</b> view, click <b>Customize Report</b>.</li>
        <li>In the <b>Report name</b> field, enter “Truncated Precursors”.</li>
        <li>In the right side list, select all column names (using click and shift-click) after the first two (Precursor and Peptide).</li>
        <li>Click the <img src="../../shared/red-x-icon.png" /> button near the right edge to delete all but the first two columns.</li>
        <li>Check “Replicates” in the tree on the left side.</li>
        <li>Expand the tree to check Proteins &gt; Peptides &gt; Precursors &gt; PrecursorResults &gt; Count Truncated.</li>
    </ul>
    <p class="keep-next">
        The <b>Customize Report</b> form should look something like this:
    </p>
    <p>
        <img src="s-17.png" />
    </p>
    <ul>
        <li>In the right side list, double-click “Count Truncated”.</li>
        <li>Click the <b>Filter</b> tab.</li>
        <li>Click the <b>Add</b> button.</li>
        <li>From the <b>Operation</b> dropdown list choose “Is Greater Than”.</li>
        <li>In the <b>Value</b> field, enter “0”.</li>
    </ul>
    <p>
        The <b>Customize Report</b> form should look something like this:
    </p>
    <p>
        <img src="s-18.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
        <li>Click the “Count Truncated” column header and click <b>Sort Descending</b>.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Grid</b> will show 221 truncated peaks that Skyline has already chosen automatically:
    </p>
    <p>
        <img src="s-19.png" />
    </p>
    <p>
        To view these peaks:
    </p>
    <ul>
        <li>Click the <b>Peptide</b> link for DFATVYVDAVK.</li>
        <li>Click each <b>Replicate</b> link for the peptide.</li>
    </ul>
    <p class="keep-next">
        Skyline will activate chromatogram graphs that look like the following:
    </p>
    <p>
        <img src="s-20.png" />
    </p>
    <ul>
        <li>Close the <b>Document Grid</b>.</li>
        <li>Close the <b>Find Results</b> view.</li>
        <li>Select the second peptide R.LGGEEVSVA<b><u>C</u></b>K.L [237, 247]</li>
    </ul>
    <h2>Beginning multi-replicate data processing</h2>
    <p>
        The peptide LGGEEVSVACK looks much better than the other two we have looked at so far. The retention times are quite stable and the horizontal
        peak apex lines match up for almost all transitions in each replicate.
    </p>
    <p>
        <img src="s-21.png" />
    </p>
    <p>
        The relative ion abundances shown in the <b>Peak Areas</b> view also appear relatively stable. You might notice that there is a
        corresponding library spectrum for this peptide and that its relative ion abundance (shown in the left-most bar in the plot) is similar to
        that of the measured peaks. To get a better sense of just how closely a peptide matches its corresponding library spectrum, do the following:
    </p>
    <ul>
        <li>Right-click the <b>Peak Areas</b> view, choose <b>Normalize To</b> and click <b>None</b>.</li>
        <li>Move the splitter bar to give the <b>Peak Areas</b> view most of the width of the screen. Conversely, if your screen space is
            limited, click-and-drag a zoom rectangle around the diseased (D_) replicates.</li>
    </ul>
    <p>
        Skyline will show you the dot-product (dotp) relationship, a measure of similarity between the library spectrum peak intensities and the
        measured peak areas ranging from 1 (best) to zero (worst).
    </p>
    <p>
        <img src="s-22.png" />
    </p>
    <p>
        Return the <b>Peak Areas</b> view to its former state by doing the following:
    </p>
    <ul>
        <li>If you used zooming, right-click and click <b>Undo All Zoom/Pan</b>.</li>
        <li>Right-click, choose <b>Normalize To</b> and click <b>Total</b>.</li>
    </ul>
    <p>
        Now take a closer look at the chromatograms for these peaks by doing the following:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Auto-Zoom</b> and click <b>Best Peak</b> (F11).</li>
        <li>Click the <b>Replicates</b> dropdown list in the <b>Targets</b> view.</li>
        <li>Use the down arrow key to scan through several of the chromatogram graphs.</li>
    </ul>
    <p>
        You might notice that some of the transitions show a signal around the edges of the main peak that is not well matched by the other peaks.
    </p>
    <p>
        <img src="s-23.png" />
    </p>
    <p>
        There is frequently signal on y5 and y6 around 13.1 minutes, which is clearly not from the same peptide.
    </p>
    <p>
        <img src="s-24.png" />
    </p>
    <p>
        When an interfering signal like this occurs inside the integration boundaries, it will add error to the quantitative measurements. For critical
        quantitative data, you might consider whether you need all 5 transitions for this peptide.
    </p>
    <p>
        For the current tutorial, however, continue to the next peptide. There you will find strong, clean, and co-eluting peaks for all 7 transitions.
    </p>
    <p>
        <img src="s-25.png" />
    </p>
    <p>
        Relative ion abundance is also extremely consistent.
    </p>
    <p>
        <img src="s-26.png" />
    </p>
    <p>
        The peak area in y3 is stronger than what was seen for the library spectrum. This might be explained by secondary fragmentation occurring in a
        triple quadrupole that might not occur with resonance excitation in an ion trap instrument, if an ion trap generated the library spectrum. In
        this case, the spectrum comes from a NIST public library, and we have no information on the instrument type that generated it.
    </p>
    <p>
        At a glance, this integration looks good enough that you should be able to continue to the next peptide, R.GSYNLQDLLAQAK.L [378, 390], without further inspection. For this next peptide, both the <b>Retention
            Times</b> and <b>Peak Areas</b> views show issues. To understand what is going on with these measurements, do the
        following:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Auto-Zoom</b> and click <b>None</b> (Shift-F11).</li>
        <li>Use the <b>Replicates</b> dropdown list in the <b>Targets</b> view to review the first few replicates until you come
            to the first problem, D_103_REP3.</li>
    </ul>
    <p>
        For the first 5 replicates the chosen peak is around 32.5 to 33.5 minutes and has signal on all 5 transitions. However, there is also another
        peak around 35 minutes with signal mostly on y4 and y5 that is surely caused by another peptide. In cases like this, interference that does
        not coincide with the peptide of interest can add information to the chromatogram “landscape” you see.
    </p>
    <p>
        <img src="s-27.png" />
    </p>
    <p class="keep-next">
        In D_103_REP3, you will see the integrated peak at 33.9 minutes with signal mostly on y4 and y5, but no other candidate peak
        to the left.
    </p>
    <p>
        <img src="s-28.png" />
    </p>
    <p>
        In this case, the scheduling window has failed to capture any signal from the targeted peptide. To remove the incorrect peak do the following:
    </p>
    <ul>
        <li>Right-click on the chromatogram graph, and click <b>Remove Peak</b>.</li>
    </ul>
    <p class="keep-next">
        This should leave a blank spot in the <b>Peak Areas</b> graph.
    </p>
    <p>
        <img src="s-29.png" />
    </p>
    <p>
        Continuing through the chromatogram plots, you will see this pattern repeated several times. Keep removing the missing peaks. You will even
        start to see some cases of peak truncation. When you get to H_148_REP2, you will see a case
        where Skyline has chosen the peak at 33.9 minutes, but part of the correct peak is still visible.
    </p>
    <p>
        <img src="s-30.png" />
    </p>
    <p>
        You can simply integrate this as a truncated peak, as you did above, by clicking and dragging beneath the x-axis. You should end up with a
        <b>Peak Areas</b> plot that looks like this:
    </p>
    <p>
        <img src="s-31.png" />
    </p>
    <p class="keep-next">
        And a retention times plot that looks like this:
    </p>
    <p>
        <img src="s-32.png" />
    </p>
    <ul>
        <li>Continue to the peptide R.DIPVNPM<b><u>C</u></b>IYR.S [46, 56].</li>
    </ul>
    <p>
        You will find no real consistent signal with integrated peaks all over the 5-minute scheduling window.
    </p>
    <ul>
        <li>You may simply delete this peptide.</li>
    </ul>
    <p>
        The next three peptides in the protein NP_001012027 all look fine with relatively stable retention times and relative ion abundances.
    </p>
    <p>
        You may feel the variance you observe in the retention times is more erratic than you would expect. This is often the case when the runs are
        logically ordered. The current order is a result of the way you imported the replicates into the document. For a study of this nature, it was
        useful to list all diseased subjects first and follow them with all the healthy subjects. All technical replicates for each subject are also
        grouped together.
    </p>
    <p class="keep-next">
        For the peptide TSDQIHFFFAK the retention times look
        like this:
    </p>
    <p>
        <img src="s-33.png" />
    </p>
    <p>
        However, it may also be quite helpful to see the runs in the order they were acquired on the instrument. You can achieve this by doing the
        following:
    </p>
    <ul>
        <li>Right-click the <b>Retention Times</b> graph, choose <b>Order By</b> and click <b>Acquired Time</b>.</li>
    </ul>
    <p class="keep-next">
        The graph will change to look like the following:
    </p>
    <p>
        <img src="s-34.png" />
    </p>
    <p>
        This retention time pattern is now a little less erratic. If you now select through the next two peptides (LQPLDFK, SQLPGIIAEGR), you will see
        how consistent the pattern is.
    </p>
    <p>
        The next peptide DFATVYVDAVK has some peak truncation. You should now be able to correct this quickly by clicking on the bars in the
        <b>Peak Areas</b> view that do not match well and clicking and dragging beneath the x-axis in the corresponding chromatogram graphs.
        There are also some peaks where the chromatography looks very noisy. An example of this is on the right in the chromatograms below. This is
        because only the very right edge of the peptide peak was measured, and the higher intensity portion is missing.
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig">
                <img src="s-35.png" />
            </td>
            <td class="comp-fig">
                <img src="s-36.png" />
            </td>
        </tr>
    </table>
    <p>
        In a case like this, you want to be very sure you avoid integrating some noise that Skyline fails to identify as a truncated peak. It may just
        be easier to remove the peaks for these replicates.
    </p>
    <p>
        Finishing this section with the peptide FGLYSDQMR, you can see in the <b>Retention
            Times</b> view that Skyline has mostly picked a peak close to 19 minutes. However, the <b>Peak Areas</b> view shows very
        poor consistency in relative ion abundance.
    </p>
    <p>
        <img src="s-37.png" />
    </p>
    <p>
        If you examine a few of the chromatogram graphs, you will find it hard to imagine consistently integrating the signal at 18.8 minutes with any
        confidence that all integrated signals come from a single peptide. In this case, you should simply delete the peptide and move on.
    </p>
    <h2>Global normalization standards</h2>
    <p>
        This data set lacks stable isotope labeled (SIL) peptides but it does contain synthetic global standard peptides that were spiked into the
        samples. The intent was to use these peptides for peak area normalization for all other peptides in an effort to reduce the effect of
        systematic variance in the LC-MS/MS phase of the experiment.<sup>2</sup> To examine these peptides now, do the following:
    </p>
    <ul>
        <li>Click in the <b>Targets</b> view.</li>
        <li>Press the End key and then the Up-Arrow key on the keyboard.</li>
    </ul>
    <p>
        This should move the selection to AFGLSSPR, the last peptide in the document. The three peptides
        HLNGFSVPR, VVLSGSDATLAYSAFK and AFGLSSPR grouped in the list named “S”
        are the injected synthetic peptides for this experiment. You can gain immediate confidence in the automatic integration of the last two
        peptides from the two summary plots we have been examining. The retention times and peak areas are relatively consistent.
    </p>
    <p>
        <img src="s-38.png" />
    </p>
    <p>
        <img src="s-39.png" />
    </p>
    <p>
        Moving to the peptide HLNGFSVPR, you will observe consistent retention times. However, there is
        much more variance in the peak areas. The relative area for y4 and sometimes y3 seems to vary throughout the data set.
    </p>
    <p>
        <img src="s-40.png" />
    </p>
    <p>
        If you click on some of the peak area bars where y4 is most abundant, you will see obvious interference on y4 and sometimes y3.
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Auto-Zoom</b> and click <b>Best Peak</b> (F11).</li>
    </ul>
    <p>
        <img src="s-41.png" />
    </p>
    <p>
        To gain an even clearer understanding of the performance of these injected peptides as potential normalization standards, you should do the
        following:
    </p>
    <ul>
        <li>Right-click the <b>Peak Areas</b> view, choose <b>Normalize To</b> and click <b>None</b>.</li>
    </ul>
    <p>
        Now you can see that the peak area for peptide HLNGFSVPR decreases dramatically from the results acquired for the first sample to the last. The
        total peak area goes from roughly 6,000,000 to 30,000, spanning a 200-fold difference. See if you able to use the <b>Results
            Grid</b> (under <b>View</b> &gt; <b>Other Grids</b> &gt; <b>Results Grid</b>) with the HLNGFSVPR
        peptide precursor 513.7776++ selected to determine the exact maximum and minimum total peak areas. Looking at the other two standard peptides,
        you will see that they too decrease over time (VVLSGSDATLAYSAFK 2.3 to 1.1 million and AFGLSSPR 23 to 1 million). All 42 runs are supposed to
        be essentially technical replicates for these three peptides. While there is clearly systematic signal degradation for all peptides over time
        in this data set, it is unlikely that very many of them experience 200x or even 20x decreases.
    </p>
    <p>
        To see where these peptides elute relative to each other do the following:
    </p>
    <ul>
        <li>Select the “S” element preceded by the <img src="../../shared/protein-icon-2.png" /> icon.</li>
    </ul>
    <p>
        The chromatogram graphs will change to show all three peptides in the “S” list together.
    </p>
    <p>
        <img src="s-42.png" />
    </p>
    <p>
        Now to see all the peptides in the document together:
    </p>
    <ul>
        <li>On the <b>Edit</b> menu, click <b>Select All</b> (Ctrl-A).</li>
    </ul>
    <p>
        <img src="s-43.png" />
    </p>
    <p>
        This makes it clear that the two problematic peptides are both hydrophilic and early eluting. In fact, they are the 3<sup>rd</sup> and
        10<sup>th</sup> eluting peptides in this experiment. You can do the following to see this in Skyline:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Retention Times</b> and click <b>Peptide Comparison</b> (Ctrl-F8).</li>
        <li>Right-click the <b>Retention Times</b> view, choose <b>Order</b> and click <b>Retention Time</b>.</li>
        <li>Click-and-drag a rectangle around the first 15 or so peptides in the <b>Retention Times</b> graph.</li>
    </ul>
    <p>
        Early eluting peptides tend to have more variance. Therefore, they are often not the best candidates for global normalization standards. These
        peptides may also be impacted by some other factor such as degradation while queuing in the auto-sampler.
    </p>
    <p>
        For another way to assess how similarly these peptides are behaving to the others around them:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Peak Areas</b> and click <b>Peptide Comparison</b> (Ctrl-F7).</li>
        <li>On the <b>View</b> menu, choose <b>Transitions</b> and click <b>Total</b> (Ctrl-F10).</li>
        <li>Right-click the <b>Peak Areas</b> view, and click <b>CV Values</b>.</li>
        <li>Click-and-drag a rectangle around the first 15 or so peptides in the <b>Peak Areas</b> graph.</li>
    </ul>
    <p>
        Before trying to understand what you see, you should remember that all of the non-standard peptides are expected to have much higher variance
        due to the fact that they are measured across 14 subjects. Also, the integration is not fully adjusted for all peptides. For the current data
        set, the automatically picked peaks looked correct for the standards. However, the HLNGFSVPR peptide has the second highest CV at about 160%
        while the AFGLSSPR peptide is close to the median for the first 15 peptides with a CV of about 50%. (The VVLSGSDATLAYSAFK peptide, which is
        more hydrophobic and later eluting, is 108<sup>th</sup> or in the 80<sup>th</sup> percentile with 18.6% CV.)
    </p>
    <p>
        <img src="s-44.png" />
    </p>
    <p>
        If you do the following, you should be able to convince yourself that the LGP peptide has a problem very similar to the HLN peptide.
    </p>
    <ul>
        <li>Click the bar for the LGP peptide in the <b>Peak Areas</b> view.</li>
        <li>On the <b>View</b> menu, choose <b>Peak Areas</b> and click <b>Replicate Comparison</b> (F7).</li>
    </ul>
    <p class="keep-next">
        Remember that there are supposed to be three cycles or technical replicates:
    </p>
    <p>
        <img src="s-45.png" />
    </p>
    <p class="keep-next">
        To convince yourself that the peaks are correctly integrated, do the following:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Transitions</b> and click <b>All</b> (F10).</li>
        <li>Right-click the <b>Peak Areas</b> view, choose <b>Normalize To</b> and click <b>Total</b>.</li>
    </ul>
    <p>
        You can see that despite the 200x range (140 to 0.7 million) from minimum to maximum, the relative product ion abundance between the runs is
        extremely stable.
    </p>
    <p>
        <img src="s-46.png" />
    </p>
    <p>
        Although this one peptide seems to have similar issues, you should exclude a peptide like HLNGFSVPR from consideration as a global
        normalization standard. Ideally, this takes place long before you inject your standard peptides into important quantitative data. In this
        case, you should only use VVLSGSDATLAYSAFK as a global normalization standard for all other peptides in the document. Further proving this
        notion is beyond the scope of this tutorial.
    </p>
    <h2>Continuing multi-replicate data processing</h2>
    <p>
        Now you will continue to review and correct peak integrations for this experiment by selecting the peptide DVFSQQADLSR above the “S” standard peptides list.
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Retention Times</b> and click <b>Replicate Comparison</b>.</li>
    </ul>
    <p>
        The consistency you will observe in the graphs for both the <b>Retention Times</b> and <b>Peak Areas</b> should give you
        the confidence that the peptide has been consistently integrated across all runs. You can continue up the <b>Targets</b> list
        without further review of this peptide. This should also be true of the next two peptides.
    </p>
    <p>
        Continue up the list until you reach the peptide IFSQQADLSR. You should notice that every
        replicate except H_146_REP1 has consistent relative ion abundances in the <b>Peak Areas</b> graph. Again, when you click on the bar
        for this run, you will see that the peak in H_146_REP1 was not fully captured by the 5-minute
        scheduling window. (You might need to turn off x-axis zooming – Shift-F11 – to see this.)
    </p>
    <p>
        <img src="s-47.png" />
    </p>
    <p>
        As you did before, you can correct this by clicking and dragging beneath the x-axis to integrate the truncated peak, or by using the
        right-click menu to remove the peak entirely.
    </p>
    <p>
        However, this is not the only issue with this peptide. When you look at the <b>Retention Times</b> graph, you will observe some
        unusual differences in the ranges of time being integrated.
    </p>
    <p>
        <img src="s-48.png" />
    </p>
    <p>
        If you review some of the chromatograms for this peptide, you will see that many show a single, very nice peak eluting over 0.2 minutes.
        Others, however, have some tailing that extends out to about 0.5 minutes while still others show the peptide eluting over 2 minutes with two
        distinct peaks.
    </p>
    <p>
        <img src="s-49.png" />
    </p>
    <p>
        A peptide like this may be very difficult to use for quantification, especially without a matching SIL peptide. You should probably delete this
        peptide from this experiment. At the very least, you should remove the peaks from the runs that show the double elution profile. 
    </p>
    <p>
        Continuing up the peptide list, you will find that the next 7 peptides show good enough consistency in the summary plots that you should not
        need to correct any of the default peak picking performed by Skyline or even give these peptides much more than just a glance.
    </p>
    <p>
        When you reach the peptide MLSGFIPLKPTVK, however, you will see much more variance in the
        <b>Peak Areas</b> graph.
    </p>
    <p>
        <img src="s-50.png" />
    </p>
    <p>
        The total peak area for this peptide is dominated by the y7 ion. Although this is also reflected in the library spectrum (as shown at the far
        left of the above graph), visual inspection of the chromatograms makes it difficult to discern if anything is actually co-eluting with the y7
        ion (see below).
    </p>
    <p>
        <img src="s-51.png" />
    </p>
    <p>
        However, it is also clear that Skyline is, for the most part, integrating the same peak for the same molecule in each run. To convince yourself
        there is evidence of more than just the y7 peak, you should look at the most intense measurements. You can do this using the following steps:
    </p>
    <ul>
        <li>Right-click the <b>Peak Areas</b> graph, choose <b>Normalize To</b> and click <b>None</b>.</li>
        <li>Click on the three obviously most intense peaks (which all come from the subject H_148).</li>
    </ul>
    <p>
        The chromatogram peaks you see in these three plots should be enough to convince you that the peptide causing the peak in y7 between 23 and 24
        minutes does, in fact, cause the co-eluting signals on all 6 monitored transitions when peptide abundance is high enough.
    </p>
    <p>
        <img src="s-52.png" />
    </p>
    <p>
        Rather than deleting this peptide, you will want to correct the one case where Skyline has chosen the wrong peak. You can see this in the
        <b>Retention Times</b> plot, as a narrower band occurring below 22.5 minutes.
    </p>
    <p>
        <img src="s-53.png" />
    </p>
    <p class="keep-next">
        And also in the <b>Peak Areas</b> plot, as the run with no visible integrated signal.
    </p>
    <p>
        <img src="s-54.png" />
    </p>
    <p>
        You can click on the bar in the <b>Retention Times</b> view to activate the chromatogram graph for this run. You will see that the
        run retains some very low signal on the y7 transition. You would never choose this transition alone to represent a peptide without the
        knowledge you have gained from the surrounding runs.
    </p>
    <p>
        <img src="s-55.png" />
    </p>
    <p>
        To correct the integration of this run, do the following:
    </p>
    <ul>
        <li>In the graph for D_154_REP3, click the peak annotation “23.4 (dotp 0.82)”.</li>
    </ul>
    <p>
        Before continuing your review of peak integration, do the following to return the <b>Peak Areas</b> graph to showing the relative ion
        abundances:
    </p>
    <ul>
        <li>Right-click the <b>Peak Areas</b> view, choose <b>Normalize To</b> and click <b>Total</b>.</li>
    </ul>
    <p>
        You will see the corrected peak now appears much more consistent with the other runs in the summary plots.
    </p>
    <p>
        Continue to the peptides above. At the peptide GMYESLPVVAVK, you will see such poor consistency
        in the summary plots that you should likely just delete this peptide and not bother adjusting the integration. The peptide ETGLMAFTNLK requires adjustment only in one run. The run should be obvious in the <b>Peak
            Areas</b> graph. By this point in the tutorial, you should be well equipped to understand what went wrong, and how to fix the
        problem.
    </p>
    <p>
        After making this correction, if you are still not fully convinced that the same peak for the same peptide is being integrated in all cases for
        the ETGLMAFTNLK peptide, you can quickly review all chromatogram graphs, by doing either of the following:
    </p>
    <ul>
        <li>Click on the <b>Replicates</b> dropdown list at the top of the <b>Targets</b> view.</li>
        <li>Use the up and down arrow keys to traverse the runs, activating chromatogram graphs.</li>
    </ul>
    <p>
        Or
    </p>
    <ul>
        <li>Press Ctrl-Up-Arrow or Ctrl-Down-Arrow to traverse the runs.</li>
    </ul>
    <p>
        You will see in many of the graphs that there is a peptide interfering on the y3 and y4 transitions about 1.5 minutes before the chosen peak.
        This gives you added confidence in the consistency of the integrated peak.
    </p>
    <p>
        <img src="s-56.png" />
    </p>
    <p>
        Continuing up the <b>Targets</b> list, you will find several peptides (YANVIAYDHSR and
        TDEDVPSGPPR) where the difference between the relative ion abundances of the measured peak and
        the matching library spectrum is pronounced. You can see these differences in the <b>Peak Areas</b> graph:
    </p>
    <p>
        <img src="s-57.png" />
    </p>
    <p>
        You can also observe this in the chromatogram graphs where Skyline displays the low dot product values “(dotp 0.28)” beneath the peak retention
        times.
    </p>
    <p>
        <img src="s-58.png" />
    </p>
    <p>
        Skyline displays the “dotp” value like this when there is another peak on the graph with a higher dotp value. In the case above, you cannot see
        the peak with the better dotp value because it is down near the noise level for the chosen peak. To see it, you can do the following:
    </p>
    <ul>
        <li>Click the chromatogram graph of interest to activate it.</li>
        <li>Place the mouse cursor to the left of the y-axis.</li>
        <li>Use a mouse scroll wheel or a scrolling gesture on a track pad to scroll upward.</li>
    </ul>
    <p>
        This will zoom the graph y-scale until you can see the smaller peak with a better dotp value.
    </p>
    <p>
        If you had high confidence that the library spectrum was collected on a mass spectrometer similar to the one used in this experiment, this
        would certainly cause you to suspect that the integrated peak is not actually measuring the peptide in question. In this case, however, you
        don’t really know the source of this spectrum and there is likely no other option for measuring this peptide with this sample prep.
    </p>
    <p>
        From the run-to-run consistency in the <b>Peak Areas</b> and <b>Retention Times</b> graphs, you should be confident the
        same peptide is being measured in all runs via the 6 co-eluting transitions. Even though the most abundant ion is the y3 ion with low
        selectivity, you can delete this transition to more closely examine the consistency and co-elution from the other ions (where y9 is the second
        most abundant ion). When you have convinced yourself of the quality of this peak, you can use the <b>Undo</b> button (Ctrl-Z) to
        return the deleted y3 transition to the Skyline document.
    </p>
    <p>
        Among the 7 peptides above TDEDVPSGPPR, you should see only one incorrectly integrated peak for the peptide SPQGLGASTAEISAR. This issue should now be relatively easy for you to identify and correct.
    </p>
    <p>
        This will bring you to the peptide <b><u>C</u></b>SSLLWAGAAWLR. Based on the variance you will observe in the
        summary plots (see below), you may initially be tempted to delete this peptide and move on:
    </p>
    <p>
        <img src="s-59.png" />
    </p>
    <p>
        However, if you look closely you will see that there are regions of consistency among the healthy subjects. To see this expressed more clearly,
        do the following:
    </p>
    <ul>
        <li>Right-click the <b>Peak Areas</b> graph, choose <b>Order By</b> and click <b>Document</b>.</li>
    </ul>
    <p class="keep-next">
        This will change the graphs to look like:
    </p>
    <p>
        <img src="s-60.png" />
    </p>
    <p>
        <img src="s-61.png" />
    </p>
    <p>
        Making it clearer that a consistent peak is being integrated for the healthy subjects but not for the diseased subjects. If you review the peak
        for the healthy subjects, you can see that it is fairly low intensity and mostly relies on co-elution of the y4, y5 and y6 ions. This
        co-elution is probably caused by a peptide. However, it is difficult to determine if this peptide is actually
        <b><u>C</u></b>SSLLWAGAAWLR or not.
    </p>
    <p>
        Looking at the chromatograms for the diseased subjects, you will sometimes find something that may be a peak around 21 minutes for the y6
        transition. However, often, it is difficult to observe any consistent elution pattern:
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig">
                D_102_REP3<br />
                <img src="s-62.png" />
            </td>
            <td class="comp-fig">
                D_108_REP1<br />
                <img src="s-63.png" />
            </td>
        </tr>
    </table>
    <p>
        There is, however, clearly no visible peak in the chromatograms for the diseased subjects that could produce a peak area like those peaks
        observed in the healthy subjects. With the experience you have now acquired with peak drift and truncation, you should feel confident that the
        peak is not simply drifting outside the measured range for all diseased subjects, but not healthy subjects. Randomization of injections might
        also increase your confidence in this. For the current study, however, the three replicate cycles should be enough. Once you have done your
        best to correct integration, you might also note that two of the diseased runs with more visible peaks for this peptide follow immediately
        after runs of healthy subjects (D_102_REP2 and D_102_REP3). This observation might suggest a carryover effect.
    </p>
    <p>
        <img src="s-64.png" />
    </p>
    <p>
        In the end, these transitions may represent one of the strongest biomarker candidates in this document. You could now order a synthetic peptide
        to increase confidence that you are, in fact, measuring the <b><u>C</u></b>SSLLWAGAAWLR peptide. You could also try acquiring MS/MS
        spectra across this peak (using parallel reaction monitoring – PRM) to see if it can be positively identified by a peptide search engine.
        Finally, if that failed, you could try <i>de novo</i> sequencing on the MS/MS spectra to identify what is causing the peak in the healthy
        subjects. In targeted proteomics, it is not always necessary to start with positive identification. Perhaps more similar to SDS-PAGE gel
        bands, finding a difference can initiate discovery.
    </p>
    <p>
        At this point, you should be able to make your way through the remaining peptides, correcting integration and removing poorly behaving
        peptides. This should take you less than an hour. You will find peptides that perform well and which were entirely captured by the scheduled
        retention time window need very little correction. You can often tell this at a glance from the <b>Peak Areas</b> and
        <b>Retention Times</b> graphs. For peptides that perform poorly or where many peaks were truncated or missing from the scheduling
        window, you may simply wish to delete the peptide. However, be careful not to jump to this conclusion too quickly or you might miss some
        interesting observations.
    </p>
    <h2>Preparing for statistical analysis with replicate annotations</h2>
    <p>
        Once you have completed careful validation and correction of any integration issues in your Skyline document, you will want to turn your
        attention to gaining a better understanding of what sort of differences in peptide abundance you might find between your experimental
        conditions. Making such inquiries, whether in Skyline or its external statistical tools, requires some further classification of the measured
        samples, generally referred to as “replicates” in Skyline. For such classification Skyline provides replicate annotations. In this tutorial,
        you will use three replicate annotations: SubjectId, BioReplicate and Condition.
    </p>
    <p>
        To define the SubjectId annotation, perform the following steps:
    </p>
    <ul>
        <li>On the <b>Settings</b> menu, click <b>Document Settings</b>.</li>
        <li>Click the <b>Edit List</b> button in the <b>Annotations</b> tab of the <b>Document Settings</b> form.</li>
        <li>Click the <b>Add</b> button in the <b>Define Annotations</b> form.</li>
        <li>In the <b>Name</b> field of the <b>Define Annotation</b> form enter ‘SubjectId’.</li>
        <li>In the <b>Applies to</b> list, check <b>Replicates</b>.</li>
    </ul>
    <p class="keep-next">
        The form should look like this:
    </p>
    <p>
        <img src="s-65.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button in the <b>Define Annotation</b> form.</li>
        <li>Click the <b>OK</b> button in the <b>Define Annotations</b> form.</li>
        <li>Check the ‘SubjectId’ annotation you just created.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Settings</b> form should look like:
    </p>
    <p>
        <img src="s-66.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p>
        Although this tutorial will not go so far as to cover statistical analysis methods available in the Skyline External Tool “MSstats”, it is well
        suited to such analysis. This dataset has been used to demonstrate the use of MSstats in many courses and workshops. If you are interested in
        exploring what MSstats has to offer for this type of analysis, you can get the two other annotations it requires by installing it into Skyline
        as follows:
    </p>
    <ul>
        <li>On the <b>Tools</b> menu, click <b>Tool Store</b>.</li>
        <li>Select ‘MSstats’ from the list of tools.</li>
    </ul>
    <p class="keep-next">
        The <b>Install from Tool Store</b> form should look like:
    </p>
    <p>
        <img src="s-67.png" />
    </p>
    <ul>
        <li>Click the <b>Install</b> button and follow the prompts until MSstats installation is completed.</li>
    </ul>
    <p>
        Perform the following steps to see the annotations added during MSstats installation or to add them directly without MSstats installation:
    </p>
    <ul>
        <li>On the <b>Settings</b> menu, click <b>Document Settings</b>.</li>
        <li>Click the <b>Edit List</b> button in the <b>Annotations</b> tab of the <b>Document Settings</b> form.</li>
        <li>If you installed MSstats, select the BioReplicate annotation, and click the <b>Edit</b> button. Otherwise, click the
            <b>Add</b> button.</li>
        <li>Repeat for the Condition annotations.</li>
    </ul>
    <p class="keep-next">
        These annotation definitions should look like:
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig" style="padding: 5px;">
                <img src="s-68.png" />
            </td>
            <td class="comp-fig" style="padding: 5px;">
                <img src="s-69.png" />
            </td>
        </tr>
    </table>
    <ul>
        <li>Click <b>OK</b> to return to the <b>Document Settings</b> form.</li>
        <li>Check the BioReplicate and Condition annotations to add them to your document.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Settings</b> form should look like this:
    </p>
    <p>
        <img src="s-70.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p>
        To set the annotations you have added to the document, do the following:
    </p>
    <ul>
        <li>On the <b>View</b> menu, click <b>Document Grid</b>.</li>
        <li>On the <b>Reports</b> menu, in the upper left corner of the <b>Document Grid</b>, click <b>Replicates</b>.
        </li>
    </ul>
    <p class="keep-next">
        The <b>Document Grid</b> should look something like this:
    </p>
    <p>
        <img src="s-71.png" />
    </p>
    <p>
        You can now manually enter the annotations for all 42 replicates into this grid. You can also paste values from a spreadsheet directly into
        this form. You can do this now by performing the following steps:
    </p>
    <ul>
        <li>In Excel, open the “Annotations.xlsx” file in the “GroupedStudies1\Heart Failure\raw” folder.</li>
        <li>Select all 42 rows of value cells beneath the header row for the 3 annotation columns.</li>
        <li>Press Ctrl-C to copy the cells.</li>
        <li>In Skyline select the top cell in the SubjectId column.</li>
    </ul>
    <p>
        <b>Note</b>: Avoid entering cell edit mode on this step. If you see a blinking cursor inside the top SubjectId cell, press the Esc
        key.
    </p>
    <p class="keep-next">
        The <b>Document Grid</b> should look like this:
    </p>
    <p>
        <img src="s-72.png" />
    </p>
    <ul>
        <li>Press Ctrl-V to paste into the <b>Document Grid</b>.</li>
    </ul>
    <p class="keep-next">
        Skyline should fill in the cells in the <b>Document Grid</b> with the values from the spreadsheet as follows:
    </p>
    <p>
        <img src="s-73.png" />
    </p>
    <h2>Annotating Peptides with Problem Peaks</h2>
    <p>
        You may also wish to annotate the peptides that either had peaks removed or truncated peaks, as both of these may cause issues during
        statistical analysis. To define the annotation you will use for this purpose, perform the following steps:
    </p>
    <ul>
        <li>On the <b>Settings</b> menu, click <b>Document Settings</b>.</li>
        <li>Click the <b>Edit List</b> button in the <b>Annotations</b> tab of the <b>Document Settings</b> form.</li>
        <li>Select the last annotation in the <b>Annotations</b> list.</li>
        <li>Click the <b>Add</b> button.</li>
        <li>In the <b>Name</b> field of the <b>Define Annotation</b> form, enter “MissingData”.</li>
        <li>From the <b>Type</b> dropdown list, choose “True/False”.</li>
        <li>In the <b>Applies To</b> list, check <b>Peptides</b>.</li>
    </ul>
    <p class="keep-next">
        The <b>Define Annotation</b> form should look like:
    </p>
    <p>
        <img src="s-74.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
        <li>Click the <b>OK</b> button in the <b>Define Annotations</b> form.</li>
        <li>Check the “MissingData” annotation you just defined.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Settings</b> form should look like:
    </p>
    <p>
        <img src="s-75.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p>
        To prepare for setting your new ‘MissingData’ annotation on all peptides with truncated peaks, do the following:
    </p>
    <ul>
        <li>In the <b>Document Grid</b>, on the <b>Reports</b> menu, click the “Truncated Precursors” report you created before.
        </li>
        <li>On the same <b>Reports</b> menu, click <b>Edit Report</b>.</li>
        <li>On the <b>Customize Report</b> form, expand the <b>Columns</b> tree to check Proteins &gt; Peptides &gt; MissingData,
            to add a column for the annotation you just created.</li>
    </ul>
    <p class="keep-next">
        The <b>Customize Report</b> form should look like this:
    </p>
    <p>
        <img src="s-76.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
        <li>Click the “Count Truncated” column header and click <b>Sort Descending</b>.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Grid</b> should look like this:
    </p>
    <p>
        <img src="s-77.png" />
    </p>
    <p>
        Depending on how you processed the rest of your document, there may or may not be exactly 223 truncated precursor peak groups. You could check
        the “MissingData” check boxes one peptide at a time. Once you check a box for any peptide, the rest will check automatically, because the
        annotations apply only once to any peptide. Try this now by doing the following:
    </p>
    <ul>
        <li>Check the checkbox in the first row of the grid, immediately under the “MissingData” header.</li>
        <li>Press the Enter key.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Grid</b> should now look like this:
    </p>
    <p>
        <img src="s-78.png" />
    </p>
    <p>
        You can also use Excel and the <b>Document Grid</b> support for pasting values to set all 163 rows in a single paste as follows:
    </p>
    <ul>
        <li>In a blank spreadsheet in Excel, scroll down to row 163.</li>
        <li>Enter “TRUE” in the first column of row 163.</li>
        <li>Select the cell you just entered.</li>
        <li>Click-and-drag the range extension square in the lower right corner of the selection rectangle to fill cells in the first column from row
            163 to 1 with the value “TRUE”.</li>
        <li>Press Ctrl-C to copy these cells.</li>
        <li>Return to Skyline and select the first cell in the “MissingData” column.</li>
        <li>Press Ctrl-V to paste ‘TRUE’ values into all 163 rows from “MissingData”.</li>
    </ul>
    <p>
        You may realize that this seems a little redundant. You only need to set the MissingData annotation to true once for every peptide. However,
        doing this achieves the desired result more quickly than by clicking a check box for each of the 31 peptides with truncated peaks.
    </p>
    <p>
        Once you have performed the steps above, you should notice that many of the peptides at the top of the <b>Targets</b> list are now
        marked with a small red triangle above and right of the peptide name.
    </p>
    <p>
        <img src="s-79.png" />
    </p>
    <p>
        If you hover the mouse cursor over one of these triangles, you will see a tip appear with the text “<b>Missing Data</b> True”.
    </p>
    <p>
        Now repeat this process for all of the peptides where you removed the peak entirely instead of integrating a truncated peak by doing the
        following:
    </p>
    <ul>
        <li>On the <b>Reports</b> menu of the <b>Document Grid</b>, click <b>Manage Reports</b>.</li>
        <li>Select the “Truncated Precursors” report you have been using.</li>
        <li>Click the <b>Copy</b> button.</li>
        <li>In the <b>Report Name</b> field of the <b>Customize Report</b> form, enter “Missing Peaks”.</li>
        <li>Remove the <b>Precursors</b> and <b>Count Truncated</b> columns using the <img src="../../shared/red-x-icon.png" /> button to the
            right of the columns list.</li>
        <li>Check the <b>Pivot Replicate Name</b> checkbox.</li>
    </ul>
    <p class="keep-next">
        The <b>Customize Report</b> form should look like this:
    </p>
    <p>
        <img src="s-80.png" />
    </p>
    <ul>
        <li>Click the <b>Filter</b> tab.</li>
        <li>Click the <img src="../../shared/red-x-icon.png" /> button to remove the existing filter on <b>Count Truncated</b>.</li>
        <li>Expand the columns tree to check Proteins &gt; Peptides &gt; Precursors &gt; Precursor Results, to add a filter for the <b>Total Area</b> column.</li>
        <li>In the <b>Operation</b> column, choose “Is Blank”.</li>
    </ul>
    <p class="keep-next">
        The <b>Customize Report</b> form should look like this:
    </p>
    <p>
        <img src="s-81.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
        <li>Click the <b>OK</b> button in the <b>Manage Reports</b> form.</li>
        <li>On the <b>Reports</b> menu in the <b>Document Grid</b>, click <b>Missing Peaks</b>.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Grid</b> should look like the following:
    </p>
    <p>
        <img src="s-82.png" />
    </p>
    <p>
        <b>NOTE</b>: If you did not fully process your document, you may instead see only one line for the peptide GSYNLQDLLAQ. The next note
        explains how to open a fully processed document.
    </p>
    <p>
        In this report, you get a simple list of peptides with a missing peak for any run. You can easily see that there are 10 peptides. Only 2 of
        these do not also have truncated peaks and 8 already have the “MissingData” annotation set. The replicate names with missing peaks are pivoted
        to the right of the <b>Peptide</b> and <b>MissingData</b> columns. To finish labeling all peptides with missing data:
    </p>
    <ul>
        <li>Check the <b>MissingData</b> check boxes for TLNSINIAVFSK and IFSQQADLSR.</li>
        <li>Close the <b>Document Grid</b>.</li>
    </ul>
    <p>
        Congratulations! You have completed initial data processing on this data set. After gaining a clearer understanding of issues that may impact
        data quality and optimizing peak integration to the extent possible for the acquired data, you are ready to begin running higher level
        statistics to further assess data quality. Hopefully, you will begin to get a sense for which peptides or proteins may actually prove useful
        as biomarkers for the conditions under study.
    </p>
    <p>
        <b>NOTE</b>: If you have not fully processed your document to this point in the tutorial or you would like to compare your processing
        to that of the tutorial author, you can open the fully processed file included with this tutorial as follows:
    </p>
    <ul>
        <li>On the <b>File</b> menu, click <b>Save</b> (Ctrl-S) to save your document.</li>
        <li>On the <b>File</b> menu, click <b>Open</b> (Ctrl-O).</li>
        <li>Navigate up one folder to “GroupedStudies1\Heart Failure” and select “Rat_plasma.sky.zip”.</li>
        <li>Click the <b>Open</b> button.</li>
    </ul>
    <h2>Initial multi-replicate inspection in Skyline</h2>
    <p>
        While you will certainly want to do a more in-depth statistical analysis of any data set like this than Skyline currently offers, Skyline does
        provide some useful ways to perform an initial inspection of variance and group mean averages. To gain some insight into variance among the
        technical replicates for each subject, do the following:
    </p>
    <ul>
        <li>Select the peptide DVFSQQADLSR just above the global internal standard peptide list.</li>
        <li>On the <b>View</b> menu, choose <b>Transitions</b> and click <b>Total</b>.</li>
        <li>Right-click the <b>Peak Areas</b> view, choose <b>Normalize To</b> and click <b>Global Standards</b>.</li>
        <li>Right-click the <b>Peak Areas</b> view, choose <b>Group By</b> and click <b>SubjectId</b>.</li>
        <li>Right-click the <b>Peak Areas</b> view, and click <b>CV Values</b> if it is not already checked.</li>
    </ul>
    <p class="keep-next">
        This should leave the <b>Peak Areas</b> graph looking something like:
    </p>
    <p>
        <img src="s-83.png" />
    </p>
    <p>
        You can now select through several more peptides to review the coefficients of variation (CV) among the technical replicates for each subject.
        You will see that most are below 25%. Ideally, however, you would prefer to see lower CVs than in the ones observed in this experiment. It is
        worth noting that a lower CV derived from 3 measurements is more likely than the same CV derived from 10 measurements. This is due to a
        statistical tendency to underestimate standard deviations with small sample sizes.
    </p>
    <p>
        Finally, you can use Skyline to get some initial insight into the difference in peptide expression between the two groups under study in this
        experiment. Perform the following steps to make a preliminary review of the differences in average peptide abundance between the healthy and
        diseased groups:
    </p>
    <ul>
        <li>Right-click the <b>Peak Areas</b> view, choose <b>Group By</b> and click <b>Condition</b>.</li>
        <li>Right-click the <b>Peak Areas</b> view, and click <b>CV Values</b> to uncheck this menu item.</li>
        <li>Select various peptides in the <b>Targets</b> view.</li>
    </ul>
    <p class="keep-next">
        You will see graphs in the <b>Peak Areas</b> view like the following for the peptides of the protein NP_872280:
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig">
                <img src="s-84.png" />
            </td>
            <td class="comp-fig">
                <img src="s-85.png" />
            </td>
            <td class="comp-fig">
                <img src="s-86.png" />
            </td>
        </tr>
    </table>
    <p>
        In these graphs, the bars represent the mean average value (in this case, peak area ratios to global standards) across all replicates and the
        whiskers show one standard deviation to either side of the mean. This gives you some idea of the sample distributions from which the means
        were derived.
    </p>
    <p>
        In interpreting this graph, it is important to understand your goal in measuring differences across groups of samples. The two most common
        goals are:
    </p>
    <ol>
        <li>Differential: attempting to identify differences between groups. (e.g. Testing whether a treatment causes reduction in activity along some
            pathway of interest.)</li>
        <li>Prediction: attempting to identify peptide expression that may be predictive of belonging to a particular group. (e.g. Testing whether a
            set of peptides might be used as biomarkers for inclusion in a certain group.)</li>
    </ol>
    <p>
        If you want to simply detect a statistically significant difference between two groups, you would be more interested in the standard errors of
        the mean values than the standard deviations of the distributions. The p value from a T-test gives you a single number expressing the
        statistical significance of a difference in means related to the standard error values for those two means. The graphs above give you no
        insight into the standard error values. Therefore, they are not as useful for gaining insight on differential expression as they could be.
    </p>
    <p>
        For prediction, the underlying distributions of the two populations are important. When you see a great deal of overlap between the two groups
        at just one standard deviation, the peptide alone is likely not a strong biomarker candidate. In that case, it would be hard to predict which
        distribution produced any single measurement. It is important to note here that it may still be possible to create a strong “biomarker panel”
        from a collection of peptides which are not individually predictive.
    </p>
    <p>
        The peptides in the graphs above have statistically significant differences in means but are not individually predictive. This is illustrated
        by the overlap in distribution ranges at one standard deviation.
    </p>
    <p class="keep-next">
        Continuing up the list of peptides / proteins, you will see that the protein immediately above, NP_036714, is a much stronger
        candidate for use as a biomarker:
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig">
                <img src="s-87.png" />
            </td>
            <td class="comp-fig">
                <img src="s-88.png" />
            </td>
        </tr>
    </table>
    <p>
        Many of the peptides in this experiment show statistically significant differences in the means between the two groups. This should not be
        surprising given the source of our target list was the literature on differential protein expression in heart disease. Far fewer, however,
        appear to be strong biomarker candidates on their own merit.
    </p>
    <p>
        When you reach the protein NP_001007697, which contains the peptide <b><u>C</u></b>SSLLWAGAAWLR, you will observe a case where the
        peptides assigned to a single protein experience very different relative expression levels between healthy and diseased groups:
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig">
                <img src="s-89.png" />
            </td>
            <td class="comp-fig">
                <img src="s-90.png" />
            </td>
        </tr>
    </table>
    <p>
        This should act as a reminder that while you can achieve some confidence in measuring the same peptide molecule over many samples, you will
        often be far less confident in assigning multiple peptide precursors to the same source protein form. The reasons for an observation of this
        nature are numerous. For instance, a peptide may have a post-translational modification (PTM) that makes it an important biomarker candidate.
        Whereas, the rest of the protein may not be affected.
    </p>
    <h2>Simple group comparisons in Skyline</h2>
    <p>
        It is also possible with Skyline to perform simple pairwise group comparisons of peptide or protein peak areas. The comparisons are performed
        by summing the available transition peak areas for a peptide or protein, optionally applying a normalization, taking the log, averaging any
        technical replicates, and performing a T-test on the resulting values. Skyline automatically discards replicates with missing values, or
        truncated peaks in label-free data.
    </p>
    <p>
        To try this now on the data set you have been processing, perform the following steps:
    </p>
    <ul>
        <li>On the <b>Settings</b> menu, click <b>Document Settings</b>.</li>
        <li>Click the <b>Add</b> button in the <b>Group Comparisons</b> tab of the <b>Document Settings</b> form.</li>
        <li>In the <b>Name</b> field of the <b>Edit Group Comparison</b> form, enter “Healthy v. Diseased”.</li>
        <li>For the <b>Control group annotations</b> choose “Condition”.</li>
        <li>For the <b>Control group value</b> choose “Healthy”.</li>
        <li>For the <b>Value to compare against</b> choose “Diseased”.</li>
        <li>For the <b>Identity annotation</b> choose “SubjectId”.</li>
        <li>For the <b>Normalization method</b> choose “Ratio to Global Standards”.</li>
        <li>In the <b>Confidence level</b> field, enter “99” %.</li>
        <li>For <b>Scope</b> choose <b>Protein</b>.</li>
    </ul>
    <p class="keep-next">
        The <b>Edit Group Comparison Form</b> should look like this:
    </p>
    <p>
        <img src="s-91.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Settings</b> form should look like this:
    </p>
    <p>
        <img src="s-92.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p>
        To inspect the group comparison you just defined do the following:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Group Comparisons</b> and click <b>Health v. Diseased</b>.</li>
    </ul>
    <p class="keep-next">
        Skyline will show a grid view that looks like this:
    </p>
    <p>
        <img src="s-93.png" />
    </p>
    <p>
        If the “Fold Change Result” column does not show the full confidence interval, double-click the vertical line between the Fold Change Result
        and Adjusted P-Value headers.
    </p>
    <p>
        To see the logged fold change values and confidence intervals graphed:
    </p>
    <ul>
        <li>Click the <b>Bar Graph</b> button in the menu above the toolbar in the grid view.</li>
    </ul>
    <p class="keep-next">
        Skyline adds a graph pane beside the <b>Healthy v. Diseased:Grid</b> view that looks like this:
    </p>
    <p>
        <img src="s-94.png" />
    </p>
    <ul>
        <li>On the grid, click the “Fold Change Result” header, and click <b>Sort Ascending</b> in the menu that appears.</li>
    </ul>
    <p>
        This will sort the proteins in the grid and in the graph. Notice that many of the confidence interval whiskers cross the zero line, indicating
        that at 99% confidence it would not be unusual for the observed data to occur by random chance. And, this is not even with any correction for
        multiple hypothesis testing.
    </p>
    <p>
        To set a cut-off based on the Benjamini-Hochberg adjusted p-value, which estimates false discovery rate (FDR), do the following in the grid
        view:
    </p>
    <ul>
        <li>Click the “Adjusted P-Value” header and click <b>Filter</b>.</li>
        <li>In the <b>Filter Type</b> field, choose “Is Less Than”.</li>
        <li>In the field below this, enter “0.01”.</li>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p class="keep-next">
        You will see the number of rows indicated in the grid toolbar drop from 48 to 11, and the graph should now look like this:
    </p>
    <p>
        <img src="s-95.png" />
    </p>
    <p>
        Note that more of the proteins that displayed significant change have positive fold-change means, indicating they were more intense in the
        diseased subjects. It is, however, important to consider the impact that failing to randomize sample order may have caused. Because in all
        three cycles of measurement, the diseased subjects preceded the healthy subjects, you should expect that any degradation not accounted for in
        your normalization will cause fold change to appear as up-regulation in the diseased group.
    </p>
    <h2>Correctly specifying technical replicates</h2>
    <p>
        If you collect technical replicates, as in this experiment, it is extremely important to your statistical inference that you specify them
        correctly in the <b>Edit Group Comparison</b> form. If you do not, then each measurement will be considered as coming from a
        distinct biological subject, incorrectly narrowing your standard errors and, hence, confidence intervals, and artificially lowering p values.
    </p>
    <p>
        To see this in action, try the following:
    </p>
    <ul>
        <li>Click the <b>Change Settings</b> text in the menu above the toolbar in the grid view. </li>
        <li>For the <b>Identity annotation</b> choose the empty element at the top.</li>
    </ul>
    <p>
        You will see the grid view show 37 rows (and the graph 37 bars) now with adjusted p values less than 0.01. This is because the statistics are
        now calculated as if you had 42 distinct subjects, instead of 14 with 3 measurements each. This is an important distinction, and as you can
        see making a mistake here could yield embarrassingly over-optimistic statistical inference. In this case, the 11 proteins showing
        statistically significant differences in means between healthy and diseased groups, at an estimated 1% FDR, has ballooned to 37 proteins, even
        including the “S” protein containing the 3 standard peptides injected at a constant concentration.
    </p>
    <h2>Refinement by differentiation</h2>
    <p>
        Returning to the original goal of this experiment, you can now reduce the document to only a subset of peptides with statistically significant
        differences in group means, at an estimated 1% FDR. Though, keep in mind that this data set may overstate the significance of up-regulation in
        the diseased group, due to the lack of randomization (diseased always measured before healthy) and systematic signal degradation over the
        runs.
    </p>
    <p>
        Nonetheless, to reduce the targets list in this document to the set of peptides with differences in group means at an estimated 1% FDR, perform
        the following steps:
    </p>
    <ul>
        <li>On the <b>Healthy v. Diseased:Settings</b> form, change the <b>Identity annotation</b> field back to “SubjectId”.</li>
        <li>On the <b>File</b> menu, click <b>Save As</b>.</li>
        <li>Save the document to the name “Rat_plasma-diff.sky”.</li>
        <li>In the <b>Healthy v. Diseased:Settings</b> form, choose the <b>Peptide</b> option in the <b>Scope</b> box.
        </li>
        <li>Click the “Adjusted P-Value” header and click <b>Filter</b>.</li>
        <li>In the <b>Filter Type</b> field, choose “Is Greater Than or Equal To”.</li>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p class="keep-next">
        The <b>Healthy v. Diseased:Grid</b> should show 92 rows above the 1% FDR cut-off, and the graph should now look
        like this:
    </p>
    <p>
        <img src="s-96.png" />
    </p>
    <p class="keep-next">
        To delete these peptides from the document do the following:
    </p>
    <ul>
        <li>Click the gray rectangle in the upper left corner of the <b>Healthy v. Diseased:Grid</b>, where the row and column headers meet,
            to select all cells.</li>
        <li>Scroll down the grid until you find the VVLSGSDATLAYSAFK peptide in the “S” protein.</li>
        <li>Hold down the Ctrl key and click on the row header for this row to deselect it.</li>
    </ul>
    <p class="keep-next">
        The grid should look like this:
    </p>
    <p>
        <img src="s-97.png" />
    </p>
    <p>
        Note that the VVLSGSDATLAYSAFK peptide has a fold-change of 1, with a 99% confidence interval of 1 to 1. This is the single peptide used as the
        Global Standard for normalization. You do not want to delete this peptide from the document. That is why you deselected it.
    </p>
    <ul>
        <li>Click the <img src="../../shared/red-x-icon.png" /> button in the grid toolbar.</li>
    </ul>
    <p class="keep-next">
        Skyline will show the following message to confirm you really want to delete the selected rows:
    </p>
    <p>
        <img src="s-98.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p>
        You can now close the group comparison windows and review the 34 peptides remaining in the document. They should all have pronounced
        differences in means between the two groups, though their distributions will not always be disjoint even at only one standard deviation. That
        is, you will see the standard deviation whiskers overlapping in the <b>Peak Areas</b> graph that should still be showing the peak
        areas grouped by condition.
    </p>
    <h1>Conclusion</h1>
    <p>
        In this tutorial, you learned some of the most effective techniques for visual assessment and correction of the peak integration that Skyline
        automatically performs on all imported LC-MS/MS chromatography data. Although you only worked with an unlabeled SRM data set, the techniques
        you applied in this tutorial would apply equally well to data from other chromatography-base quantification methods. The manual nature of this
        inspection and correction makes it best suited for data sets containing not much more than two thousand peptides. In this case, 137 peptides
        took about 1 hour to process, which means a data set with 2,000 peptides might take upwards of 2 days of effort.
    </p>
    <p>
        For much larger data sets you might want to first perform some very rough screening of peptides based on their initial promise in showing some
        difference between the conditions you are studying. In that case, you only spend time on this type of manual inspection and correction for
        potential peptides of interest. You should now be able to achieve this kind of filtering with the approach you learned using Skyline group
        comparison support. You may want to relax the limit on adjusted p value (or FDR cut-off) to something less stringent than the 1% used in this
        tutorial, but the approach would be similar.
    </p>
    <p>
        You have now seen how this type of processing and expert use of the features Skyline provides can help you both understand and correct
        potential sources of error in your quantitative experiments. You have also dealt first-hand with mis-assigned peaks, interference, peak
        truncation, peptides at abundance levels that are difficult to detect, 200-fold peptide signal degradation, and doubly eluting peptides. With
        a mastery of the tools Skyline provides for understanding, correcting, and annotating these issues, you should be able to reduce error in your
        quantitative targeted proteomics experiments and get to biological insights more effectively.
    </p>
    <h1>Bibliography</h1>
    <p class="bibliography">
        1. Bereman, M. S., MacLean, B., Tomazela, D. M., Liebler, D. C. &amp; MacCoss, M. J. The development of selected reaction
        monitoring methods for targeted proteomics via empirical refinement. <i>PROTEOMICS</i> <b>12</b>, 1134–1141 (2012).
    </p>
    <p class="bibliography">
        2. Zhang, H. <i>et al</i>. Methods for Peptide and Protein Quantitation by Liquid Chromatography-Multiple Reaction
        Monitoring Mass Spectrometry. <i>Mol. Cell. Proteomics MCP</i> <b>10</b>, (2011).
    </p>
</body>

</html>