<html>

<head>
    <meta charset="utf-8">
    <link rel="stylesheet" type="text/css" href="../../shared/SkylineStyles.css">
    <script src="../../shared/skyline.js" type="text/javascript"></script>
</head>

<body onload="skylineOnload();">
    <h1 class="document-title">Comparing PRM, DIA, and DDA in Skyline </h1>
    <p>
        In this tutorial, you will perform a targeted analysis in Skyline with PRM, DIA and DDA measurement data.
    </p>
    <p>
        The four samples used for this tutorial were generated by mixing HeLa protein digest with PROCAL peptide standard<sup><a href="#footnote" id="footnote-ref">1</a></sup> at defined
        concentrations. Each sample was measured with three technical replicate injections in PRM, DIA and DDA mode on a Quadrupole-Orbitrap-Iontrap
        (Tribrid) mass spectrometer (Eclipse, Thermo).
    </p>
    <p>
        <img src="image-figure1.png" />
    </p>
    <p>
        <strong>Credits:</strong> The samples for this tutorial were measured by the BayBioMS@MRI at TU Munich. The tutorial was written by Julia
        Mergner and Christina Ludwig.
    </p>
    <h1>Getting Started</h1>
    <p>
        To start this tutorial, download the following ZIP file:
    </p>
    <p>
        <a href="https://skyline.ms/tutorials/AcquisitionComparisonMzml.zip">https://skyline.ms/tutorials/AcquisitionComparisonMzml.zip</a>
    </p>
    <p>
        Extract the files in it to a folder on your computer, like:
    </p>
    <p>
        C:\Users\brendanx\Documents
    </p>
    <p>
        This will create a new folder:
    </p>
    <p>
        C:\Users\brendanx\Documents\AcquisitionComparisonMzml
    </p>
    <p>
        If you have been using Skyline prior to starting this tutorial, it is a good idea to revert Skyline to its default settings. To do so: 
    </p>
    <ul>
        <li>Start Skyline.</li>
        <li>On the <strong>Start Page, </strong>click <strong>Blank Document </strong>which looks like this: </li>
    </ul>
    <p>
        <img src="../../shared/en/skyline-blank-document.png" />
    </p>
    <ul>
        <li>On the <strong>Settings</strong> menu, click <strong>Default. </strong></li>
        <li>Click <strong>No </strong>on the form asking if you want to save the current settings.</li>
    </ul>
    <p>
        The document settings in this instance of Skyline have now been reset to the default.
    </p>
    <p>
        Since this tutorial covers a proteomics topic, you can choose the proteomics interface by doing the following:
    </p>
    <ul>
        <li>Click the user interface control in the upper right-hand corner of the Skyline window, and click <strong>Proteomics interface
            </strong>which looks like this: </li>
    </ul>
    <p>
        <img src="../../shared/en/proteomics-interface.png" />
    </p>
    <p>
        Skyline is operating in proteomics mode which is displayed by the protein icon <img src="../../shared/protein-icon.png" /> in the upper
        right-hand corner of the Skyline window.
    </p>
    <h1>Skyline ‚Äì Loading Libraries</h1>
    <h2>Loading an experimental spectral library file</h2>
    <p>
        To start this tutorial, you will first load a previously generated experimental spectral library called PROCAL.blib. This library file contains
        experimental spectra for all the 41 PROCAL peptides (proteome tools project, Zolg et al., 2017), which are the target peptides of this
        tutorial. The PROCAL.blib file was generated with Skyline by loading a DDA measurement file of the PROCAL peptides together with the MaxQuant
        search result file (msms.txt).
    </p>
    <p>
        To load this experimental spectra library file:
    </p>
    <ul>
        <li>On the <strong>Settings</strong> menu, click <strong>Peptide Settings</strong>.</li>
        <li>Click the <strong>Library</strong> tab.</li>
        <li>Click the <strong>Edit list</strong> button.</li>
        <li>In the <strong>Edit Libraries</strong> window, click the <strong>Add</strong> button.</li>
        <li>In the <strong>Name</strong> field, enter "PROCAL".</li>
        <li>Click the <strong>Browse</strong> button and navigate to the <strong>Tutorial\02_PRM</strong> subfolder.</li>
        <li>Open the ‚Äú<strong>PROCAL.blib</strong>‚Äù file.</li>
        <li>Click <strong>Ok</strong> in the <strong>Edit Library</strong> and <strong>Edit Libraries</strong> window.</li>
    </ul>
    <p>
        Skyline has now added ‚Äú<strong>PROCAL</strong>‚Äù to the <strong>Libraries</strong> list in the <strong>Library</strong> tab of the
        <strong>Peptide Settings</strong> window.
    </p>
    <ul>
        <li>Check the box for ‚Äú<strong>PROCAL-Library</strong>‚Äù to tell Skyline to use this library in picking peptides and transitions.</li>
    </ul>
    <p class="keep-next">
        The <strong>Peptide Settings - Library</strong> tab should now look like this:
    </p>
    <p>
        <img src="s-01.png" />
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>
        You can explore the loaded PROCAL library:
    </p>
    <ul>
        <li>On the <strong>View</strong> menu, choose <strong>Libraries</strong>, and click <strong>Library Explorer</strong>.</li>
    </ul>
    <p>
        A <strong>Spectral Library Explorer</strong> window will open, which will show you all peptides entailed in the loaded library file (for the
        PROCAL library it is 44 peptides in total), their corresponding spectra as well as raw file and retention time information:
    </p>
    <p>
        <img src="s-02.png" />
    </p>
    <ul>
        <li>Click the <strong>Close</strong> button to close the <strong>Spectral Library Explorer</strong> when you are done.</li>
    </ul>
    <h2>Inserting FASTA sequences</h2>
    <p>
        To specify your proteins of interest you can import a FASTA file containing exclusively your target proteins via the <strong>Import</strong>
        function.
    </p>
    <ul>
        <li>On the <strong>File</strong> menu, choose <strong>Import</strong> and click <strong>FASTA</strong>.</li>
        <li>Navigate to the <strong>1_PRM</strong> subfolder and select the "<strong>PROCAL.fasta</strong>" file.</li>
        <li>Click the <strong>Open</strong> button.</li>
    </ul>
    <p class="keep-next">
        The Skyline main window should now look like this:
    </p>
    <p>
        <img src="s-03.png" />
    </p>
    <ul>
        <li>In case no <strong>Library Match</strong> window with spectrum information is visible, on the <strong>View</strong> menu, choose
            <strong>Libraries</strong>, and click <strong>Library Match</strong> (Alt-1).</li>
        <li>Right-click the <strong>Library Match</strong> window, choose <strong>Ion Types</strong>, and make sure that the ion types<strong> Y
            </strong>and<strong> B</strong> selected.</li>
    </ul>
    <p>
        Skyline now highlights the y-ions in blue and the b-ions in purple in the spectrum graph.
    </p>
    <p>
        By default, Skyline chooses the <em>3 most intense singly-charged product y-ions</em> as the transitions it will target for doubly charged
        precursors. This selection can be seen when opening a peptide in the <strong>Targets</strong> window by clicking on the box with "+" in front
        it:
    </p>
    <p>
        <img src="s-04.png" />
    </p>
    <p>
        To change the product ion selection setting from the default values, perform the following steps:
    </p>
    <ul>
        <li>On the <strong>Settings</strong> menu, click <strong>Transition Settings</strong>.</li>
        <li>Click the <strong>Filter</strong> tab.</li>
        <li>In the <strong>Precursor charges</strong> and <strong>Ion charges</strong> fields confirm the values are ‚Äú2‚Äù and ‚Äú1‚Äù, respectively.</li>
        <li>In the <strong>Ion types</strong> field, change from ‚Äúy‚Äù to ‚Äúp,y,b‚Äù (precursor, y-ions, b-ions).</li>
        <li>In the <strong>Product ion selection</strong> field change <strong>From</strong> to ‚Äú<strong>ion 2</strong>‚Äù and <strong>To</strong> to
            ‚Äú<strong>last ion</strong>‚Äù.</li>
        <li>Uncheck ‚Äú<strong>N-terminal to Proline</strong>‚Äù.</li>
    </ul>
    <p class="keep-next">
        The <strong>Transition Settings ‚Äì Filter </strong>tab should look like this:
    </p>
    <p>
        <img src="s-05.png" />
    </p>
    <ul>
        <li>Click the <strong>Library</strong> tab.</li>
        <li>Check that the <strong>Ion match tolerance</strong> field is set to the default ‚Äú0.5‚Äù.</li>
        <li>In the <strong>Pick</strong> field, enter ‚Äú6‚Äù for <strong>product ions</strong> and <strong>minimum product ions</strong>.</li>
    </ul>
    <p class="keep-next">
        The <strong>Transition Settings - Library</strong> tab should look like this:
    </p>
    <p>
        <img src="s-06.png" />
    </p>
    <ul>
        <li>Click the <strong>Instrument</strong> tab.</li>
        <li>Change the <strong>Min m/z</strong> field to ‚Äú350‚Äù.</li>
    </ul>
    <p class="keep-next">
        The <strong>Transition Settings - Instrument</strong> tab should look like this:
    </p>
    <p>
        <img src="s-07.png" />
    </p>
    <ul>
        <li>Select the <strong>Full-Scan</strong> tab.</li>
        <li>Change the <strong>Isotope peaks included</strong> field to ‚ÄúCount‚Äù.</li>
        <li>Change the <strong>Precursor mass analyzer</strong> field to ‚ÄúCentroided‚Äù.</li>
        <li>Check the <strong>Mass Accuracy</strong> field is by default set to ‚Äú10‚Äù ppm.</li>
    </ul>
    <p class="keep-next">
        The <strong>Transition Settings - Full-Scan</strong> tab should look like this:
    </p>
    <p>
        <img src="s-08.png" />
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p class="keep-next">
        The Skyline <strong>Targets</strong> tree should now look like this:
    </p>
    <p>
        <img src="s-09.png" />
    </p>
    <p>
        Skyline has picked the top 6 ranked product ions for each peptide precursor, including b-ions, based on the spectral library file
        (‚Äú<strong>PROCAL.blib</strong>‚Äù) you provided in the first step of this tutorial, as well as 3 precursor isotopes (irank 1,2,3) for each
        peptide.
    </p>
    <h2>Creating a predicted spectral library using Koina</h2>
    <p>
        Skyline is not limited to using just a single spectral library. You can for example use the Koina prediction framework to predict spectra for
        <strong>unmodified</strong> peptides directly in Skyline.
    </p>
    <ul>
        <li>On the <strong>Settings</strong> menu, click <strong>Peptide Settings</strong>.</li>
        <li>Click the <strong>Library</strong> tab.</li>
        <li>Click the <strong>Build</strong> button.</li>
        <li>In the <strong>Name</strong> field of the <strong>Build Library</strong> window, enter ‚Äú<strong>PROCAL-Koina</strong>‚Äù.</li>
        <li>Click the <strong>Browse</strong> button.</li>
        <li>Navigate to the ‚Äú1_PRM‚Äù subfolder.</li>
        <li>Click the <strong>Save</strong> button.</li>
        <li>In the <strong>Data source</strong> field select <strong>Koina</strong>.</li>
        <li>In the Pop-up window click <strong>Yes</strong> to confirm that you want to change the Koina settings now. If there is no Pop-up window
            click on Koina - <strong>Info/Settings</strong>.</li>
        <li>In the <strong>Intensity model</strong> field of the <strong>Options</strong> form select ‚ÄúProsit_2020_intensity_HCD‚Äù.</li>
        <li>In the <strong>NCE</strong> field, select ‚Äú31‚Äù.</li>
    </ul>
    <p class="keep-next">
        The <strong>Options</strong> form should look like this:
    </p>
    <p>
        <img src="s-10.png" />
    </p>
    <p>
        In case you cannot get an online connection to the Koina prediction server please add the PROCAL_Koina.blib spectral library file from the
        <strong>4_backup</strong> subfolder. Just follow the same steps as for the PROCAL library. You can still perform the Mirror spectrum
        comparison of both libraries. Only the online Koina prediction used later in this tutorial will not work.
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
        <li>In the <strong>NCE</strong> field of the <strong>Build Library</strong> form Select ‚Äú31‚Äù.</li>
    </ul>
    <p class="keep-next">
        The <strong>Build Library</strong> form should look like this:
    </p>
    <p>
        <img src="s-11.png" />
    </p>
    <ul>
        <li>Click the <strong>Finish</strong> button.</li>
        <li>Make sure both Libraries are ticked in the <strong>Peptide Settings - Library</strong> tab.</li>
    </ul>
    <p>
        <img src="s-12.png" />
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>
        The PROCAL_Koina library is loading in the background.
    </p>
    <p>
        In the <strong>Library Match</strong> window <strong>Spectrum</strong> dropdown list, you can now select one of the two Libraries that are
        active in the file. Which library spectrum is currently shown is always printed on top of the spectrum together with the peptide sequence and
        charge state.
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig">
                <img src="s-13.png" />
            </td>
            <td class="comp-fig">
                <img src="s-14.png" />
            </td>
        </tr>
    </table>
    <p>
        Skyline can only base product-ion selection on a single spectrum from one library. Skyline will search the libraries in the order they appear
        in the list and use the first spectrum match it finds. With our current settings the PROCAL library is always selected first.
    </p>
    <p>
        To compare the libraries in a mirror plot:
    </p>
    <ul>
        <li>Right-click the <strong>Library Match</strong> graph and click <strong>Mirror</strong>.</li>
        <li>In the <strong>Spectrum</strong> list select ‚Äú<strong>PROCAL</strong>‚Äù and in the <strong>Mirror</strong> list
            ‚Äú<strong>PROCAL_Koina</strong>‚Äù.</li>
    </ul>
    <p>
        <img src="s-15.png" />
    </p>
    <p>
        Compare the dotp match factor for different peptides. (TODO: Need properties grid for this)
    </p>
    <ul>
        <li>Right-click the <strong>Library Match</strong> graph and click <strong>Koina</strong>. </li>
    </ul>
    <p>
        Now you get the same mirror plot with a live Koina prediction for CE 31.
    </p>
    <ul>
        <li>Change the CE setting to ‚Äú<strong>18</strong>‚Äù and ‚Äú<strong>39</strong>‚Äù.</li>
    </ul>
    <p>
        Observe how the fragmentation pattern for a peptide changes.
    </p>
    <table class="comp-fig">
        <tr>
            <td class="comp-fig">
                <img src="s-16.png" />
            </td>
            <td class="comp-fig">
                <img src="s-17.png" />
            </td>
        </tr>
    </table>
    <p>
        Find for several peptides the optimal CE setting, i.e. the setting where experimental and predicted peptides are as similar as possible. Can
        the initial setting of ‚ÄúNCE = 31‚Äù be further optimized?
    </p>
    <ul>
        <li>On the <strong>File</strong> menu, click <strong>Save</strong>.</li>
        <li>Save the Skyline document as ‚Äú<strong>Tutorial_Libraries</strong>‚Äù in the tutorial folder.</li>
    </ul>
    <h1>PRM data analysis</h1>
    <p>
        You have created your spectral libraries and selected your target peptide list. In the next step you will extract the chromatogram information
        for the precursor and transition ions in your list from Thermo mzML files recorded with a PRM method (for settings see <em>Appendix:
            Background information</em>).
    </p>
    <ul>
        <li>On the <strong>File</strong> menu, click <strong>Save As</strong>.</li>
        <li>Save the document as "<strong>Tutorial_PRM</strong>".</li>
    </ul>
    <p>
        Before importing data go again to the <strong>Settings</strong> menu and click <strong>Transition Settings</strong>
    </p>
    <ul>
        <li>Select the <strong>Full-Scan</strong> tab and specify the settings for <strong>MS/MS filtering</strong>.</li>
        <li>In the <strong>Acquisition method</strong> field select ‚Äú<strong>PRM</strong>‚Äù.</li>
        <li>In the <strong>Product mass analyzer</strong> field select ‚Äú<strong>Centroided</strong>‚Äù.</li>
        <li>In the <strong>Mass Accuracy</strong> field enter ‚Äú10‚Äù ppm.</li>
        <li><strong>IMPORTANT</strong>: select the option ‚Äú<strong>Include all matching scans</strong>‚Äù. You need to change this from the standard
            setting if the retention times recorded in the library do not match the retention times in the file you are importing. Consider this
            whenever you are combining libraries and files from different LCs, different method settings, different columns etc.</li>
    </ul>
    <p class="keep-next">
        The <strong>Transition Settings - Full-Scan</strong> tab should look like this:
    </p>
    <p>
        <img src="s-18.png" />
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <table>
        <tr>
            <td>
                <p>
                    <strong>mzML data:</strong>
                </p>
                <p>
                    All measurements in this tutorial were recorded as Thermo .raw files. To decrease file size .raw files were converted to .mzML
                    files using MSConvert.
                </p>
                <p>
                    Kessner et al., 2008. ProteoWizard: Open Source Software for Rapid Proteomics Tools Development. Bioinformatics; doi:
                    10.1093/bioinformatics/btn323).
                </p>
                <p>
                    Martens et al., 2011. mzML ‚Äì a community standard for mass spectrometry data. MCP; <a
                        href="https://doi.org/10.1074/mcp.R110.000133">https://doi.org/10.1074/mcp.R110.000133</a>
                </p>
            </td>
        </tr>
    </table>
    <p>
        To import the mzML data:
    </p>
    <ul>
        <li>On the <strong>File</strong> menu, choose <strong>Import</strong> and click <strong>Results</strong> (Ctrl-R).</li>
    </ul>
    <p>
        <img src="s-19.png" />
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
        <li>In the <strong>Import Results </strong>Files form, navigate to the ‚Äú<strong>1_PRM</strong>‚Äù subfolder and select the four mzML files.</li>
        <li>Click the <strong>Open</strong> button.</li>
        <li>Choose not to remove the common ‚ÄúPRM_‚Äù prefix.</li>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>
        The Chromatogram information is extracted from the mzML files. With four or more cores, all files will be processed in parallel. On most
        laptops with two cores, the import will process two files at a time.
    </p>
    <p>
        Once the import is finished, you want to adjust the Skyline window view. You can drag and dock any window in Skyline by left clicking on the
        window's top border, holding the left mouse button down, and dragging this window to a new position. Hover over one of the arrow symbols and
        release the window.
    </p>
    <ul>
        <li>Drag and drop the <strong>Library Match</strong> window below the <strong>Targets</strong> window.</li>
        <li>On the <strong>View</strong> menu, choose <strong>Arrange Graphs</strong> and click <strong>Tiled</strong>.</li>
        <li>On the <strong>View</strong> menu, choose <strong>Retention Times</strong> and click <strong>Replicate Comparison</strong> (F8).</li>
        <li>On the <strong>View</strong> menu, choose <strong>Peak Areas</strong> and click <strong>Replicate Comparison</strong> (F7).</li>
    </ul>
    <p>
        If the <strong>Replicate Comparison</strong> windows are floating on your Skyline document, drag and dock the <strong>Peak Areas - Replicate
            Comparison</strong> and <strong>Retention Times - Replicate comparison</strong> to a position on the right side of the chromatogram
        windows so that all information is easily visible.
    </p>
    <ul>
        <li>On the <strong>View</strong> menu, choose <strong>Auto-Zoom</strong> and click <strong>Best Peak</strong> (F11).</li>
        <li>On the <strong>View</strong> menu, choose <strong>Transitions</strong> and click <strong>Split Graph</strong>.</li>
        <li>On the <strong>View</strong> menu, choose <strong>Transform</strong> and click <strong>None</strong>.</li>
    </ul>
    <p>
        On a small screen the legends take up too much space.
    </p>
    <ul>
        <li>Right-click a <strong>Chromatogram</strong> graph and click <strong>Legend</strong>.</li>
        <li>Do the same for the <strong>Replicate Comparison</strong> windows.</li>
        <li>Right-click a <strong>Chromatogram window</strong> and select <strong>Synchronize Zooming</strong>.</li>
    </ul>
    <p class="keep-next">
        The main Skyline window should now look like this:
    </p>
    <p>
        <img src="s-20.png" />
    </p>
    <p>
        <em>Investigate the different peptide chromatograms as follows:</em>
    </p>
    <ul>
        <li><em>Compare the MS1 and MS2 chromatograms for the four PROCAL dilutions.</em></li>
        <li><em>Which peptides are not well detected at which concentration?</em></li>
        <li><em>Can you find three peptides for which at low concentrations the MS1 signal (precursor ion) has already vanished in the noise, while
                MS2 signal (product ions) still provide decent chromatogram quality?</em></li>
        <li><em>Can you find a peptide that behaves the other way around, i.e. for which MS2 signal is of lower quality than MS1?</em></li>
        <li><em>Can you find a peptide with a strong interference in MS1?</em></li>
    </ul>
    <p>
        Before moving to the next section:
    </p>
    <ul>
        <li>On the <strong>File</strong> menu, click <strong>Save</strong> (Ctrl-S).</li>
    </ul>
    <h1>DIA Data Analysis</h1>
    <p>
        Now that you are already familiar with some Skyline functions, you can easily explore how the PROCAL peptide dilution chromatograms look when
        measured with DIA (for settings see <em>Appendix: Background information</em>).
    </p>
    <p>
        For this you need to start a new Skyline document and adjust the settings for DIA.
    </p>
    <ul>
        <li>Click the <strong>New Document</strong> button on the toolbar.</li>
        <li>On the <strong>File</strong> menu, click <strong>Start</strong>.</li>
        <li>Select the option ‚Äú<strong>Import DIA Peptide Search</strong>‚Äù.</li>
    </ul>
    <p>
        <img src="../../shared/en/image-start-dia.png" />
    </p>
    <p>
        The <strong>Import Peptide Search</strong> wizard opens. You are presented with the <strong>Spectral Library</strong> page, which allows you to
        build a project-specific spectral library. Here you use the database search results and library prediction performed in DIA-NN<sup>3</sup>
        (For settings see <em>Appendix: Background information</em>).
    </p>
    <p>
        <sup>3</sup> Demichev V, Messner CB, Vernardis SI, Lilley KS, Ralser M. DIA-NN: neural networks and interference correction enable deep
        proteome coverage in high throughput. Nature Methods 17, 41-44 (2020)
    </p>
    <ul>
        <li>In the <strong>Spectral Library</strong> window, click <strong>Add Files</strong> and navigate to the ‚Äú<strong>2_DIA</strong>‚Äù subfolder.
        </li>
        <li>Open the ‚Äú<strong>DIA_PROCAL.tsv.speclib</strong>‚Äù file.</li>
    </ul>
    <table>
        <tr>
            <td><strong>Note: </strong>The ‚ÄúDIA_PROCAL.tsv.speclib‚Äù file was predicted with DIA-NN based on the PROCAL.fasta (see <em>Appendix:
                    Background information</em>). To read in the spectra information from the measurement files requires in addition the ‚Äúreport.tsv‚Äù
                result file from the DIA-NN database search. Both files, the .speclib and report.tsv file, need to have the same naming. The
                measurement data files (here mzML) do not need to be located in the same folder.</td>
        </tr>
    </table>
    <p>
        <img src="s-21.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>
        You are presented with the <strong>Extract Chromatograms </strong>form, which shows the DIA data files Skyline has auto-detected for
        chromatogram extraction, peak detection, and peak area calculation. The form should look like this:
    </p>
    <p>
        <img src="s-22.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
        <li>Choose not to remove the common ‚ÄúDIA_‚Äù prefix.</li>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>
        The Import wizard asks if you want to add modifications.
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>
        In the <strong>Configure Transition Settings</strong> page do the following:
    </p>
    <ul>
        <li>In the <strong>Precursor charges</strong> field enter ‚Äú2,3‚Äù.</li>
        <li>In the <strong>Ion charges</strong> field enter ‚Äú1‚Äù</li>
        <li>In the <strong>Ion types</strong> field enter ‚Äúy, b, p‚Äù.</li>
        <li>In the <strong>Product ions</strong> fields choose <strong>From</strong> ‚Äúion 2‚Äù <strong>To</strong> <strong>‚Äú</strong>Last
            ion<strong>‚Äù.</strong></li>
        <li>In the <strong>Min m/z</strong> field enter ‚Äú350‚Äù.</li>
        <li>Check the <strong>Use DIA precursor isolation window for exclusion</strong> checkbox.</li>
        <li>In the <strong>Ion match tolerance</strong> field use the default ‚Äú0.05‚Äù.</li>
        <li>In the <strong>Pick</strong> fields enter ‚Äú6‚Äù <strong>product ions</strong> and ‚Äú3‚Äù <strong>min product ions</strong>.</li>
    </ul>
    <p>
        <img src="s-23.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>
        In the <strong>Configure Full-Scan Settings</strong> page do the following:
    </p>
    <ul>
        <li>For MS1 filtering, in the<strong> Precursor mass analyzer</strong> field choose ‚Äú<strong>Centroided</strong>‚Äù.</li>
        <li>In the <strong>Mass Accuracy</strong> field enter<strong> </strong>‚Äú10‚Äù <strong>ppm</strong>.</li>
        <li>For MS/MS filtering, in the <strong>Isolation scheme</strong> field select <strong>Add</strong>.</li>
        <li>In the <strong>Name</strong> field enter ‚ÄúDIA_40windows‚Äù.</li>
        <li>Choose the option <strong>Prespecified isolation windows</strong>.</li>
        <li>Click the <strong>Import</strong> button.</li>
        <li>Navigate to the <strong>2_DIA </strong>subfolder and open the first DIA mzML file.</li>
    </ul>
    <p>
        Skyline automatically reads in the DIA windows settings from the mzML file.
    </p>
    <ul>
        <li>Click the <strong>Graph</strong> button to have a look at the isolation scheme and then <strong>Close</strong>.</li>
        <li>Click the <strong>OK</strong> button in the <strong>Edit Isolation Scheme</strong> form.</li>
    </ul>
    <p class="keep-next">
        The <strong>Configure Full-Scan Settings</strong> page should look like this:
    </p>
    <p>
        <img src="s-24.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>
        In the <strong>Import FASTA</strong> page do the following:
    </p>
    <ul>
        <li>Click the <strong>Browse</strong> button.</li>
        <li>Navigate to the <strong>1_PRM</strong> subfolder.</li>
        <li>Open the ‚Äú<strong>PROCAL.fasta</strong>‚Äú file.</li>
        <li>In the <strong>Decoy generation method</strong> field select ‚Äú<strong>Shuffle Sequence</strong>‚Äù. (WHY?)</li>
    </ul>
    <p class="keep-next">
        The <strong>Import FASTA </strong>page should look like this:
    </p>
    <p>
        <img src="s-25.png" />
    </p>
    <ul>
        <li>Click the <strong>Finish</strong> button.</li>
    </ul>
    <p class="keep-next">
        Skyline shows the <strong>Associate Proteins</strong> form with the following values for the default options:
    </p>
    <p>
        <img src="s-26.png" />
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>
        Skyline begins extracting chromatograms from the DIA mzML files. During this time, you can already organize the Skyline windows as before. You
        should only need to do the following:
    </p>
    <ul>
        <li>On the <strong>View</strong> menu, choose <strong>Arrange Graphs</strong>, and click <strong>Tiled</strong>.</li>
    </ul>
    <p class="keep-next">
        When the import has completed, Skyline should look like this:
    </p>
    <p>
        <img src="s-27.png" />
    </p>
    <p>
        As you have loaded at the beginning of the Import DIA Peptide Search procedure the results file from a DIA-NN analysis (the PROCAL-tsv.speclib
        and the PROCAL-report.tsv present in the same folder), Skyline is able to show you the retention time at which DIA-NN has successfully
        identified each peptide in each raw file. 
    </p>
    <p>
        A successful peptide identification is indicated with a dark blue line annoted as ‚ÄúID‚Äù, with the retention time. You can add or remove this
        peptide identification information by doing the following:
    </p>
    <ul>
        <li>Right-click any chromatogram graph, choose <strong>Peptide ID Times</strong>,<strong> </strong>and click <strong>Matching</strong> to
            check or uncheck this item<strong>.</strong></li>
    </ul>
    <p>
        <img src="s-28.png" />
    </p>
    <p>
        <em>Now quantitatively investigate the different peptide chromatograms.</em>
    </p>
    <ul>
        <li><em>Compare the MS1 and MS2 chromatograms for the four PROCAL dilutions.</em></li>
        <li><em>Which peptides are not well detected? At which concentrations?</em></li>
        <li><em>Look up the peptides you investigated in detail in the PRM data before and compare those in the DIA data.</em></li>
        <li><em>Compare the retention time in the 0.1fmol sample for "TASGVGGFSTK". Compare the mass errors for the selected peaks of this peptide
                over all concentration. Manually correct the peak area integration in the 0.1fmol sample.</em></li>
        <li><em>Check the peptide "HDTVFGSYLYK". What is the interfered product ion here? Remove it from the document.</em></li>
        <li><em>Manually correct peak area integration and remove interfered product ions for the whole document.</em></li>
        <li><em>Have a look at the decoy peptides "GAYDSSIHEHK", "LGSYGTSAGVK" or "TILGDDIVFGK". What is the dotp score for the 0.1fmol sample?</em>
        </li>
    </ul>
    <p>
        Before moving to the next section:
    </p>
    <ul>
        <li>On the <strong>File</strong> menu, click <strong>Save</strong> (Ctrl-S).</li>
    </ul>
    <h1>DDA Data Analysis</h1>
    <p>
        Now that you are already familiar with some Skyline functions, you can easily explore how the PROCAL peptide dilution chromatograms look when
        measured with DDA (for settings see <em>Appendix: Background information</em>).
    </p>
    <p>
        For this you need to start a new Skyline document and adjust the settings for DDA.
    </p>
    <ul>
        <li>Click the <strong>New Document</strong> button on the toolbar.</li>
        <li>On the <strong>File</strong> menu, click <strong>Start</strong>.</li>
        <li>Select the option ‚Äú<strong>Import DDA Peptide Search</strong>‚Äù.</li>
    </ul>
    <p>
        <img src="../../shared/en/image-start-dda.png" />
    </p>
    <p>
        The <strong>Import Peptide Search </strong>wizard opens. You are presented with the <strong>Spectral Library </strong>page, which allows you to
        build a project-specific spectral library. Here you use the database search results from<strong> MaxQuant</strong><sup>4</sup> (For settings
        see <em>Appendix: Background information</em>)
    </p>
    <p>
        <sup>4</sup> Cox J, &amp; Mann M. MaxQuant enables high peptide identification rates, individualized p.p.b.-range mass accuracies and
        proteome-wide protein quantification. Nature Biotechnology 26, 1367-1372 (2008)
    </p>
    <ul>
        <li>In the <strong>Spectral Library</strong> window, click <strong>Add Files</strong> and navigate to the ‚Äú<strong>3_DDA</strong>‚Äù subfolder.
        </li>
        <li>Open the ‚Äú<strong>msms.txt</strong>‚Äù file.</li>
    </ul>
    <table>
        <tr>
            <td><strong>Note: </strong>Reading the spectra information form the measurement files requires msms.txt result file and mqpar.xml file
                from the MaxQuant search results. The measurement data files (here mzML) must be located in the same folder.</td>
        </tr>
    </table>
    <p>
        <img src="s-29.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>
        Skyline reads in the identified spectra from the MaxQuant search and creates a new spectral library file.
    </p>
    <p>
        You are presented with the <strong>Extract Chromatograms </strong>form where the mzML file(s) used in the search are usually automatically
        selected if it located in the same folder as the msms.txt folder. If not navigate to the <strong>3_DDA</strong> subfolder and select the four
        DDA mzML files. 
    </p>
    <p>
        <img src="s-30.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
        <li>Choose not to remove the common ‚ÄúDDA_‚Äù prefix.</li>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>
        <img src="s-31.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>
        In the <strong>Configure Full-Scan Settings</strong> page do the following:
    </p>
    <ul>
        <li>Change the <strong>Precursor charges</strong> field to ‚Äú2,3‚Äù and otherwise accept defaults.</li>
        <li>In the <strong>Precursor mass analyzer</strong> field leave ‚Äú<strong>Centroided</strong>‚Äù.</li>
        <li>In the <strong>Mass Accuracy</strong> field leave<strong> </strong>‚Äú10‚Äù <strong>ppm</strong>.</li>
    </ul>
    <p>
        The <strong>Configure Full-Scan Settings </strong>page<strong> </strong>should look like this:
    </p>
    <p>
        <img src="s-32.png" />
    </p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>
        In the <strong>Import FASTA</strong> page do the following:
    </p>
    <ul>
        <li>Click the <strong>Browse</strong> button.</li>
        <li>Navigate to the <strong>1_PRM</strong> subfolder.</li>
        <li>Open the ‚Äú<strong>PROCAL.fasta</strong>‚Äú file.</li>
    </ul>
    <p class="keep-next">
        The <strong>Import FASTA </strong>page should look like this:
    </p>
    <p>
        <img src="s-33.png" />
    </p>
    <ul>
        <li>Click the <strong>Finish</strong> button.</li>
    </ul>
    <p class="keep-next">
        Skyline shows the <strong>Associate Proteins</strong> form with the following values for the default options:
    </p>
    <p>
        <img src="s-34.png" />
    </p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>
        Skyline begins extracting chromatograms from the MS1 spectra in the DDA mzML files. During this time, you can already organize the Skyline
        windows as before. You should only need to do the following:
    </p>
    <ul>
        <li>On the <strong>View</strong> menu, choose <strong>Arrange Graphs</strong>, and click <strong>Tiled</strong>.</li>
    </ul>
    <p class="keep-next">
        When the import has completed, Skyline should look like this:
    </p>
    <p>
        <img src="s-35.png" />
    </p>
    <p>
        As you have imported DDA data, Skyline will only show you MS1 chromatograms but no product ion (MS2) chromatograms. However, the information if
        and at which retention time an MS2 spectrum was recorded that led to a successful peptide identification with MaxQuant is indicated with a
        dark blue line annoted as ‚ÄúID‚Äù, with the retention time. You can add or remove this peptide identification information by doing the following:
    
    </p>
    <ul>
        <li>Right-click any chromatogram graph, choose <strong>Peptide ID Times </strong>and click <strong>Matching</strong> to check or uncheck this
            item<strong>.</strong></li>
    </ul>
    <p>
        <img src="s-36.png" />
    </p>
    <p>
        <em>Investigate the identified MS2 spectra for all peptides in your Skyline document. </em>
    </p>
    <ul>
        <li><em>How many identified spectra do you tend to get per precursor in each raw file? </em></li>
        <li><em>At which concentrations do you tend to get more identifications?</em></li>
        <li><em>Do you tend to get MS2 identifications rather in the beginning, the end, or in the middle of the peak/chromatogram?</em></li>
    </ul>
    <p>
        <em>Now quantitatively investigate the different peptide MS1 chromatograms. </em>
    </p>
    <ul>
        <li><em>Compare the MS1 chromatogram peak areas for the four PROCAL dilutions for all peptides.</em></li>
        <li><em>Which peptides are not well detected? At which concentration(s)? </em></li>
        <li><em>Specifically, look up the peptides ‚ÄúTASGVGGFSTK‚Äù and ‚ÄúHLTGLTFDTYK‚Äù. Carefully check retention time, idotp and mass errors and how
                those values change between the different concentrations. What is the issue with the 0.1 fmol sample? </em></li>
        <li><em>Manually correct the wrongly integrated peptides in the document.</em></li>
    </ul>
    <p>
        Before moving to the next section:
    </p>
    <ul>
        <li>On the <strong>File</strong> menu, click <strong>Save</strong> (Ctrl-S).</li>
    </ul>
    <h1>Conclusion</h1>
    <p>
        You have completed this Skyline tutorial! You have learned to import and build spectral libraries and how to investigate target peptides in
        PRM, DIA or DDA measurement files. The doors are open to go crazy and explore your data in more detail with the many interactive
        visualizations offered by Skyline.
    </p>
    <h1>Extra task for quick people</h1>
    <ul>
        <li><em>Export precursor and product ion total areas from the Skyline document using </em><strong><em>File ü°™ Export ü°™
                    Report</em></strong><em>.</em></li>
        <li><em>Visualize those ion intensities using your favourite data visualization tool. Think about optimal ways how to plot the data. For
                example, plot intensity ratios per peptide (measured peptide intensity at each concentration divided by the peptide intensity at 100
                fmol) versus peptide retention time for all three data acquisition methods on MS1 and/or MS2 level.</em></li>
        <li><em>What general trends do you observe?</em></li>
        <li><em>Do you see something that does not fit your expectations?</em></li>
    </ul>
    <h1>Background information</h1>
    <ol>
        <li>Spectral Libraries</li>
    </ol>
    <p>
        In this tutorial we created spectral libraries for the PROCAL peptide selection. You can do the same for any peptide selection for example
        using public available spectral library sources like:
    </p>
    <ul>
        <li>Peptide Atlas (<a href="http://www.peptideatlas.org/speclib/"><u>http://www.peptideatlas.org/speclib/</u></a>)</li>
        <li>National Institute of Standards and Technology (NIST) (<a href="https://peptide.nist.gov/"><u>https://peptide.nist.gov/</u></a>)</li>
        <li>The Global Proteome Machine (GPM) (<a
                href="ftp://ftp.thegpm.org/projects/xhunter/libs/"><u>ftp://ftp.thegpm.org/projects/xhunter/libs/</u></a>)</li>
    </ul>
    <p>
        The Proteome Tools project offers spectral libraries recorded from synthesized peptides at different collision energies and fragmentation
        settings.
    </p>
    <ul>
        <li>Proteome Tools (<a href="https://www.proteometools.org/index.php?id=53"><u>https://www.proteometools.org/index.php?id=53</u></a>)</li>
    </ul>
    <p>
        You can also create new spectral libraries in Skyline using other publicly available data, or peptide search results from your laboratory
        experiments. Skyline supports building libraries from the following search result formats:
    </p>
    <p>
        BlibBuild (<a
            href="https://skyline.ms/wiki/home/software/BiblioSpec/page.view?name=BlibBuild"><u>https://skyline.ms/wiki/home/software/BiblioSpec/page.view?name=BlibBuild</u></a>)
    
    </p>
    <p>
        <img src="image-figure-ssl.png" />
    </p>
    <ul>
        <li>Custom (SSL) (<a href="https://skyline.ms/blib-formats.url"><u>https://skyline.ms/blib-formats.url</u></a>)</li>
    </ul>
    <ol>
        <li>Eclipse instrument settings</li>
    </ol>
    <p>
        LC-Settings for PRM, DIA &amp; DDA
    </p>
    <p>
        Ultimate 3000 RSLCnano system
    </p>
    <ul>
        <li>Trap column (ReproSil-pur C18-AQ, 5 Œºm, Dr. Maisch, 20 mm √ó 75 Œºm, self-packed) at a flow rate of 5 ŒºL/min in HPLC grade water with 0.1%
            (v/v) formic acid</li>
        <li>Analytical column (ReproSil Gold C18-AQ, 3 Œºm, Dr. Maisch, 450 mm √ó 75 Œºm, self-packed)</li>
        <li>50 min linear gradient from 4% to 32% of solvent B (0.1% formic acid and 5% (v/v) DMSO in acetonitrile) at 300 nL/min flow rate. nanoLC
            solvent A was 0.1% (v/v) formic acid, 5% (v/v) DMSO in HPLC grade water</li>
    </ul>
    <p>
        Orbitrap Eclipse
    </p>
    <div style="margin-left : 0px;">
        <table>
            <tr>
                <td><strong>Parameter</strong></td>
                <td><strong>PRM</strong></td>
                <td><strong>DIA</strong></td>
                <td><strong>DDA</strong></td>
            </tr>
            <tr>
                <td colspan="4">MS1</td>
            </tr>
            <tr>
                <td>Orbitrap Resolution</td>
                <td>60,000</td>
                <td>120,000</td>
                <td>60,000</td>
            </tr>
            <tr>
                <td>Scan Range (m/z)</td>
                <td>360-1300</td>
                <td>360-1300</td>
                <td>360-1300</td>
            </tr>
            <tr>
                <td>Max IT (ms)</td>
                <td>50</td>
                <td>50</td>
                <td>50</td>
            </tr>
            <tr>
                <td>Norm.AGC Target</td>
                <td>100%</td>
                <td>100%</td>
                <td>100%</td>
            </tr>
            <tr>
                <td colspan="4">MS2</td>
            </tr>
            <tr>
                <td>Isolation Window</td>
                <td>1.3</td>
                <td>-</td>
                <td>1.3</td>
            </tr>
            <tr>
                <td>NCE (%)</td>
                <td>30</td>
                <td>30</td>
                <td>30</td>
            </tr>
            <tr>
                <td>Orbitrap Resolution</td>
                <td>30,000</td>
                <td>30,000</td>
                <td>15,000</td>
            </tr>
            <tr>
                <td>Scan Range (m/z)</td>
                <td>140-2000</td>
                <td>200-1800</td>
                <td>-</td>
            </tr>
            <tr>
                <td>Max IT (ms)</td>
                <td>120</td>
                <td>54</td>
                <td>22</td>
            </tr>
            <tr>
                <td>Norm.AGC Target</td>
                <td>400%</td>
                <td>1000%</td>
                <td>200%</td>
            </tr>
            <tr>
                <td>Cycle time</td>
                <td>-</td>
                <td>-</td>
                <td>2</td>
            </tr>
            <tr>
                <td>Dynamic exclusion</td>
                <td>-</td>
                <td>-</td>
                <td>30s</td>
            </tr>
            <tr>
                <td>Windows</td>
                <td>-</td>
                <td>40 variable</td>
                <td></td>
            </tr>
        </table>
    </div>
    <ol>
        <li>MaxQuant &amp; DIA-NN settings</li>
    </ol>
    <p>
        MaxQant v.2.4.0.0
    </p>
    <div style="margin-left : 0px;">
        <table>
            <tr>
                <td><strong>Parameter</strong></td>
                <td><strong>Settings</strong></td>
            </tr>
            <tr>
                <td>Type</td>
                <td>Standard</td>
            </tr>
            <tr>
                <td>Enzyme</td>
                <td>Trypsin/P</td>
            </tr>
            <tr>
                <td>Missed clevages</td>
                <td>1</td>
            </tr>
            <tr>
                <td>Modifications</td>
                <td>
                    <p>
                        Fixed: Carbamidomethyl (C)
                    </p>
                    <p>
                        Variable: Oxidation (M);Acetyl (Protein N-term)
                    </p>
                </td>
            </tr>
            <tr>
                <td>Label-free quantification</td>
                <td>None</td>
            </tr>
            <tr>
                <td>Sequences</td>
                <td>PROCAL.fasta; human reference UP000005640 Swiss Prot fasta</td>
            </tr>
            <tr>
                <td>Match between runs</td>
                <td>FALSE</td>
            </tr>
            <tr>
                <td>PSM FDR</td>
                <td>1%</td>
            </tr>
            <tr>
                <td>Protein FDR</td>
                <td>1%</td>
            </tr>
            <tr>
                <td>Min peptide length</td>
                <td>7</td>
            </tr>
        </table>
    </div>
    <p>
        DIA-NN v.1.8.1
    </p>
    <div style="margin-left : 0px;">
        <table>
            <tr>
                <td><strong>Parameter</strong></td>
                <td><strong>Settings</strong></td>
            </tr>
            <tr>
                <td>Spectral library prediction</td>
                <td>
                    <p>
                        PROCAL.fasta; human reference UP000005640 Swiss Prot fasta ; contaminants.fastaTrypsin/P; 1 Missed cleavages
                    </p>
                    <p>
                        0 Maximum number of variable modifications
                    </p>
                </td>
            </tr>
            <tr>
                <td>DIA search</td>
                <td></td>
            </tr>
            <tr>
                <td>Spectral library</td>
                <td>PROCAL predicted library (.speclib)</td>
            </tr>
            <tr>
                <td>Peptide length range</td>
                <td>7-30</td>
            </tr>
            <tr>
                <td>Precursor m/z range</td>
                <td>360-1300</td>
            </tr>
            <tr>
                <td>Fragment ion m/z range</td>
                <td>200-1800</td>
            </tr>
            <tr>
                <td>MBR</td>
                <td>TRUE</td>
            </tr>
            <tr>
                <td>Protein inference</td>
                <td>Genes</td>
            </tr>
            <tr>
                <td>Neural network classifier</td>
                <td>Single-pass mode</td>
            </tr>
            <tr>
                <td>Quantification strategy</td>
                <td>Robust LC (high precision)</td>
            </tr>
            <tr>
                <td>Cross-run normalization</td>
                <td>RT-dependent</td>
            </tr>
            <tr>
                <td>Library generation</td>
                <td>Smart profiling</td>
            </tr>
            <tr>
                <td>Speed and RAM usage</td>
                <td>Optimal results</td>
            </tr>
        </table>
    </div>
    <ol>
        <li>File conversion</li>
    </ol>
    <p>
        Thermo .raw files were converted to mzML files using MSConvert (ProteoWizard: Open Source Software for Rapid Proteomics Tools Development.
        Darren Kessner; Matt Chambers; Robert Burke; David Agus; Parag Mallick. Bioinformatics 2008; doi: 10.1093/bioinformatics/btn323). 
    </p>
    <div style="margin-left : 0px;">
        <table>
            <tr>
                <td><strong>Parameter</strong></td>
                <td><strong>Settings</strong></td>
            </tr>
            <tr>
                <td>Output format</td>
                <td>mzML</td>
            </tr>
            <tr>
                <td>Binary encoding precision</td>
                <td>64-bit</td>
            </tr>
            <tr>
                <td>Write Index;Use zlib compression;TPP compatibility</td>
                <td>TRUE;TRUE;TRUE</td>
            </tr>
            <tr>
                <td>Filters</td>
                <td>Peak Picking; MS Levels 1-2</td>
            </tr>
        </table>
    </div>
    <ol>
        <li id="footnote">
            <p>
                <sup>1</sup> Zolg DP, Wilhelm M, Yu P, Knaute T, Zerweck J, Wenschuh H, Reimer U, Schnatbaum K, Kuster B. PROCAL: A set of 40 peptide
                standards for retention time indexing, column performance monitoring, and collision energy calibration. Proteomics. 2017 Nov; 17(21)
                <a href="#footnote-ref-2">‚Üë</a>
            </p>
        </li>
    </ol>
</body>

</html>