<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns:st1="urn:schemas-microsoft-com:office:smarttags"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=us-ascii">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 12">
<meta name=Originator content="Microsoft Word 12">
<link rel=File-List href="index_files/filelist.xml">
<title>MyriMatch</title>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="place"/>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="City"/>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Matt Chambers</o:Author>
  <o:LastAuthor>Matt Chambers</o:LastAuthor>
  <o:Revision>446</o:Revision>
  <o:TotalTime>2149</o:TotalTime>
  <o:Created>2007-05-31T15:23:00Z</o:Created>
  <o:LastSaved>2011-07-22T19:05:00Z</o:LastSaved>
  <o:Pages>5</o:Pages>
  <o:Words>4039</o:Words>
  <o:Characters>23024</o:Characters>
  <o:Company>MSRC</o:Company>
  <o:Lines>191</o:Lines>
  <o:Paragraphs>54</o:Paragraphs>
  <o:CharactersWithSpaces>27009</o:CharactersWithSpaces>
  <o:Version>12.00</o:Version>
 </o:DocumentProperties>
</xml><![endif]-->
<link rel=themeData href="index_files/themedata.thmx">
<link rel=colorSchemeMapping href="index_files/colorschememapping.xml">
<!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:Zoom>90</w:Zoom>
  <w:DisplayBackgroundShape/>
  <w:TrackMoves>false</w:TrackMoves>
  <w:TrackFormatting/>
  <w:PunctuationKerning/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:DoNotPromoteQF/>
  <w:LidThemeOther>EN-US</w:LidThemeOther>
  <w:LidThemeAsian>X-NONE</w:LidThemeAsian>
  <w:LidThemeComplexScript>X-NONE</w:LidThemeComplexScript>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:DontGrowAutofit/>
   <w:SplitPgBreakAndParaMark/>
   <w:DontVertAlignCellWithSp/>
   <w:DontBreakConstrainedForcedTables/>
   <w:DontVertAlignInTxbx/>
   <w:Word11KerningPairs/>
   <w:CachedColBalance/>
  </w:Compatibility>
  <w:BrowserLevel>MicrosoftInternetExplorer4</w:BrowserLevel>
  <m:mathPr>
   <m:mathFont m:val="Cambria Math"/>
   <m:brkBin m:val="before"/>
   <m:brkBinSub m:val="&#45;-"/>
   <m:smallFrac m:val="off"/>
   <m:dispDef/>
   <m:lMargin m:val="0"/>
   <m:rMargin m:val="0"/>
   <m:defJc m:val="centerGroup"/>
   <m:wrapIndent m:val="1440"/>
   <m:intLim m:val="subSup"/>
   <m:naryLim m:val="undOvr"/>
  </m:mathPr></w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" DefUnhideWhenUsed="false"
  DefSemiHidden="false" DefQFormat="false" LatentStyleCount="267">
  <w:LsdException Locked="false" QFormat="true" Name="Normal"/>
  <w:LsdException Locked="false" QFormat="true" Name="heading 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 6"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 7"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 8"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 9"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="caption"/>
  <w:LsdException Locked="false" QFormat="true" Name="Title"/>
  <w:LsdException Locked="false" Priority="1" Name="Default Paragraph Font"/>
  <w:LsdException Locked="false" QFormat="true" Name="Subtitle"/>
  <w:LsdException Locked="false" QFormat="true" Name="Strong"/>
  <w:LsdException Locked="false" QFormat="true" Name="Emphasis"/>
  <w:LsdException Locked="false" Priority="99" Name="No List"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   Name="Placeholder Text"/>
  <w:LsdException Locked="false" Priority="1" QFormat="true" Name="No Spacing"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 1"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true" Name="Revision"/>
  <w:LsdException Locked="false" Priority="34" QFormat="true"
   Name="List Paragraph"/>
  <w:LsdException Locked="false" Priority="29" QFormat="true" Name="Quote"/>
  <w:LsdException Locked="false" Priority="30" QFormat="true"
   Name="Intense Quote"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 1"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 1"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 1"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 2"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 2"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 2"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 2"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 3"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 3"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 3"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 3"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 4"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 4"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 4"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 4"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 5"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 5"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 5"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 5"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 6"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 6"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 6"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 6"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="19" QFormat="true"
   Name="Subtle Emphasis"/>
  <w:LsdException Locked="false" Priority="21" QFormat="true"
   Name="Intense Emphasis"/>
  <w:LsdException Locked="false" Priority="31" QFormat="true"
   Name="Subtle Reference"/>
  <w:LsdException Locked="false" Priority="32" QFormat="true"
   Name="Intense Reference"/>
  <w:LsdException Locked="false" Priority="33" QFormat="true" Name="Book Title"/>
  <w:LsdException Locked="false" Priority="37" SemiHidden="true"
   UnhideWhenUsed="true" Name="Bibliography"/>
  <w:LsdException Locked="false" Priority="39" SemiHidden="true"
   UnhideWhenUsed="true" QFormat="true" Name="TOC Heading"/>
 </w:LatentStyles>
</xml><![endif]--><!--[if !mso]><object
 classid="clsid:38481807-CA0E-42D2-BF39-B33AF135CC4D" id=ieooui></object>
<style>
st1\:*{behavior:url(#ieooui) }
</style>
<![endif]-->
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-charset:1;
	mso-generic-font-family:roman;
	mso-font-format:other;
	mso-font-pitch:variable;
	mso-font-signature:0 0 0 0 0 0;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:-520081665 -1073717157 41 0 66047 0;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoCommentText, li.MsoCommentText, div.MsoCommentText
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-link:"Comment Text Char";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
span.MsoCommentReference
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-ansi-font-size:8.0pt;
	mso-bidi-font-size:8.0pt;}
a:link, span.MsoHyperlink
	{mso-style-unhide:no;
	color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{mso-style-unhide:no;
	color:purple;
	text-decoration:underline;
	text-underline:single;}
p.MsoCommentSubject, li.MsoCommentSubject, div.MsoCommentSubject
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-parent:"Comment Text";
	mso-style-link:"Comment Subject Char";
	mso-style-next:"Comment Text";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";
	font-weight:bold;}
p.MsoAcetate, li.MsoAcetate, div.MsoAcetate
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-link:"Balloon Text Char";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:8.0pt;
	font-family:"Tahoma","sans-serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraph, li.MsoListParagraph, div.MsoListParagraph
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraphCxSpFirst, li.MsoListParagraphCxSpFirst, div.MsoListParagraphCxSpFirst
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraphCxSpMiddle, li.MsoListParagraphCxSpMiddle, div.MsoListParagraphCxSpMiddle
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraphCxSpLast, li.MsoListParagraphCxSpLast, div.MsoListParagraphCxSpLast
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
span.CommentTextChar
	{mso-style-name:"Comment Text Char";
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"Comment Text";}
span.CommentSubjectChar
	{mso-style-name:"Comment Subject Char";
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-parent:"Comment Text Char";
	mso-style-link:"Comment Subject";
	font-weight:bold;}
span.BalloonTextChar
	{mso-style-name:"Balloon Text Char";
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"Balloon Text";
	mso-ansi-font-size:8.0pt;
	mso-bidi-font-size:8.0pt;
	font-family:"Tahoma","sans-serif";
	mso-ascii-font-family:Tahoma;
	mso-hansi-font-family:Tahoma;
	mso-bidi-font-family:Tahoma;}
.MsoChpDefault
	{mso-style-type:export-only;
	mso-default-props:yes;
	font-size:10.0pt;
	mso-ansi-font-size:10.0pt;
	mso-bidi-font-size:10.0pt;}
@page Section1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in 1.0in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
 /* List Definitions */
 @list l0
	{mso-list-id:59913793;
	mso-list-type:hybrid;
	mso-list-template-ids:-1923173740 1549957072 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l0:level1
	{mso-level-start-at:2;
	mso-level-number-format:alpha-lower;
	mso-level-text:"%1\)";
	mso-level-tab-stop:36.75pt;
	mso-level-number-position:left;
	margin-left:36.75pt;
	text-indent:-18.75pt;
	mso-ansi-font-size:14.0pt;
	mso-ansi-font-weight:normal;}
@list l0:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1
	{mso-list-id:180971617;
	mso-list-type:hybrid;
	mso-list-template-ids:-972648950 -1767603584 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l1:level1
	{mso-level-number-format:roman-upper;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.5in;
	mso-ansi-font-size:14.0pt;
	mso-bidi-font-size:14.0pt;}
@list l1:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2
	{mso-list-id:451241682;
	mso-list-type:hybrid;
	mso-list-template-ids:1598447620 -654440768 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l2:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:.85in;
	text-indent:-.25in;
	font-family:"Arial","sans-serif";
	mso-fareast-font-family:"Times New Roman";}
@list l2:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3
	{mso-list-id:589196660;
	mso-list-type:hybrid;
	mso-list-template-ids:409901062 1606998620 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l3:level1
	{mso-level-number-format:alpha-lower;
	mso-level-text:"%1\)";
	mso-level-tab-stop:37.5pt;
	mso-level-number-position:left;
	margin-left:37.5pt;
	text-indent:-19.5pt;
	mso-ansi-font-size:14.0pt;}
@list l3:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
 table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";}
table.MsoTableGrid
	{mso-style-name:"Table Grid";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-unhide:no;
	border:solid windowtext 1.0pt;
	mso-border-alt:solid windowtext .5pt;
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-border-insideh:.5pt solid windowtext;
	mso-border-insidev:.5pt solid windowtext;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="3074"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=Section1>

<p class=MsoNormal><a name=Top></a><span style='font-size:36.0pt;font-family:
"Arial","sans-serif"'>MyriMatch 1.6<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:18.0pt;mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'>For MS-MS Database Search<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:18.0pt;mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:14.0pt;mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'>Table of Contents:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>I.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><a href="#Introduction">Introduction</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>II.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><a href="#Usage">Usage</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>a.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><a href="#UsageBasic">Basic</a> <o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>b.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><a href="#UsagePTMs">Specifying static or dynamic
mass modifications</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>c.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><a href="#UsageConfigurationParametersGuide">Configuration
parameters guide</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>III.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><a href="#InterpretingResults">Interpreting
results</a></span><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>a.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'><a
href="#InterpretingResultsSearchOutput">Search-time output</a></span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>b.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'><a
href="#InterpretingResultsSQTFile">PepXML format guide</a></span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>c.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'><a
href="#InterpretingResultsValidation">Validation</a></span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><a name=Introduction><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>Introduction</span></a><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>MyriMatch is
a tool designed to take experimental data from shotgun proteomics experiments
and compare those spectra against sequences in a known database of proteins.
Whether the program is being run in a single-computer environment or across an
entire cluster of processing nodes, it is able to optimally divide work in a
much more efficient way than many other database search programs. This is
because it only generates candidate sequences from the known database once for
the entire set of spectra instead of once for every spectrum. Thus, for each
candidate sequence generated, it is compared against every spectrum. The
spectra keep a certain (user-defined) number of candidate sequences that had
the highest scores.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><a name=Usage></a><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>Usage<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>a)</span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp; </span><a name=UsageBasic></a>The basic usage of
MyriMatch is quite simple:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:.25in'><b
style='mso-bidi-font-weight:normal'><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'>myrimatch [flags] &#8211;ProteinDatabase &lt;FASTA
database filepath&gt; &lt;MS/MS data filepath in a supported file format&gt;
&lt;another MS/MS data filepath&gt;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.25in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>When running
it from the command line, the command line parser first determines what flags
you have specified in the command. The flags can be anywhere on the command
line. The following basic flags are supported:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><b
style='mso-bidi-font-weight:normal'>-cfg &lt;file&gt;</b><span
style='mso-tab-count:6'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>specifies
a runtime configuration [default: myrimatch.cfg]<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><b
style='mso-bidi-font-weight:normal'>-workdir &lt;path&gt;</b><span
style='mso-tab-count:5'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>specifies
an absolute path to use as the working directory during execution [default:
current working directory]<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>-cpus
&lt;integer&gt;<span style='mso-tab-count:5'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></b><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>specifies
the number of worker threads to use during search [default: all available
processors]<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>If a flag is
specified that expects an argument but no argument is provided, it might be
treated as a spectrum data file which probably undesirable. If you do not
specify a runtime configuration file with -cfg and the default configuration
file is not found, then default runtime values are used (and a warning that no
configuration file was found will be shown). <o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>There is
another type of flag that is supported that has a unique pattern: the override
flags. Instead of having a name like cfg, the override flags have the same name
as the variable that they override. Overriding a variable is specifying a
different value on the command line than the one that is in the configuration
file (just like the configuration file overrides the built-in values). For
example, to override the variable DynamicMods to have the value &#8220;M @
16&#8221;, use the override flag:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><b
style='mso-bidi-font-weight:normal'>-DynamicMods &quot;M @ 16&quot;<o:p></o:p></b></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>The double
quotes are necessary on the command line because the value of the variable has
spaces in it.<o:p></o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>After the
flags are parsed, the file arguments are processed. The first file argument
must be the relative or absolute path to the FASTA protein database you want to
search against. Every file argument after that is a relative or absolute path
to a MS/MS spectra data file from which to extract experimental spectra. The
FASTA database filepath must be a valid filepath, it does not support
wildcards. The MS/MS spectra data filepaths, however, do support wildcards. The
provided spectra must be in one of the following supported formats:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><b
style='mso-bidi-font-weight:normal'>mzML 1.x (version 1.1 is recommended)<o:p></o:p></b></span></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>mzXML
3.x<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Mascot
Generic (MGF)<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Bruker
BioTools Data Exchange (btdx.xml)<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Bruker
BAF,FID, Compass .d<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Agilent/Bruker
YEP<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Agilent
MassHunter .d<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Thermo
RAW<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>Waters
RAW<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>ABI
WIFF<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>Note:</span></b><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'> Reading
instrument native formats requires installation of appropriate vendor readers
on the computer (Follow the <a
href="http://proteowizard.sourceforge.net/technical/formats/">link</a> and see
the section &#8220;Vendor-specific requirements&#8221;). Most vendor readers
are available for windows platform only. Linux/Unix/Mac users are encouraged to
use mzML 1.1 format.<o:p></o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:36.75pt;text-indent:-18.75pt;mso-list:
l0 level1 lfo4;tab-stops:list 36.75pt'><a name=UsageResidueMasses></a><a
name=UsagePTMs></a><![if !supportLists]><span style='font-size:14.0pt;
mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>b)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>A
static mass modification is something like carboxymethylation of cysteines,
where all cysteines should be treated as about +57 in MyriMatch and all
subsequent downstream analysis. Refer to the StaticMods variable in the
configuration parameters guide. A dynamic mass modification is something like a
potential oxidation of methionine, where each methionine may be occur as either
its natural mass or about +16. Refer to the DynamicMods variable in the
configuration parameters guide.<o:p></o:p></span></p>

<span style='font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
"Times New Roman";mso-ansi-language:EN-US;mso-fareast-language:EN-US;
mso-bidi-language:AR-SA'><br clear=all style='mso-special-character:line-break;
page-break-before:always'>
</span>

<p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:36.75pt;text-indent:-18.75pt;mso-list:
l0 level1 lfo4;tab-stops:list 36.75pt'><a
name=UsageConfigurationParametersGuide></a><![if !supportLists]><span
style='font-size:14.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>c)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;font-family:"Arial","sans-serif"'>Configuration
parameters guide<o:p></o:p></span></p>

<table class=MsoTableGrid border=1 cellspacing=0 cellpadding=0
 style='border-collapse:collapse;border:none;mso-border-alt:solid windowtext .5pt;
 mso-yfti-tbllook:480;mso-padding-alt:.15in .1in .15in .1in'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;height:30.55pt'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;mso-border-alt:
  solid windowtext .5pt;padding:.15in .1in .15in .1in;height:30.55pt'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:14.0pt;font-family:"Arial","sans-serif"'>Name<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:14.0pt;font-family:"Arial","sans-serif"'>(Type, Default
  Value)<o:p></o:p></span></p>
  </td>
  <td width=852 style='width:639.2pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in;height:30.55pt'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:14.0pt;font-family:"Arial","sans-serif"'>Description<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>NumChargeStates<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 3)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Controls
  the number of charge states that MyriMatch will handle during all stages of
  the program. It is especially important during determination of charge state
  (see DuplicateSpectra for more information).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>OutputSuffix<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The output
  of a MyriMatch job will be a pepXML file for each input file. The string
  specified by this parameter will be appended to each pepXML filename. It is
  useful for differentiating jobs within a single directory.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>ProteinDatabase<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=852 style='width:639.2pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Specifies
  the FASTA protein database to be searched.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>StartSpectraScanNum<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 0)<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>EndSpectraScanNum<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, -1)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A useful
  feature to focus a search on a subset of spectra in a particular data file,
  these two parameters can be set in order to limit the possible range of scan
  numbers that MyriMatch will read from the input data files. By default, all
  tandem mass spectra in the input files are read in for processing.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>StartProteinIndex<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 0)<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>EndProteinIndex<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, -1)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A useful
  feature to focus a search on a subset of proteins in the protein database,
  these two parameters can be set in order to limit the range of proteins that
  MyriMatch will read from the protein database. By default, all proteins in
  the protein database are read in for processing.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>StatusUpdateFrequency<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 5 seconds)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Preprocessing
  spectra and scoring candidates may take a long time. A measure of progress
  through the protein database will be given on intervals that are specified by
  this parameter, measured in seconds.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>UseChargeStateFromMS<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(boolean, false)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If true,
  MyriMatch will use the charge state from the input data if it is available.
  If false, or if charge state is not available from a particular spectrum,
  MyriMatch will use its internal algorithm to determine charge state. If, for
  a given spectrum, MyriMatch uses its internal algorithm to determine charge
  state and the result is multiply charged, that spectrum may be duplicated to
  other charge states (see DuplicateSpectra for more information).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>DuplicateSpectra<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(boolean, true)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If
  MyriMatch determines a spectrum to be multiply charged and this parameter is
  true, the spectrum will be copied and treated as if it was all possible
  charge states from +2 to +&lt;NumChargeStates&gt;. If this parameter is
  false, the spectrum will simply be treated as a +2.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:9'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>MinSequenceMass<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0 Da)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  preprocessing the experimental spectra, any spectrum with a precursor mass
  that is less than the specified mass will be disqualified. This parameter is
  useful to eliminate inherently unidentifiable spectra from an input data set.
  A setting of 500 for example, will eliminate most 3-residue matches and clean
  up the output file quite a lot.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:10'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>MaxSequenceMass<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 10000 Da)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  preprocessing the experimental spectra, any spectrum with a precursor mass
  that exceeds the specified mass will be disqualified.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:11'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>TicCutoffPercentage<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0.98)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>In order to
  maximize the effectiveness of the MVH scoring algorithm, an important step in
  preprocessing the experimental spectra is filtering out noise peaks. Noise
  peaks are filtered out by sorting the original peaks in descending order of
  intensity, and then picking peaks from that list until the cumulative ion
  current of the picked peaks divided by the total ion current (TIC) is greater
  than or equal to this parameter. Lower percentages mean that less of the
  spectrums total intensity will be allowed to pass through preprocessing. See
  the section on Advanced Usage for tips on how to use this parameter
  optimally.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:12'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>NumIntensityClasses<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 3)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Before
  scoring any candidates, experimental spectra have their peaks stratified into
  the number of intensity classes specified by this parameter. Spectra that are
  very dense in peaks will likely benefit from more intensity classes in order
  to best take advantage of the variation in peak intensities. Spectra that are
  very sparse will not see much benefit from using many intensity classes.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:13'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>AdjustPrecursorMass<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(boolean, false)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If true,
  the preprocessing step will correct the precursor mass by adjusting it
  through a specified range in steps of a specified length, finally choosing
  the optimal adjustment. The optimal adjustment is the one that maximizes the
  sum of products of all complementary peaks in the spectrum.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:14'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>MinPrecursorAdjustment<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, -2.5 Da)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When adjusting
  the precursor mass, this parameter sets the lower mass limit of adjustment
  allowable from the original precursor mass, measured in <st1:place w:st="on"><st1:City
   w:st="on">Daltons</st1:City></st1:place>.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:15'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>MaxPrecursorAdjustment<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 2.5 Da)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  adjusting the precursor mass, this parameter sets the upper mass limit of
  adjustment allowable from the original precursor mass, measured in <st1:place
  w:st="on"><st1:City w:st="on">Daltons</st1:City></st1:place>.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:16'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>PrecursorAdjustmentStep<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0.1 Da)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  adjusting the precursor mass, this parameter sets the size of the steps
  between adjustments, measured in <st1:place w:st="on"><st1:City w:st="on">Daltons</st1:City></st1:place>.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:17'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>DeisotopingMode<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 0)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Deisotoping
  a spectrum (consolidating isotopic peak intensities into the monoisotopic
  peaks intensity) during preprocessing will significantly improve precursor
  adjustment, and it may be desirable to keep the deisotoped spectrum around
  for candidate scoring as well. Set to 0, no deisotoping will be used. Set to
  1, deisotoping will be used for precursor adjustment only. Set to 2,
  deisotoping will be used for both precursor adjustment and for candidate
  scoring.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:18;height:12.1pt'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in;height:12.1pt'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>IsotopeMzTolerance<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0.25 m/z)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in;
  height:12.1pt'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  deisotoping a spectrum, an isotopic peak is one that is the mass of a neutron
  higher than another peak, tolerating variation based on the value of this
  parameter. Deisotoping actually traverses the spectrum at multiple charge
  states, starting from the highest (NumChargeStates) and ending at the lowest.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:19'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>ComplementMzTolerance<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0.5 m/z)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  adjusting the precursor mass, this parameter controls how much tolerance
  there is on each side of the calculated m/z when looking for a peaks
  complement.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:20'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>StaticMods<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If a
  residue (or multiple residues) should always be treated as having a
  modification on their natural mass, set this parameter to inform the search engine
  which residues are modified. Residues are entered into this string as a
  space-delimited list of pairs. Each pair is of the form:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>&lt;AA residue character&gt; &lt;mod
  mass&gt;<o:p></o:p></span></b></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Thus, to
  treat cysteine as always being carboxymethylated, this parameter would be set
  to something like the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;C
  57&#8221;<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:21'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>DynamicMods<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>Note: avoid using the
  &#8220;#&#8221; symbol in a configuration file since it begins a comment
  section. Using the &#8220;#&#8221; symbol in a command-line override works
  fine.<o:p></o:p></span></b></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>In order to
  search a database for potential post-translational modifications of candidate
  sequences, the user must configure this parameter to inform the search engine
  which residues may be modified. Residues that are modifiable are entered into
  this string in a space-delimited list of triplets. Each triplet is of the
  form:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>&lt;AA motif&gt; &lt;character to
  represent mod&gt; &lt;mod mass&gt;<o:p></o:p></span></b></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></b></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Thus, to
  search for potentially oxidized methionines and phosphorylated serines, this
  parameter would be set to something like the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;M *
  15.995 S $ 79.966&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The AA
  motif can include multiple residues, and the peptide termini are represented
  by opening &#8220;(&#8220; and closing &#8220;)&#8221; parentheses for the N
  and C termini, respectively. For example, an N terminal deamidation of
  glutamine may be specified by the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;(Q ^
  -17&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If the last
  residue in the motif is not the residue intended to be modifiable, then use
  an exclamation mark to indicate that the residue preceding the mark is the
  modifiable residue. Using the previous example, &#8220;(Q! ^ -17&#8221; is an
  equivalent way to specify it. Another example would be specifying the
  demidation of asparagine when it is N terminal to a glycine, which might look
  like:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;N!G
  @ -17&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Another
  possibility is to specify a block of interchangeable residues in the motif,
  which is supported by the &#8220;[&#8220; and &#8220;]&#8221; brackets. For
  example, to specify a potential phosphorylation on any serine, threonine, or
  tyrosine, use the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;[STY]
  * 79.966&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The
  &#8220;{&#8220; and &#8220;}&#8221; brackets work in the opposite way as the
  &#8220;[&#8220; and &#8220;]&#8221; brackets, i.e. &#8220;{STY} *
  79.966&#8221; specifies a potential phosphorylation on every residue EXCEPT
  serine, threonine, or tyrosine. Both kinds of brackets can be combined with
  the exclamation mark, in which case the exclamation mark should come after
  the block (because the block counts as a single residue). Using the previous
  example, &#8220;[STY]! * 79.966&#8221; is an equivalent way to specify it.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Using the
  negative multi-residue brackets is the best way to indicate the &#8220;any
  residue except&#8221; concept, and it works on single residues as well. For
  example, to specify a mod on lysine except when it is at the C terminus of a
  peptide, use something like the string &#8220;K!{)} # 144&#8221;. Another
  example is specifying the cleavage-blocking homoserine mod in a CnBr digest
  when a serine or threonine is C terminal to a methionine:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;M![ST]
  * -29.99&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Note that
  it is not currently possible to specify (for example) the
  non-cleavage-blocking homoserine lactone mod in a CnBr digest, because the
  motif would extend outside of the peptide sequence itself. In the future a string
  like &#8220;M!){ST} * -17&#8221; might work for that, but for now, if
  &#8220;(&#8220; is used it must be the first character in the motif, and
  likewise if &#8220;)&#8221; is used it must be the last character in a motif.
  <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:22'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>MaxDynamicMods<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 2)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter sets the maximum number of modified residues that may be in any
  candidate sequence.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:23'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>MaxResults<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 5)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter sets the maximum number of candidate sequence matches to report for
  each spectrum.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:24'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>CleavageRules<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, &#8220;Trypsin/P&#8221;)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This important
  parameter allows the user to control the way peptides are generated from the
  protein database. It can be used to configure the search on tryptic peptides
  only, on non-tryptics, or anything in between. It can even be used to test
  multiple residue motifs at a potential cleavage site. This parameter
  describes which amino acids are valid on the N and C termini of a digestion
  site. The parameter is specified in <a
  href="http://www.ebi.ac.uk/ontology-lookup/browse.do?ontName=MS&amp;termId=MS:1001180&amp;termName=cleavage%20agent%20regular%20%25expression">PSI-MS
  regular expression syntax</a> (a limited Perl regular expression syntax).
  MyriMatch can recognize the following protease names and automatically use
  the corresponding regular expression for this parameter.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-spacerun:yes'>&nbsp;</span><o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span><b style='mso-bidi-font-weight:normal'>Protease names:<o:p></o:p></b></span></p>
  <p class=MsoListParagraphCxSpFirst style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Trypsin&#8221;
  (allows for cut after K or R)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Trypsin/P&#8221;
  (normal trypsin cut, disallows cutting when the site is before a proline)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&quot;Chymotrypsin&#8221;
  (allows cut after F,Y,W,L. Disallows cutting before proline)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&quot;TrypChymo&#8221;
  (combines &#8220;Trypsin/P&#8221; and &#8220;Chymotrypsin&#8221; cleavage
  rules)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Lys-C&#8221;<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Lys-C/P&#8221;
  (Lys-C, disallowing cutting before proline)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Asp-N&#8221;<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;PepsinA&#8221;
  (Cuts right after F, L)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;CNBr&#8221;
  (Cyanogen bromide)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Formic_acid&#8221;
  (Formic acid)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;NoEnzyme&#8221;
  (Cuts everywhere)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpLast style='margin-left:.85in;mso-add-space:
  auto'><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A complete list
  of supported protease names can be found <a
  href="http://www.ebi.ac.uk/ontology-lookup/browse.do?ontName=MS&amp;termId=MS:1001045&amp;termName=cleavage%20agent%20name">here</a>.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>Note:</span></b><span
  style='font-family:"Arial","sans-serif"'> CleavageRules can also work with an
  earlier but deprecated regular expression syntax. <b style='mso-bidi-font-weight:
  normal'>We highly discourage users from using the old syntax.</b> Briefly,
  the old syntax is a space-delimited list of cleavage rules, where each
  cleavage rule itself is a space-delimited pair of strings. The first string
  of the cleavage rule specifies the residue or residues that must be
  N-terminal to a potential cleavage site. The second string specifies the
  residue or residues that must be C-terminal to the site. Either string in the
  pair can contain multiple sequences of one or more residues, separated by the
  &#8216;|&#8217; character. A &#8216;.&#8217; character is a wildcard that
  will accept anything. Additionally, the &#8216;[&#8216; and &#8216;]&#8217;
  characters refer to the N and C termini of a protein.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Now that
  you are thoroughly confused, here are examples of single cleavage rules:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>R .</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if the N-terminal residue is R<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>R|K .</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if the N-terminal residue is R <i
  style='mso-bidi-font-style:normal'>or</i> K<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>[ .</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage at the N terminus of a protein<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>. ]</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage at the C terminus of a protein<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The <b
  style='mso-bidi-font-weight:normal'>&#8220;.&#8221;</b> wildcard is important
  because it allows the cleavage routine to work very quickly. However, if you
  wanted to leave out proline from a tryptic digest, you would have to
  explicitly declare the valid residues for both sides of a cleavage site:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>R|K
  A|C|D|E|F|G|H|I|K|L|M|N|Q|R|S|T|V|W|Y</span></b><span style='font-family:
  "Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Remember
  that this parameter is a <i style='mso-bidi-font-style:normal'>list</i> of
  cleavage rules; a real tryptic digest can be declared with two cleavage
  rules:<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:"Arial","sans-serif"'>[|R|K
  . . ]</span></b><span style='font-family:"Arial","sans-serif"'><span
  style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if it is at the N terminus of a protein,<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>or
  if the N-terminal residue is R or K; a site is valid for cleavage<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>if
  it is at the C terminus of a protein<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><span
  style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Also note
  that a cleavage rule can have a residue string of more than one residue,
  allowing for multiple-residue cleavage motifs:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>[M|[ .</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if it is at the N terminus of a protein,<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>or
  if the N-terminal sequence of residues is [M (i.e. the M must<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>be
  at the N terminus of a protein)<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:25'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>NumMinTerminiCleavages<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 2)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>By default,
  when generating peptides from the protein database, a peptide must start
  after a cleavage and end before a cleavage. Setting this parameter to 0 or 1
  will reduce that requirement, so that neither terminus or only one terminus
  of the peptide must match one of the cleavage rules specified in the
  CleavageRules parameter. This parameter is useful to turn a tryptic digest
  into a semi-tryptic digest.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:26'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>NumMaxMissedCleavages<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, -1)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>By default,
  when generating peptides from the protein database, a peptide may contain any
  number of missed cleavages. A missed cleavage is a site within the peptide
  that matches one of the cleavage rules (refer to CleavageRules). Settings
  this parameter to some other number will stop generating peptides from a
  sequence if it contains more than the specified number of missed cleavages.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:27'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>UseAvgMassOfSequences<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(boolean, true)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If true,
  the mass of candidate sequences will be calculated using the average masses of
  its amino acid residues. This parameter should be set based on whether the
  experimental data has precursor masses that are monoisotopic. For example,
  LCQ/LTQ-derived precursors are generally measured by average masses and
  FT-ICR/Orbitrap-derived precursors are generally measured by monoisotopic
  masses.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:28'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>UseSmartPlusThreeModel<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(boolean, true)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Once a
  candidate sequence has been generated from the protein database, MyriMatch
  determines which spectra will be compared to the sequence. For each unique
  charge state of those spectra, a set of theoretical fragment ions is
  generated by one of several different algorithms.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>For +1 and
  +2 precursors, a +1 b and y ion is always predicted at each peptide bond.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>For +3 and
  higher precursors, the fragment ions predicted depend on the way this
  parameter is set. When this parameter is true, then for each peptide bond, an
  internal calculation is done to estimate the basicity of the b and y fragment
  sequence. The precursors protons are distributed to those ions based on that
  calculation, with the more basic sequence generally getting more of the
  protons. For example, when this parameter is true, each peptide bond of a +3
  precursor will either generate a +2 b<sub>i</sub> and a +1 y<sub>i</sub> ion,
  or a +1 b<sub>i</sub> and a +2 y<sub>i</sub> ion. For a +4 precursor,
  depending on basicity, a peptide bond breakage may result in a +1 b<sub>i</sub>
  and a +3 y<sub>i</sub> ion, a +2 b<sub>i</sub> and a +2 y<sub>i</sub> ion, or
  a +3 b<sub>i</sub> and a +1 y<sub>i</sub> ion. When this parameter is false,
  however, ALL possible charge distributions for the fragment ions are
  generated for every peptide bond. So a +3 sequence of length 10 will always
  have theoretical +1 y5, +2 y5, +1 b5, and +2 b5 ions.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:29'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>PrecursorMzTolerance<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 1.25 daltons/z)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A generated
  sequence candidate is only compared to an experimental spectrum if the candidates
  mass is within this tolerance of the experimental spectrums precursor mass.
  This value is given in Daltons/z units, but the actual tolerance is
  calculated by multiplying by the charge state. This parameter should be set
  to the tolerance that is desired for +1 spectra. At the default value, the
  precursor mass tolerances are 1.25, 2.5, and 3.75 Da for the first three
  charge states, respectively. This parameter can also be set to use PPM
  (parts-per-million) accuracy when searching ultra-high resolution data (see
  below). <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:30'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>PrecursorMzToleranceUnits<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, daltons)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter specifies the mass units used for PrecursorMzTolerance. Allowed
  values are &#8220;daltons&#8221; or &#8220;ppm&#8221; (parts-per-million).
  This parameter is case-insensitive. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:31'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>FragmentMzTolerance<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0.5 m/z)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter controls how much tolerance there is on each side of the calculated
  m/z when looking for an ion fragment peak during candidate scoring.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:32'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>FragmentMzToleranceUnits<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, daltons)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This parameter
  specifies the mass units used for FragmentMzTolerance. Allowed values are
  &#8220;daltons&#8221; or &#8220;ppm&#8221; (parts-per-million). This
  parameter is case-insensitive.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:33'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>ProteinSampleSize<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 100)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Before
  beginning sequence candidate generation and scoring, MyriMatch will do a
  random sampling of the protein database to get an estimate of the number of
  comparisons that will be done by the job. The bigger the sample size, the
  longer this estimate will take and the more accurate it will be. Of course,
  if there are fewer proteins in the database than the sample size, all
  proteins will be used in the sampling and the number of comparisons will be
  exact.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:34;height:12.1pt'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in;height:12.1pt'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>ClassSizeMultiplier<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 2)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in;
  height:12.1pt'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  stratifying peaks into a specified, fixed number of intensity classes, this
  parameter controls the size of each class relative to the class above it
  (where the peaks are more intense). At default values, if the best class, A,
  has 1 peak in it, then class B will have 2 peaks in it and class C will have
  4 peaks.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:35'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>NumBatches<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 50)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter sets a number of batches per node to strive for when using the
  MPI-based parallelization features. Setting this too low means that some
  nodes will finish before others (idle processor time), while setting it too
  high means more overhead in network transmission as each batch is smaller. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:36'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>ThreadCountMultiplier<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 10)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>MyriMatch
  is designed to take advantage of (symmetric) multiprocessor systems by
  multithreading the database search. A search process on an SMP system will
  spawn one worker thread for each processing unit (where a processing unit can
  be either a core on a multi-core CPU or a separate CPU entirely). The main
  thread then generates a list of worker numbers which is equal to the number
  of worker threads multiplied by this parameter. The worker threads then take
  a worker number from the list and use that number to iterate through the
  protein list. It is possible that one thread will be assigned all the
  proteins that generate a few candidates while another thread is assigned all
  the proteins that generate many candidates, resulting in one thread finishing
  its searching early. By having each thread use multiple worker numbers, the
  chance of one thread being penalized for picking all the easy proteins is
  reduced because if it finishes early it can just pick a new number. The only
  disadvantage to this system is that picking the new number incurs some
  overhead because of synchronizing with the other worker threads that might be
  trying to pick a worker number at the same time. The default value is a nice
  compromise between incurring that overhead and minimizing wasted time.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:37;mso-yfti-lastrow:yes'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>UseMultipleProcessors<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(boolean, true)<o:p></o:p></span></p>
  </td>
  <td width=852 valign=top style='width:639.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If true,
  each process will use all the processing units available on the system it is
  running on.<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal><span style='font-size:14.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:14.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
font-family:"Arial","sans-serif"'>Int<a name=InterpretingResults></a>erpreting
results<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:37.5pt;text-indent:-19.5pt;mso-list:l3 level1 lfo8;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-size:14.0pt;
mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>a)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>S<a
name=InterpretingResultsSearchOutput></a>earch-time output of MyriMatch serves
several purposes. The majority of the output will usually be progress
information, telling the user which part of the job that MyriMatch is currently
working on, and in some cases how far along into that part the job is. There
will be periodic updates when MyriMatch is preprocessing spectra and when it is
generating candidates from the database and comparing those candidates against
the spectra. In a multi-process (MPI) job, there will also be progress
information on bulk transfers of data over the network. Additionally, MyriMatch
will display statistics on the spectra that remain after preprocessing,
specifically the average number of peaks in a spectrum before and after
preprocessing. Also provided is the average number of the percentage of peaks
that were filtered out by the preprocessing step. Finally, in the case of an
MPI job, when the database search is complete each node that took part in the
search will display statistics detailing the work that node did. The lines will
be like:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='font-family:"Arial","sans-serif"'>Process #1 (foohost)
stats: &lt;numBatches&gt; / &lt;numProteins&gt; /
&lt;numCandidatesGenerated&gt; / &lt;numCandidatesSearched&gt; /
&lt;numComparisonsDone&gt;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:37.5pt;text-indent:-19.5pt;mso-list:l3 level1 lfo8;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-size:14.0pt;
mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>b)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>O<a
name=InterpretingResultsSQTFile></a>ne pepXML (a database search output format
originally developed at the SPC) file is produced for every input spectra file
that a MyriMatch job searches. The file contains an entry for each spectrum
kept during the search (i.e. only the spectra that were not obviously junk)
that was compared to any candidate. Only the best MaxResults (a config
parameter, defaulting to 5) matches for each spectrum are kept and output in
the file. The pepXML format is quite versatile, but does have a few
limitations. It is not possible to fully encode MyriMatch&#8217;s CleavageRules
variable if multi-residue motifs are used (in the rule &#8220;[|[M|K|R .&#8221;
the &#8220;[M&#8221; part is a multi-residue motif), and it is not possible to
fully encode Static/DynamicMods if multi-residue motifs are used (in the mod
&#8220;(Q ^ -17&#8221; the &#8220;(Q&#8221; part is a multi-residue motif). We
have proposed extensions to pepXML to support these motif-oriented
capabilities.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:37.5pt;text-indent:-19.5pt;mso-list:l3 level1 lfo8;
tab-stops:list 37.5pt'><a name=InterpretingResultsValidation></a><![if !supportLists]><span
style='font-size:14.0pt;mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";
mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>c)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp; </span></span></span><![endif]><span
style='font-family:"Arial","sans-serif"'>To validate results generated by
MyriMatch (and also several other popular search engines), users can pass
pepXML files to IDPickerQonvert (in preparation for analyzing the results in
the IDPicker suite). This approach only works when the protein database that
was searched included distracter (decoy) proteins with a common prefix in their
name that is not a prefix in any of the valid proteins. Most often, the
database that was searched will include proteins in their forward and reverse
orders, with the reversed protein having a prefix added to the name like
&#8220;rev_&#8221;.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

</div>

</body>

</html>
