<html>

<head>
    <meta charset="utf-8">
    <link rel="stylesheet" type="text/css" href="../../shared/SkylineStyles.css">
    <script src="../../shared/skyline.js" type="text/javascript"></script>
</head>

<body onload="skylineOnload();">
    <h1 class="document-title">Skyline Absolute Quantification</h1>
    <h1>Introduction</h1>
    <p>
        This tutorial covers how to determine the absolute abundance of a target protein using Selected Reaction Monitoring (SRM) mass spectrometry.
        Specifically, we will demonstrate how to use an external calibration curve with an internal standard heavy labeled peptide for normalization.
    </p>
    <p>
        Peptide absolute abundance measurements can be obtained using either a single-point or a multiple-point calibration. Single-point internal
        calibration absolute abundance measurements are generated by spiking into a target sample a heavy labeled “standard” version of the target
        peptide that is of known abundance. The absolute abundance of the “sample” target peptide is obtained by calculating the relative abundance of
        the light “sample” target peptide to the heavy “standard” target peptide<sup>1</sup>. One drawback is that this approach assumes that a
        light-to-heavy ratio of 2 implies that the light peptide is actually twice as abundant as the heavy peptide – this is referred to as having a
        peptide response with a slope of 1. Furthermore, this approach of using a single point internal calibration makes the assumption that both the
        light and the heavy peptide are within the linear range of the mass spectrometry detector. However, these assumptions are not always
        correct<sup>2,3,4,5</sup>.
    </p>
    <p>
        Multiple-point external calibration experiments correct for situations where the peptide response does not have a slope of 1. This calibration
        is done by measuring the signal intensity of a “standard” peptide at multiple calibration points of known abundance and generating a
        calibration curve. This calibration curve can then be used to calculate the concentration of the target peptide in a sample, given the signal
        intensity of that peptide in that sample<sup>3</sup>. One drawback is that this method requires multiple injections into the mass spectrometer
        to build a calibration curve. 
    </p>
    <p>
        To improve the precision of absolute abundance measurements using an external calibration curve, stable isotope labeled internal standards are
        often used<sup>6</sup>. Imprecise measurements of the ion intensity of a peptide often arise from sample preparation, autosampler or
        chromatographic irregularities. By adding an identical quantity of a standard heavy labeled peptide to each of the calibrants and the sample,
        one is able to measure the ratio of calibrant-to-standard or sample-to-standard, also known as normalization. This approach is favored as this
        ratio is unaffected by some sample preparation, autosampler or chromatographic irregularities. Consequently, by performing peptide absolute
        quantification using an external calibration curve and normalization to an internal standard heavy labeled peptide one is able to obtain the
        most accurate and precise measurements while minimizing the amount of valuable sample that has to be used.
    </p>
    <h1>Experimental Overview</h1>
    <p>
        This tutorial will work with data published in Stergachis et al.<sup>7</sup> where the absolute abundance of GST-tagged proteins were measured
        using a “proteotypic” peptide present within the GST-tag (<b>Tutorial Figure 1A</b>). For any absolute quantification experiment, it
        is critical to first identify one or more “proteotypic” peptides that will be used to quantify the protein of interest. The peptide IEAIPQIDK
        was identified as “proteotypic” based on its strong signal intensity relative to other tryptic peptides in the GST-tag (unpublished). Also,
        this peptide uniquely identifies this schistosomal GST-tag as opposed to other human glutathione-binding proteins. 
    </p>
    <p>
        For this experiment, FOXN1 protein containing an in frame GST-tag was generated using <i>in vitro</i> transcription/translation and
        full-length proteins were purified using glutathione resin (<b>Tutorial Figure 1B</b>). Heavy labeled IEAIPQIDK peptide was then
        spiked into the elution buffer and the sample was digested and analyzed using selected reaction monitoring (SRM) on a Thermo TSQ Vantage
        triple-quadrupole mass spectrometer. An external calibration curve was generated using different quantities of a light IEAIPQIDK peptide that
        was purified to &gt;97% purity and the concentration determined by amino acid analysis. Heavy labeled IEAIPQIDK peptide was also spiked into
        these calibrants at the same concentration as in the FOXN1-GST sample (<b>Tutorial Figure 1C</b>). It is important to note that it
        does not matter what the concentration of the heavy peptide is in each of the samples, so long as it is the same. However, it is best if the
        amount of heavy peptide in the samples is similar to the amount of light peptide originating from FOXN1-GST. Also, it is best if the
        concentration of the light peptide originating from FOXN1-GST falls somewhere in the middle of the concentration range tested using the
        different calibrants.
    </p>
    <p>
        <img src="s-experimental-overview.png" />
    </p>
    <p>
        <b>Tutorial Figure 1. Experimental Overview</b>
    </p>
    <p>
        (<b>A</b>) Schistosomal GST-tag protein sequence. The tryptic peptide used for quantification purposes is indicated in red.
    </p>
    <p>
        (<b>B</b>) Schematic of the synthesis, enrichment, digestion and analysis of tagged proteins.
    </p>
    <p>
        (<b>C</b>) Samples monitored and the abundance of light and heavy IEAIPQIDK peptide in each.
    </p>
    <h1>Getting Started</h1>
    <p>
        To start this tutorial, download the following ZIP file:
    </p>
    <p>
        <a href="https://skyline.gs.washington.edu/tutorials/AbsoluteQuant.zip">https://skyline.gs.washington.edu/tutorials/AbsoluteQuant.zip</a>
    </p>
    <p>
        Extract the files in it to a folder on your computer, like:
    </p>
    <p>
        C:\Users\absterga\Documents
    </p>
    <p>
        This will create a new folder:
    </p>
    <p>
        C:\Users\absterga\Documents\AbsoluteQuant
    </p>
    <p>
        If you have been using Skyline prior to starting this tutorial, it is a good idea to revert Skyline to its default settings. To do so: 
    </p>
    <ul>
        <li>Start Skyline.</li>
        <li>On the <b>Start Page</b>, click <b>Blank Document</b> which looks like this: </li>
    </ul>
    <p>
        <img src="../../shared/en/skyline-blank-document.png" />
    </p>
    <ul>
        <li>On the <b>Settings</b> menu, click <b>Default</b>. </li>
        <li>Click <b>No</b> on the form asking if you want to save the current settings.</li>
    </ul>
    <p>
        The document settings in this instance of Skyline have now been reset to the default.
    </p>
    <p>
        Since this tutorial covers a proteomics topic, you can choose the proteomics interface by doing the following:
    </p>
    <ul class="keep-next">
        <li>Click the user interface control in the upper right-hand corner of the Skyline window, and click <b>Proteomics interface</b> which looks like this: </li>
    </ul>
    <p>
        <img src="../../shared/en/proteomics-interface.png" />
    </p>
    <p>
        Skyline is operating in proteomics mode which is displayed by the protein icon <img src="../../shared/protein-icon.png" /> in the upper
        right-hand corner of the Skyline window.
    </p>
    <p>
        Now, you should be looking at Skyline with a new empty document.
    </p>
    <h1>Generating a Transition List</h1>
    <p>
        Before you insert a peptide sequence into Skyline, it is important to make sure that all of the peptide and transition settings are correctly
        configured for this experiment. The settings described below are designed for <sup>13</sup>C<sub>6</sub><sup>15</sup>N<sub>2</sub> L-Lysine
        labeled internal standard peptides. If you are using a different isotope, please choose the appropriate isotope modification in the Peptide
        settings configuration. 
    </p>
    <h2>Configuring transition settings:</h2>
    <ul>
        <li>On <b>Settings</b> menu, click <b>Transition Settings</b>.</li>
        <li>Click the <b>Prediction</b> tab. </li>
        <li>From the <b>Collision energy</b> dropdown list choose the instrument that you will be using for your measurements. For this
            experiment, a <b>Thermo TSQ Vantage</b> was used for all measurements. </li>
    </ul>
    <p class="keep-next">
        The form should now look like this:
    </p>
    <p>
        <img src="s-1.png" />
    </p>
    <ul>
        <li>Click the <b>Filter</b> tab. </li>
        <li>Here you can accept the default settings for ion charges and types. </li>
        <li>In the <b>Special ions</b> list, uncheck the choice “N-terminal to Proline”.</li>
        <li>In the <b>Product ion selection</b> – <b>From</b> field, choose “ion 3”.</li>
        <li>In the <b>Product ion selection</b> – <b>To</b> field, choose “last ion - 1”.</li>
    </ul>
    <div style="margin-left : 40px;">
        <table>
            <tr>
                <td>Note: Normally you would want to choose “last ion” which is equivalent to y<sub>n-1</sub> (or y8 for a 9 amino acid peptide like
                    IEAIPQIDK used in this experiment. However, the experiment used “last ion – 1” due to a confusion between it and “n-1”.</td>
            </tr>
        </table>
    </div>
    <p class="keep-next">
        The <b>Transition Settings</b> form should look like this: 
    </p>
    <p>
         <img src="s-2.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <h2>Configuring peptide settings:</h2>
    <ul>
        <li>On the <b>Settings</b> menu, click <b>Peptide Settings</b>.</li>
        <li>Click the <b>Modifications</b> tab. </li>
        <li>Click the <b>Edit List</b> button to the right of the <b>Isotope modifications</b> list.</li>
        <li>Click the <b>Add</b> button in the <b>Edit Isotope Modifications</b> form that appears.</li>
        <li>In <b>Name</b> dropdown list, choose “Label:13C(6)15N(2) (C-term K)”.</li>
    </ul>
    <p class="keep-next">
        The <b>Edit Isotope Modification</b> form should now look like this:
    </p>
    <p>
        <img src="s-3.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
        <li>In the <b>Isotope modifications</b> list, check the new “Label:13C(6)15N(2) (C-term K)” modification.</li>
    </ul>
    <p class="keep-next">
        The <b>Peptide Settings</b> form should now look like this:
    </p>
    <p>
        <img src="s-4.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button. </li>
    </ul>
    <p>
        Since the experiment uses a heavy labeled internal standard peptide, the <b>Internal standard type</b> drop-list can be left as the
        default <b>heavy</b>.
    </p>
    <h2>Inserting a peptide sequence:</h2>
    <ul>
        <li>On the <b>Edit</b> menu, choose <b>Insert</b> and click <b>Peptides</b>.</li>
        <li>Paste “IEAIPQIDK” into the <b>Peptide Sequence</b> cell and “GST-tag” into the <b>Protein Name</b> cell.</li>
    </ul>
    <p>
        <img src="s-5.png" />
    </p>
    <ul>
        <li>Click the <b>Insert</b> button.</li>
    </ul>
    <p class="keep-next">
        After performing the above steps, the main Skyline should now look like this:
    </p>
    <p>
        <img src="s-6.png" />
    </p>
    <p>
        Before exporting your first transition list, first save your document to the AbsoluteQuant folder by doing the following:
    </p>
    <ul>
        <li>On the <b>File</b> menu, click <b>Save</b> (ctrl-S).</li>
        <li>Navigate to the AbsoluteQuant folder.</li>
        <li>In the <b>File name</b> field, enter “AbsoluteQuantTutorial”.</li>
        <li>Click the <b>Save</b> button.</li>
    </ul>
    <h2>Exporting a transition list:</h2>
    <p>
        If you were running this experiment yourself on a mass spectrometer, you would need an instrument method. Skyline can export directly to an
        instrument method, given a template method, but here you will export a transition list, which you could import into a template instrument
        method yourself at another time. Do the following to achieve this:
    </p>
    <ul>
        <li>On the <b>File</b> menu, choose <b>Export</b> and click <b>Transition List</b>.</li>
    </ul>
    <p>
        The <b>Export Transition List</b> form should appear as shown below, with “Thermo” automatically chosen in the <b>Instrument
            type</b> list, based on the “Thermo TSQ Vantage” value you chose from the <b>Collision energy</b> dropdown list in the
        <b>Transition Settings</b> – <b>Prediction</b> tab.
    </p>
    <p>
        <img src="s-7.png" />
    </p>
    <p>
        You can also see that all of the other settings are appropriate for this very simple target list.
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
        <li>In the <b>File name</b> field, enter “GST-tag”.</li>
        <li>Click the <b>Save</b> button. </li>
    </ul>
    <p>
        The resulting transition list was imported into a Thermo TSQ Vantage triple-quadrupole mass spectrometer template method file. You may now want
        to open the resulting GST-tag.csv file in Excel or a text editor to see what it looks like.
    </p>
    <h1>Analyzing SRM Data from Calibrants</h1>
    <p>
        In this next section you will work with the nine samples indicated in <b>Tutorial Figure 1C</b>. You will import the .RAW files into
        Skyline to view the data. Data will be imported into the saved Skyline document that was generated in the previous section. The files that you
        will import are contained in the AbsoluteQuant folder you created for this tutorial and are named:
    </p>
    <ul>
        <li>Standard_1.RAW</li>
        <li>Standard_2.RAW</li>
        <li>Standard_3.RAW</li>
        <li>Standard_4.RAW</li>
        <li>Standard_5.RAW</li>
        <li>Standard_6.RAW</li>
        <li>Standard_7.RAW</li>
        <li>Standard_8.RAW</li>
        <li>FOXN1-GST.RAW</li>
    </ul>
    <p>
        These RAW files were collected in a random order and were interspersed among a larger set of runs. The results as fully processed with Skyline
        can be found in the <b>Supplemental Data 2</b> for the original paper (<a
            href="http://proteome.gs.washington.edu/supplementary_data/IVT_SRM/Supplementary%20Data%202.sky.zip">http://proteome.gs.washington.edu/supplementary_data/IVT_SRM/Supplementary%20Data%202.sky.zip</a>).
    </p>
    <p>
        Before you look at the FOXN1-GST sample, you should first become familiar with the standards.
    </p>
    <h2>Importing RAW files into Skyline:</h2>
    <ul>
        <li>On the <b>File</b> menu, choose <b>Import</b> and click <b>Results</b>.</li>
        <li>Accept the default option <b>Add single-injection replicates in files</b>.</li>
    </ul>
    <p class="keep-next">
        The <b>Import Results</b> form should look like this:
    </p>
    <p>
        <img src="s-8.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <ul class="keep-next">
        <li>In the <b>Import Results Files</b> form, find and select all eight “Standard” RAW files as shown below.</li>
    </ul>
    <p>
        <img src="s-9.png" />
    </p>
    <ul>
        <li>Click <b>Open</b> to import the files.</li>
        <li>When presented with the option to remove the “Standard_” prefix in creating replicate names, choose the <b>Do not remove</b>
            option.</li>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <p>
        It may take a few moments for Skyline to import all of the RAW files.
    </p>
    <p>
        To ensure that the chromatographic peaks for each of the standards looks good, it is best to view all of the chromatograms next to each other
        in a tiled view. Do the following to achieve this layout:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Arrange Graphs</b> and click <b>Tiled</b> (Ctrl-T). </li>
        <li>Select the IEAIPQIDK peptide in the <b>Targets</b> view.</li>
    </ul>
    <p>
        You will see the heavy (Blue) and light (Red) total chromatograms displayed together in a graph for each standard as shown below:
    </p>
    <p>
        <img src="s-10.png" />
    </p>
    <p>
        What to inspect when looking at the chromatographic traces for the standards:
    </p>
    <ul>
        <li>Make sure that the correct peak is selected for both the heavy and light trace of each standard.</li>
        <li>Make sure the peak shapes look Gaussian and do not show an excessively jagged appearance. If this is the case, it may be best to rerun
            your samples.</li>
        <li>Make sure that the retention time is similar for the different standards. Widely varying retention times often indicate poor
            chromatography.</li>
    </ul>
    <h1>Analyzing SRM Data from FOXN1-GST Sample</h1>
    <p>
        Next you will want to import the FOXN1-GST.RAW file into the current Skyline document using the same instructions as detailed above.
    </p>
    <ul>
        <li>On the <b>File</b> menu, choose <b>Import</b> and click <b>Results</b>.</li>
        <li>Click the <b>OK</b> button.</li>
        <li>Select the FOXN1-GST.RAW file.</li>
        <li>Click the <b>Open</b> button.</li>
    </ul>
    <p>
        To ensure that this sample looks good, inspect the chromatographic trace, the fragmentation pattern and the retention time of both the heavy
        and light peak. To help, you will want to show summary plots as follows:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Retention Times</b> and click <b>Replicate Comparison</b> (F8).</li>
        <li>On the <b>View</b> menu, choose <b>Peak Areas</b> and click <b>Replicate Comparison</b> (F7).</li>
        <li>Right-click the <b>Peak Areas</b> graph, choose <b>Normalized To</b> and click <b>Total</b> to view the
            relative contribution of each transition to the total signal intensity.</li>
        <li>On the <b>Settings</b> menu, click <b>Integrate All</b> to tell Skyline this is a refined method where all transitions
            are expected to have measurable signal.</li>
    </ul>
    <p>
        Arrange the windows as follows for better viewing:
    </p>
    <ul>
        <li>On the <b>View</b> menu, choose <b>Arrange Graphs</b> and click <b>Tabbed</b> (Ctrl-Shift-T). </li>
        <li>Click on the title bars of the <b>Replicate Comparison</b> graphs and drag them to the right edge of the main Skyline window.
        </li>
    </ul>
    <p class="keep-next">
        You can then select either the light precursor:
    </p>
    <p>
        <img src="s-11.png" />
    </p>
    <p class="keep-next">
        Or the heavy precursor:
    </p>
    <p>
        <img src="s-12.png" />
    </p>
    <p>
         Verify the following:
    </p>
    <ul>
        <li>The peak shape looks Gaussian and does not show an excessively jagged appearance.</li>
        <li>The retention time is similar for the standards and the FOXN1-GST sample.</li>
        <li>The relative contribution of each transition to the total signal is similar for each sample. </li>
    </ul>
    <p>
        If this does not appear to be so, then an incorrect peak is likely selected for one of the samples, or a transition may be experiencing
        noticeable interference. 
    </p>
    <p>
        Another way to view the data is to follow these steps: 
    </p>
    <ul>
        <li>Select the IEAIPQIDK peptide in <b>Targets</b> view again. </li>
        <li>Right-click on the <b>Peak Areas</b> graph, choose <b>Normalized To</b> and click <b>Heavy</b> to view the
            light-to-heavy ratio for each standard and the FOXN1-GST sample.</li>
    </ul>
    <p>
        The values displayed in the <b>Peak Areas</b> graph will be the ones used later in the calibration curve. It can be easily observed
        from this graph that the light-to-heavy ratio for the FOXN1-GST sample falls somewhere in the middle of the ratios from the calibration
        points. This is ideal, as this portion of the calibration curve is best for quantification purposes. 
    </p>
    <p>
        <img src="s-13.png" />
    </p>
    <h1>Generating a Calibration Curve</h1>
    <p>
        For this tutorial, you will make a calibration curve in Skyline. 
    </p>
    <h2>Configuring quantification settings:</h2>
    <ul>
        <li>On the <b>Settings</b> menu, click <b>Peptide Settings</b>.</li>
        <li>Click the <b>Quantification</b> tab.</li>
        <li>In the <b>Regression Fit</b> dropdown list, choose “Linear”.</li>
        <li>In the <b>Normalization Method</b> dropdown list, choose “Ratio to Heavy”.</li>
        <li>In the <b>Units</b> field, enter “fmol/ul”.</li>
    </ul>
    <p class="keep-next">
        The <b>Peptide Settings</b> form should look like this:
    </p>
    <p>
        <img src="s-14.png" />
    </p>
    <ul>
        <li>Click the <b>OK</b> button.</li>
    </ul>
    <h2>Specify the analyte concentrations of the external standards:</h2>
    <ul>
        <li>On the <b>View</b> menu, click <b>Document Grid</b>.</li>
        <li>In the top left of the <b>Document Grid</b>, click the <b>Reports</b> dropdown list and choose
            <b>Replicates</b>.</li>
        <li>Copy the following data and paste it into the <b>Document Grid</b> to set each of the “Standard_#” replicates <b>Sample
                Type</b> to “Standard” and their <b>Analyte Concentrations</b> to the values below.</li>
    </ul>
    <div style="margin-left : 20px;">
        <table>
            <tr>
                <td>Standard</td>
                <td>40</td>
            </tr>
            <tr>
                <td>Standard</td>
                <td>12.5</td>
            </tr>
            <tr>
                <td>Standard</td>
                <td>5</td>
            </tr>
            <tr>
                <td>Standard</td>
                <td>2.5</td>
            </tr>
            <tr>
                <td>Standard</td>
                <td>1</td>
            </tr>
            <tr>
                <td>Standard</td>
                <td>0.5</td>
            </tr>
            <tr>
                <td>Standard</td>
                <td>0.25</td>
            </tr>
            <tr>
                <td>Standard</td>
                <td>0.1</td>
            </tr>
        </table>
    </div>
    <ul>
        <li>In the row for FOXN1-GST, set its <b>Sample Type</b> to “Unknown”.</li>
    </ul>
    <p class="keep-next">
        The <b>Document Grid: Replicates</b> form should look like this:
    </p>
    <p>
        <img src="s-15.png" />
    </p>
    <h2>View the calibration curve</h2>
    <ul>
        <li>On the <b>View</b> menu, click <b>Calibration Curve</b>.</li>
        <li>In the <b>Document Grid</b>, click the FOXN1-GST link in the <b>Replicate</b> column.</li>
    </ul>
    <p class="keep-next">
        You should see a graph that looks like this:
    </p>
    <p>
        <img src="s-16.png" />
    </p>
    <p>
        The slope and intercept are displayed on the calibration curve. They can be used to convert between peak area ratio and concentration with a
        standard <b>y = m * x + b</b> equation 
    </p>
    <p>
        <i>concentration = (ratio - intercept) / slope</i>
    </p>
    <p>
        Also, calculated concentration of the currently selected replicate is displayed on the calibration form. Here the concentration of the unknown
        sample is shown as 1.8554 fmol/ul.
    </p>
    <h1>Conclusion</h1>
    <p>
        This tutorial presented the advantages of different absolute abundance experimental setups and demonstrated how to determine absolute
        abundances using an external calibration curve with an internal standard heavy labeled peptide for locating the correct integration range even
        at very low analyte abundance and for normalization. This method provides accurate and precise absolute measurements while minimizing the
        amount of valuable sample that has to be used during the experiment. 
    </p>
    <h1>Reference List</h1>
    <ol>
        <li>Gerber, S.A., Rush, J., Stemman, O., Kirschner, M.W. &amp; Gygi, S.P. Absolute quantification of proteins and phosphoproteins from cell
            lysates by tandem MS. <i>Proceedings of the National Academy of Sciences of the United States of America</i> <b>100</b>,
            6940-6945 (2003).</li>
        <li>MacCoss, M.J., Wu, C.C., Matthews, D.E. &amp; Yates, J.R. Measurement of the isotope enrichment of stable isotope-labeled proteins using
            high-resolution mass spectra of peptides. <i>Analytical Chemistry</i> <b>77</b>, 7646-53 (2005).</li>
        <li>Lavagnini, I. &amp; Magno, F. A statistical overview on univariate calibration, inverse regression, and detection limits: Application to
            gas chromatography/mass spectrometry technique. <i>Mass spectrometry reviews</i> <b>26</b>, 1-18 </li>
        <li>Watson, J.T. Mass Spectrometry. <i>Methods in Enzymology</i> <b>193</b>, 86–106 (1990).</li>
        <li>Patterson, B.W. &amp; Wolfe, R.R. Concentration dependence of methyl palmitate isotope ratios by electron impact ionization gas
            chromatography/mass spectrometry. <i>Biological mass spectrometry</i> <b>22</b>, 481-6 (1993). </li>
        <li>MacCoss, M.J., Toth, M.J. &amp; Matthews, D.E. Evaluation and optimization of ion-current ratio measurements by selected-ion-monitoring
            mass spectrometry. <i>Analytical chemistry</i> <b>73</b>, 2976-84 (2001).</li>
        <li>Stergachis, A., MacLean, B., Lee, K., Stamatoyannopoulos, J. A., &amp; MacCoss, M. J., Rapid empirical discovery of optimal peptides for
            targeted proteomics <i>Nature Methods</i> In press</li>
    </ol>
</body>

</html>