<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=us-ascii">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 12">
<meta name=Originator content="Microsoft Word 12">
<link rel=File-List href="index_files/filelist.xml">
<title>MyriMatch</title>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Matt Chambers</o:Author>
  <o:LastAuthor>Chambers, Matthew</o:LastAuthor>
  <o:Revision>451</o:Revision>
  <o:TotalTime>2321</o:TotalTime>
  <o:Created>2007-05-31T15:23:00Z</o:Created>
  <o:LastSaved>2013-10-11T18:52:00Z</o:LastSaved>
  <o:Pages>5</o:Pages>
  <o:Words>4032</o:Words>
  <o:Characters>22989</o:Characters>
  <o:Company>MSRC</o:Company>
  <o:Lines>191</o:Lines>
  <o:Paragraphs>53</o:Paragraphs>
  <o:CharactersWithSpaces>26968</o:CharactersWithSpaces>
  <o:Version>12.00</o:Version>
 </o:DocumentProperties>
</xml><![endif]-->
<link rel=themeData href="index_files/themedata.thmx">
<link rel=colorSchemeMapping href="index_files/colorschememapping.xml">
<!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:Zoom>90</w:Zoom>
  <w:DisplayBackgroundShape/>
  <w:SpellingState>Clean</w:SpellingState>
  <w:GrammarState>Clean</w:GrammarState>
  <w:TrackMoves>false</w:TrackMoves>
  <w:TrackFormatting/>
  <w:PunctuationKerning/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:DoNotPromoteQF/>
  <w:LidThemeOther>EN-US</w:LidThemeOther>
  <w:LidThemeAsian>ZH-CN</w:LidThemeAsian>
  <w:LidThemeComplexScript>X-NONE</w:LidThemeComplexScript>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:DontGrowAutofit/>
   <w:SplitPgBreakAndParaMark/>
   <w:DontVertAlignCellWithSp/>
   <w:DontBreakConstrainedForcedTables/>
   <w:DontVertAlignInTxbx/>
   <w:Word11KerningPairs/>
   <w:CachedColBalance/>
  </w:Compatibility>
  <w:BrowserLevel>MicrosoftInternetExplorer4</w:BrowserLevel>
  <m:mathPr>
   <m:mathFont m:val="Cambria Math"/>
   <m:brkBin m:val="before"/>
   <m:brkBinSub m:val="&#45;-"/>
   <m:smallFrac m:val="off"/>
   <m:dispDef/>
   <m:lMargin m:val="0"/>
   <m:rMargin m:val="0"/>
   <m:defJc m:val="centerGroup"/>
   <m:wrapIndent m:val="1440"/>
   <m:intLim m:val="subSup"/>
   <m:naryLim m:val="undOvr"/>
  </m:mathPr></w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" DefUnhideWhenUsed="false"
  DefSemiHidden="false" DefQFormat="false" LatentStyleCount="267">
  <w:LsdException Locked="false" QFormat="true" Name="Normal"/>
  <w:LsdException Locked="false" QFormat="true" Name="heading 1"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 2"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 3"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 4"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 5"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 6"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 7"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 8"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="heading 9"/>
  <w:LsdException Locked="false" SemiHidden="true" UnhideWhenUsed="true"
   QFormat="true" Name="caption"/>
  <w:LsdException Locked="false" QFormat="true" Name="Title"/>
  <w:LsdException Locked="false" Priority="1" Name="Default Paragraph Font"/>
  <w:LsdException Locked="false" QFormat="true" Name="Subtitle"/>
  <w:LsdException Locked="false" QFormat="true" Name="Strong"/>
  <w:LsdException Locked="false" QFormat="true" Name="Emphasis"/>
  <w:LsdException Locked="false" Priority="99" Name="No List"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true"
   Name="Placeholder Text"/>
  <w:LsdException Locked="false" Priority="1" QFormat="true" Name="No Spacing"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 1"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="99" SemiHidden="true" Name="Revision"/>
  <w:LsdException Locked="false" Priority="34" QFormat="true"
   Name="List Paragraph"/>
  <w:LsdException Locked="false" Priority="29" QFormat="true" Name="Quote"/>
  <w:LsdException Locked="false" Priority="30" QFormat="true"
   Name="Intense Quote"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 1"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 1"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 1"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 1"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 1"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 1"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 1"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 2"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 2"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 2"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 2"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 2"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 2"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 2"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 2"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 3"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 3"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 3"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 3"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 3"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 3"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 3"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 3"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 4"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 4"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 4"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 4"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 4"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 4"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 4"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 4"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 5"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 5"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 5"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 5"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 5"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 5"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 5"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 5"/>
  <w:LsdException Locked="false" Priority="60" Name="Light Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="61" Name="Light List Accent 6"/>
  <w:LsdException Locked="false" Priority="62" Name="Light Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="63" Name="Medium Shading 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="64" Name="Medium Shading 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="65" Name="Medium List 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="66" Name="Medium List 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="67" Name="Medium Grid 1 Accent 6"/>
  <w:LsdException Locked="false" Priority="68" Name="Medium Grid 2 Accent 6"/>
  <w:LsdException Locked="false" Priority="69" Name="Medium Grid 3 Accent 6"/>
  <w:LsdException Locked="false" Priority="70" Name="Dark List Accent 6"/>
  <w:LsdException Locked="false" Priority="71" Name="Colorful Shading Accent 6"/>
  <w:LsdException Locked="false" Priority="72" Name="Colorful List Accent 6"/>
  <w:LsdException Locked="false" Priority="73" Name="Colorful Grid Accent 6"/>
  <w:LsdException Locked="false" Priority="19" QFormat="true"
   Name="Subtle Emphasis"/>
  <w:LsdException Locked="false" Priority="21" QFormat="true"
   Name="Intense Emphasis"/>
  <w:LsdException Locked="false" Priority="31" QFormat="true"
   Name="Subtle Reference"/>
  <w:LsdException Locked="false" Priority="32" QFormat="true"
   Name="Intense Reference"/>
  <w:LsdException Locked="false" Priority="33" QFormat="true" Name="Book Title"/>
  <w:LsdException Locked="false" Priority="37" SemiHidden="true"
   UnhideWhenUsed="true" Name="Bibliography"/>
  <w:LsdException Locked="false" Priority="39" SemiHidden="true"
   UnhideWhenUsed="true" QFormat="true" Name="TOC Heading"/>
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:"Cambria Math";
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-charset:1;
	mso-generic-font-family:roman;
	mso-font-format:other;
	mso-font-pitch:variable;
	mso-font-signature:0 0 0 0 0 0;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:-520081665 -1073717157 41 0 66047 0;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoCommentText, li.MsoCommentText, div.MsoCommentText
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-link:"Comment Text Char";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
span.MsoCommentReference
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-ansi-font-size:8.0pt;
	mso-bidi-font-size:8.0pt;}
a:link, span.MsoHyperlink
	{mso-style-unhide:no;
	color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{mso-style-unhide:no;
	color:purple;
	text-decoration:underline;
	text-underline:single;}
p.MsoCommentSubject, li.MsoCommentSubject, div.MsoCommentSubject
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-parent:"Comment Text";
	mso-style-link:"Comment Subject Char";
	mso-style-next:"Comment Text";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";
	font-weight:bold;}
p.MsoAcetate, li.MsoAcetate, div.MsoAcetate
	{mso-style-noshow:yes;
	mso-style-unhide:no;
	mso-style-link:"Balloon Text Char";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:8.0pt;
	font-family:"Tahoma","sans-serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraph, li.MsoListParagraph, div.MsoListParagraph
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraphCxSpFirst, li.MsoListParagraphCxSpFirst, div.MsoListParagraphCxSpFirst
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraphCxSpMiddle, li.MsoListParagraphCxSpMiddle, div.MsoListParagraphCxSpMiddle
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
p.MsoListParagraphCxSpLast, li.MsoListParagraphCxSpLast, div.MsoListParagraphCxSpLast
	{mso-style-priority:34;
	mso-style-unhide:no;
	mso-style-qformat:yes;
	mso-style-type:export-only;
	margin-top:0in;
	margin-right:0in;
	margin-bottom:0in;
	margin-left:.5in;
	margin-bottom:.0001pt;
	mso-add-space:auto;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman","serif";
	mso-fareast-font-family:"Times New Roman";}
span.CommentTextChar
	{mso-style-name:"Comment Text Char";
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"Comment Text";}
span.CommentSubjectChar
	{mso-style-name:"Comment Subject Char";
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-parent:"Comment Text Char";
	mso-style-link:"Comment Subject";
	font-weight:bold;}
span.BalloonTextChar
	{mso-style-name:"Balloon Text Char";
	mso-style-unhide:no;
	mso-style-locked:yes;
	mso-style-link:"Balloon Text";
	mso-ansi-font-size:8.0pt;
	mso-bidi-font-size:8.0pt;
	font-family:"Tahoma","sans-serif";
	mso-ascii-font-family:Tahoma;
	mso-hansi-font-family:Tahoma;
	mso-bidi-font-family:Tahoma;}
span.SpellE
	{mso-style-name:"";
	mso-spl-e:yes;}
span.GramE
	{mso-style-name:"";
	mso-gram-e:yes;}
.MsoChpDefault
	{mso-style-type:export-only;
	mso-default-props:yes;
	font-size:10.0pt;
	mso-ansi-font-size:10.0pt;
	mso-bidi-font-size:10.0pt;}
@page WordSection1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in 1.0in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.WordSection1
	{page:WordSection1;}
 /* List Definitions */
 @list l0
	{mso-list-id:59913793;
	mso-list-type:hybrid;
	mso-list-template-ids:-1923173740 1549957072 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l0:level1
	{mso-level-start-at:2;
	mso-level-number-format:alpha-lower;
	mso-level-text:"%1\)";
	mso-level-tab-stop:36.75pt;
	mso-level-number-position:left;
	margin-left:36.75pt;
	text-indent:-18.75pt;
	mso-ansi-font-size:14.0pt;
	mso-ansi-font-weight:normal;}
@list l0:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1
	{mso-list-id:180971617;
	mso-list-type:hybrid;
	mso-list-template-ids:-972648950 -1767603584 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l1:level1
	{mso-level-number-format:roman-upper;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.5in;
	mso-ansi-font-size:14.0pt;
	mso-bidi-font-size:14.0pt;}
@list l1:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2
	{mso-list-id:451241682;
	mso-list-type:hybrid;
	mso-list-template-ids:1598447620 -654440768 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l2:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:-;
	mso-level-tab-stop:none;
	mso-level-number-position:left;
	margin-left:.85in;
	text-indent:-.25in;
	font-family:"Arial","sans-serif";
	mso-fareast-font-family:"Times New Roman";}
@list l2:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3
	{mso-list-id:589196660;
	mso-list-type:hybrid;
	mso-list-template-ids:409901062 1606998620 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l3:level1
	{mso-level-number-format:alpha-lower;
	mso-level-text:"%1\)";
	mso-level-tab-stop:37.5pt;
	mso-level-number-position:left;
	margin-left:37.5pt;
	text-indent:-19.5pt;
	mso-ansi-font-size:14.0pt;}
@list l3:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
 table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-priority:99;
	mso-style-qformat:yes;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";}
table.MsoTableGrid
	{mso-style-name:"Table Grid";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-unhide:no;
	border:solid windowtext 1.0pt;
	mso-border-alt:solid windowtext .5pt;
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-border-insideh:.5pt solid windowtext;
	mso-border-insidev:.5pt solid windowtext;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman","serif";}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="8194"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=WordSection1>

<p class=MsoNormal><a name=Top></a><span class=SpellE><span style='font-size:
36.0pt;font-family:"Arial","sans-serif"'>MyriMatch</span></span><span
style='font-size:36.0pt;font-family:"Arial","sans-serif"'> 2.1<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:18.0pt;mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'>For MS-MS Database Search<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:18.0pt;mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:14.0pt;font-family:"Arial","sans-serif"'>For
<span class=SpellE>MyriMatch</span> 1.6 documentation, click <a
href="http://forge.fenchurch.mc.vanderbilt.edu/scm/viewvc.php/*checkout*/trunk/doc/1.6/index.html?revision=95&amp;root=myrimatch">here</a>.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:18.0pt;mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:14.0pt;mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'>Table of Contents:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>I.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><a href="#Introduction">Introduction</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>II.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><a href="#Usage">Usage</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>a.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><a href="#UsageBasic">Basic</a> <o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>b.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><a href="#UsagePTMs">Specifying static or
dynamic mass modifications</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>c.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'><a href="#UsageConfigurationParametersGuide">Configuration
parameters guide</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>III.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><a href="#InterpretingResults">Interpreting
results</a></span><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>a.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'><a
href="#InterpretingResultsSearchOutput">Search-time output</a></span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>b.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'><a
href="#InterpretingResultsSQTFile"><span class=SpellE>PepXML</span> format
guide</a></span><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
style='mso-list:Ignore'>c.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'><a
href="#InterpretingResultsValidation">Validation</a></span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><a name=Introduction><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>Introduction</span></a><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in;text-indent:.25in'><span
class=SpellE><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>MyriMatch</span></span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'> is a tool
designed to take experimental data from shotgun proteomics experiments and
compare those spectra against sequences in a known database of proteins.
Whether the program is being run in a single-computer environment or across an
entire cluster of processing nodes, it is able to optimally divide work in a
much more efficient way than many other database search programs. This is
because it only generates candidate sequences from the known database once for
the entire set of spectra instead of once for every spectrum. Thus, for each
candidate sequence generated, it is compared against every spectrum. The
spectra keep a certain (user-defined) number of candidate sequences that had
the highest scores.<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><a name=Usage></a><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>Usage<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>a)</span><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp; </span><a name=UsageBasic></a>The basic usage of
<span class=SpellE>MyriMatch</span> is quite simple:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:.25in'><span
class=SpellE><span class=GramE><b style='mso-bidi-font-weight:normal'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>myrimatch</span></b></span></span><b
style='mso-bidi-font-weight:normal'><span style='mso-bidi-font-size:36.0pt;
font-family:"Arial","sans-serif"'> [flags] -<span class=SpellE>ProteinDatabase</span>
&lt;FASTA database <span class=SpellE>filepath</span>&gt; &lt;MS/MS data <span
class=SpellE>filepath</span> in a supported file format&gt; &lt;another MS/MS
data <span class=SpellE>filepath</span>&gt;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.25in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>When running
it from the command line, the command line parser first determines what flags
you have specified in the command. The flags can be anywhere on the command
line. The following basic flags are supported:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><b
style='mso-bidi-font-weight:normal'>-<span class=SpellE>cfg</span> &lt;file&gt;</b><span
style='mso-tab-count:6'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>specifies
a runtime configuration [default: myrimatch.cfg]<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><b
style='mso-bidi-font-weight:normal'>-<span class=SpellE>workdir</span>
&lt;path&gt;</b><span style='mso-tab-count:5'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>specifies
a path to use as the working directory during execution [default: current
working directory]<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>-<span
class=SpellE>cpus</span> &lt;integer&gt;<span style='mso-tab-count:5'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span></span></b><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>specifies
the number of worker threads to use during search [default: all available
processors]<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>If a flag is
specified that expects an argument but no argument is provided, it might be
treated as a spectrum data file which probably undesirable. If you do not
specify a runtime configuration file with -<span class=SpellE>cfg</span> and
the default configuration file is not found, then default runtime values are
used (and a warning that no configuration file was found will be shown). <o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>There is
another type of flag that is supported that has a unique pattern: the override
flags. Instead of having a name like <span class=SpellE>cfg</span>, the
override flags have the same name as the variable that they override.
Overriding a variable is specifying a different value on the command line than
the one that is in the configuration file (just like the configuration file
overrides the built-in values). For example, to override the variable <span
class=SpellE>DynamicMods</span> to have the value &#8220;M @ 16&#8221;, use the
override flag:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><span
style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><b
style='mso-bidi-font-weight:normal'>-<span class=SpellE>DynamicMods</span>
&quot;M @ 16&quot;<o:p></o:p></b></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>The double
quotes are necessary on the command line because the value of the variable has
spaces in it.<o:p></o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'>After the
flags are parsed, the file arguments are processed. The first argument is
usually &#8220;-<span class=SpellE>ProteinDatabase</span>&#8221; followed by
the relative or absolute path to the FASTA protein database you want to search
against. Every file argument after that is a relative or absolute path to a
MS/MS spectra data file from which to extract experimental spectra. The FASTA
database <span class=SpellE>filepath</span> must be a valid <span class=SpellE>filepath</span>,
it does not support wildcards. The MS/MS spectra data <span class=SpellE>filepaths</span>,
however, do support wildcards. The provided spectra can be in any of the
formats that <span class=SpellE>ProteoWizard</span> <span class=SpellE>MSData</span>
supports. Click <a href="http://proteowizard.sourceforge.net/formats.shtml">here</a>
for a list.<o:p></o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:36.75pt;text-indent:-18.75pt;mso-list:
l0 level1 lfo4;tab-stops:list 36.75pt'><a name=UsageResidueMasses></a><a
name=UsagePTMs></a><![if !supportLists]><span style='font-size:14.0pt;
mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>b)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>A
static mass modification is something like <span class=SpellE>carboxymethylation</span>
of <span class=SpellE>cysteines</span>, where all <span class=SpellE>cysteines</span>
should be treated as about +57 in <span class=SpellE>MyriMatch</span> and all
subsequent downstream analysis. Refer to the <span class=SpellE>StaticMods</span>
variable in the configuration parameters guide. A dynamic mass modification is
something like <span class=GramE>a potential oxidation</span> of <span
class=SpellE>methionine</span>, where each <span class=SpellE>methionine</span>
may be occur as either its natural mass or about +16. Refer to the <span
class=SpellE>DynamicMods</span> variable in the configuration parameters guide.<o:p></o:p></span></p>

<span style='font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
"Times New Roman";mso-ansi-language:EN-US;mso-fareast-language:EN-US;
mso-bidi-language:AR-SA'><br clear=all style='mso-special-character:line-break;
page-break-before:always'>
</span>

<p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:36.75pt;text-indent:-18.75pt;mso-list:
l0 level1 lfo4;tab-stops:list 36.75pt'><a
name=UsageConfigurationParametersGuide></a><![if !supportLists]><span
style='font-size:14.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>c)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;font-family:"Arial","sans-serif"'>Configuration
parameters guide<o:p></o:p></span></p>

<table class=MsoTableGrid border=1 cellspacing=0 cellpadding=0
 style='border-collapse:collapse;border:none;mso-border-alt:solid windowtext .5pt;
 mso-yfti-tbllook:480;mso-padding-alt:.15in .1in .15in .1in'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;height:30.55pt'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;mso-border-alt:
  solid windowtext .5pt;padding:.15in .1in .15in .1in;height:30.55pt'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:14.0pt;font-family:"Arial","sans-serif"'>Name<o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:14.0pt;font-family:"Arial","sans-serif"'>(Type, Default
  Value)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in;height:30.55pt'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:14.0pt;font-family:"Arial","sans-serif"'>Description<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>NumChargeStates</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 3)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Controls the
  number of charge states that <span class=SpellE>MyriMatch</span> will handle
  during all stages of the program. It is especially important during
  determination of charge state (see <span class=SpellE>DuplicateSpectra</span>
  for more information).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>OutputSuffix</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The output
  of a <span class=SpellE>MyriMatch</span> job will be an identification file
  for each input file. The string specified by this parameter will be appended
  to each output filename. It is useful for differentiating jobs within a
  single directory.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>OutputFormat</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, &#8220;<span class=SpellE>pepXML</span>&#8221;)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span class=SpellE><span style='font-family:"Arial","sans-serif"'>MyriMatch</span></span><span
  style='font-family:"Arial","sans-serif"'> can write identifications in either
  &#8220;<span class=SpellE>mzIdentML</span>&#8221; or &#8220;<span
  class=SpellE>pepXML</span>&#8221; format.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>ProteinDatabase</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Specifies
  the FASTA protein database to be searched.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>DecoyPrefix</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, &#8220;rev_&#8221;)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Specifying
  a decoy prefix enables <span class=SpellE>MyriMatch</span> to know if it is
  making a target or a decoy comparison for each PSM. If the protein database
  has proteins that begin with <span class=SpellE>DecoyPrefix</span>, then
  those proteins are decoys. If not, decoy proteins are created on-the-fly by
  reversing each target protein in the database (so there is one decoy protein
  per target protein). The automatic reversal (as well as the ability to
  distinguish between target and decoy comparisons) can be disabled by setting
  the <span class=SpellE>DecoyPrefix</span> to the empty string
  (&#8220;&#8221;).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>SpectrumListFilters</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, &#8220;<span class=SpellE>peakPicking</span>
  false 2-&#8220;)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A <b
  style='mso-bidi-font-weight:normal'>semicolon-delimited</b> list of filters
  applied to spectra as it is read in. Supported filters are defined by <span
  class=SpellE>ProteoWizard</span>:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <table class=MsoTableGrid border=1 cellspacing=0 cellpadding=0 width=957
   style='width:717.45pt;border-collapse:collapse;border:none;mso-border-alt:
   solid windowtext .5pt;mso-yfti-tbllook:1184;mso-padding-alt:0in 5.4pt 0in 5.4pt'>
   <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal align=center style='text-align:center'><b
    style='mso-bidi-font-weight:normal'><span style='font-family:"Arial","sans-serif"'>Filter
    Name<o:p></o:p></span></b></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border:solid windowtext 1.0pt;
    border-left:none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal align=center style='text-align:center'><b
    style='mso-bidi-font-weight:normal'><span style='font-family:"Arial","sans-serif"'>Definition<o:p></o:p></span></b></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border:solid windowtext 1.0pt;
    border-left:none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal align=center style='text-align:center'><b
    style='mso-bidi-font-weight:normal'><span style='font-family:"Arial","sans-serif"'>Arguments<o:p></o:p></span></b></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:1'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal>index</p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by position in the spectrum list<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE><span style='font-family:"Arial","sans-serif"'>Int_set</span></span><span
    style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:2'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>msLevel</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by MS level<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE><span style='font-family:"Arial","sans-serif"'>Int_set</span></span><span
    style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:3'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>scanNumber</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by scan number or by index+1<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE><span style='font-family:"Arial","sans-serif"'>Int_set</span></span><span
    style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:4'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>scanEvent</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by scan event<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE><span style='font-family:"Arial","sans-serif"'>Int_set</span></span><span
    style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:5'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>scanTime</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by scan start time<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>[<span
    class=SpellE>scanTimeLow,scanTimeHigh</span>]<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:6'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>mzPrecursors</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by precursor m/z<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>[mz1,
    mz2, &#8230; <span class=SpellE>mzN</span>]<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:7'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>defaultArrayLength</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by number of primary data points<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE><span style='font-family:"Arial","sans-serif"'>Int_set</span></span><span
    style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:8'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal>activation</p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by activation type<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;ETD|CID|SA|HCD|BIRD|ECD|<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>IRMPD|PD|PSD|PQD|SID|SORI&gt;<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:9'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal>analyzer</p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by analyzer type<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;<span
    class=SpellE>quad|orbi|FT|IT|TOF</span>&gt;<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:10'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal>polarity</p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectra by scan polarity<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;<span
    class=SpellE>positive|negative</span>|+|-&gt;<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:11'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>peakPicking</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Replaces
    profile peaks with <span class=SpellE>centroided</span> peaks<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;<span
    class=SpellE>prefer_vendor</span>&gt;:<span class=SpellE>boolean</span>(<b
    style='mso-bidi-font-weight:normal'>true</b>) &lt;<span class=SpellE>msLevels</span>&gt;:<span
    class=SpellE>int_set</span><o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:12'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal>threshold</p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectrum data points by intensity<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;<span
    class=SpellE>count|count</span>-after-<span class=SpellE>ties|absolute</span>|<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>bpi-<span
    class=SpellE>relative|tic</span>-<span class=SpellE>relative|tic</span>-cutoff&gt;
    &lt;threshold&gt;<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;most-<span
    class=SpellE>intense|least</span>-intense&gt; [&lt;<span class=SpellE>msLevels</span>&gt;:<span
    class=SpellE>int_set</span>]<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:13'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>mzWindow</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    spectrum data points by m/z<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>[<span
    class=SpellE>mzLow</span>, <span class=SpellE>mzHigh</span>]<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:14'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal>MS2Denoise</p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Applies a
    moving window filter to MS2 spectra<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;window
    peak count&gt;:<span class=SpellE>int</span>(<b style='mso-bidi-font-weight:
    normal'>6</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;window
    m/z width&gt;:<span class=SpellE>int</span>(<b style='mso-bidi-font-weight:
    normal'>30</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;<span
    class=SpellE>multicharge</span> relaxation&gt;:<span class=SpellE>bool</span>(<b
    style='mso-bidi-font-weight:normal'>true</b>)<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:15'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal>MS2Deisotope</p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE><span style='font-family:"Arial","sans-serif"'>Deisotopes</span></span><span
    style='font-family:"Arial","sans-serif"'> MS2 spectra using Markey method<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:16'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>ETDFilter</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters
    ETD <span class=SpellE>MSn</span> spectrum data points, removing <span
    class=SpellE>unreacted</span> precursors, charge-reduced precursors, and
    neutral losses<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;remove
    precursor&gt;:<span class=SpellE>bool</span>(<b style='mso-bidi-font-weight:
    normal'>true</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;remove
    charge-reduced&gt;:<span class=SpellE>bool</span>(<b style='mso-bidi-font-weight:
    normal'>true</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;remove
    neutral losses&gt;:<span class=SpellE>bool</span>(<b style='mso-bidi-font-weight:
    normal'>true</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;blanket
    removal&gt;:<span class=SpellE>bool</span>(<b style='mso-bidi-font-weight:
    normal'>false</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;matching
    tolerance&gt;:real(<b style='mso-bidi-font-weight:normal'>3</b>) &lt;PPM|<b
    style='mso-bidi-font-weight:normal'>MZ</b>&gt;<o:p></o:p></span></p>
    </td>
   </tr>
   <tr style='mso-yfti-irow:17;mso-yfti-lastrow:yes'>
    <td width=145 valign=top style='width:108.85pt;border:solid windowtext 1.0pt;
    border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:
    solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span class=SpellE>chargeStatePredictor</span></p>
    </td>
    <td width=409 valign=top style='width:307.1pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Predicts <span
    class=SpellE>MSn</span> spectrum precursors to be singly or multiply
    charged depending on the ratio of intensity above and below the precursor
    m/z<o:p></o:p></span></p>
    </td>
    <td width=402 valign=top style='width:301.5pt;border-top:none;border-left:
    none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
    mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
    mso-border-alt:solid windowtext .5pt;padding:0in 5.4pt 0in 5.4pt'>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;override
    existing charge&gt;<span class=SpellE>bool</span>(<b style='mso-bidi-font-weight:
    normal'>false</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span lang=FR-CH style='font-family:"Arial","sans-serif";
    mso-ansi-language:FR-CH'>&lt;max. multiple charge&gt;:<span class=SpellE>int</span>(<b
    style='mso-bidi-font-weight:normal'>3</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span lang=FR-CH style='font-family:"Arial","sans-serif";
    mso-ansi-language:FR-CH'>&lt;min. multiple charge&gt;:<span class=SpellE>int</span>(<b
    style='mso-bidi-font-weight:normal'>2</b>)<o:p></o:p></span></p>
    <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&lt;TIC
    fraction threshold&gt;:real(<b style='mso-bidi-font-weight:normal'>0.9</b>)<o:p></o:p></span></p>
    </td>
   </tr>
  </table>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>'<span
  class=SpellE>int_set</span>' means that a set of integers must be specified,
  as a list of intervals of the form [<span class=SpellE>a,b</span>] or a[-][b]<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If no <span
  class=SpellE>chargeStatePredictor</span> is specified, a default one will be
  added like: &#8220;<span class=SpellE>chargeStatePredictor</span> false &lt;<span
  class=SpellE>NumChargeStates</span>&gt; 2 0.9&#8221;<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>TicCutoffPercentage</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0.98)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>In order to
  maximize the effectiveness of the MVH scoring algorithm, an important step in
  preprocessing the experimental spectra is filtering out noise peaks. Noise peaks
  are filtered out by sorting the original peaks in descending order of
  intensity, and then picking peaks from that list until the cumulative ion
  current of the picked peaks divided by the total ion current (TIC) is greater
  than or equal to this parameter. Lower percentages mean that less of the
  spectrums total intensity will be allowed to pass through preprocessing. See
  the section on Advanced Usage for tips on how to use this parameter
  optimally.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MaxPeakCount</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Filters out
  all peaks except the <span class=SpellE>MaxPeakCount</span> most intense
  peaks.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:9'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>AvgPrecursorMzTolerance</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 1.5 m/z)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A generated
  sequence candidate is only compared to an experimental spectrum if the
  candidates mass is within this tolerance of the experimental spectrum&#8217;s
  precursor mass. The units (&#8220;<span class=SpellE>daltons</span>&#8221; or
  &#8220;<span class=SpellE>ppm</span>&#8221;) must be provided as well as the
  magnitude. The actual tolerance used for the search is calculated by
  multiplying the tolerance by the charge state, so this parameter should be
  set to the tolerance that is desired for +1 <span class=GramE>spectra</span>.
  At the default value, the precursor mass tolerances are 1.5, 3, and 4.5 <span
  class=SpellE>Da</span> for the first three charge states, respectively.<span
  style='mso-spacerun:yes'>&nbsp; </span><o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:10'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MonoPrecursorMzTolerance</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 10 <span class=SpellE>ppm</span>)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A generated
  sequence candidate is only compared to an experimental spectrum if the candidates
  mass is within this tolerance of the experimental spectrum&#8217;s precursor
  mass. The units (&#8220;<span class=SpellE>daltons</span>&#8221; or &#8220;<span
  class=SpellE>ppm</span>&#8221;) must be provided as well as the magnitude.
  The actual tolerance used for the search is calculated by multiplying the
  tolerance by the charge state, so this parameter should be set to the
  tolerance that is desired for +1 <span class=GramE>spectra</span>. At the
  default value, the precursor mass tolerances are 10, 20, and 30 <span
  class=SpellE>ppm</span> for the first three charge states, respectively.<span
  style='mso-spacerun:yes'>&nbsp; </span><o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:11'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MonoisotopeAdjustmentSet</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer set, [-1,2])<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Sometimes a
  mass spectrometer will pick the wrong isotope as the <span class=SpellE>monoisotope</span>
  of an eluting peptide. When using narrow tolerances for <span class=SpellE>monoisotopic</span>
  precursors, this can cause identifiable spectra to be missed. This parameter
  defines a set of isotopes (0 being the instrument-called <span class=SpellE>monoisotope</span>)
  to try as the <span class=SpellE>monoisotopic</span> precursor m/z. To
  disable this technique, set the value to &#8220;0&#8221;. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:12'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>PrecursorMzToleranceRule</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, &#8220;auto&#8221;)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter controls the automatic selection of precursor mass type. For data
  from Thermo instruments, using the &#8220;auto&#8221; setting on a RAW, <span
  class=SpellE>mzML</span>, or mz5 file will automatically choose <span
  class=SpellE>monoisotopic</span> or average mass values (and the
  corresponding precursor tolerance). For other instruments or older data
  formats, the &#8220;mono&#8221; or &#8220;<span class=SpellE>avg</span>&#8221;
  tolerance should be set explicitly.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:13'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>FragmentMzTolerance</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0.5 m/z)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter controls how much tolerance there is on each side of the calculated
  m/z when looking for an ion fragment peak during candidate scoring. The units
  (&#8220;<span class=SpellE>daltons</span>&#8221; or &#8220;<span
  class=SpellE>ppm</span>&#8221;) must be provided as well as the magnitude.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:14'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>FragmentationRule</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, &#8220;CID&#8221;)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter determines which ion series are used to build the theoretical
  spectrum for each candidate peptide. Possible values are:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>CID: b, y<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>ETD: c, z*<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>manual:
  user-defined (a comma-separated list of [<span class=SpellE>abcxyz</span>] or
  z* (z+1), e.g. <span class=SpellE>manual:b,y,z</span><o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:15'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>FragmentationAutoRule</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(<span class=SpellE>boolean</span>,
  true)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If true, <span
  class=SpellE>MyriMatch</span> will automatically choose the fragmentation
  rule based on the activation type of each <span class=SpellE>MSn</span>
  spectrum. This allows a single search to handle CID and ETD spectra (i.e. an
  interleaved or decision tree run). If false or if the input format does not
  specify the input format then <span class=SpellE>FragmentationRule</span> is
  used (see above).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:16'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>StaticMods</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If a
  residue (or multiple residues) should always be treated as having a
  modification on their natural mass, set this parameter to inform the search
  engine which residues are modified. Residues are entered into this string as
  a space-delimited list of pairs. Each pair is of the form:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>&lt;AA residue character&gt; &lt;mod
  mass&gt;<o:p></o:p></span></b></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Thus, to
  treat <span class=SpellE>cysteine</span> as always being <span class=SpellE>carboxymethylated</span>,
  this parameter would be set to something like the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;C
  57&#8221;<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:17'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>DynamicMods</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, none)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>Note: avoid using the
  &#8220;#&#8221; symbol in a configuration file since it begins a comment
  section. Using the &#8220;#&#8221; symbol in a command-line override works
  fine.<o:p></o:p></span></b></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>In order to
  search a database for potential post-translational modifications of candidate
  sequences, the user must configure this parameter to inform the search engine
  which residues may be modified. Residues that are modifiable are entered into
  this string in a space-delimited list of triplets. Each triplet is of the
  form:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>&lt;AA motif&gt; &lt;character to
  represent mod&gt; &lt;mod mass&gt;<o:p></o:p></span></b></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></b></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Thus, to
  search for potentially oxidized <span class=SpellE>methionines</span> and <span
  class=SpellE>phosphorylated</span> <span class=SpellE>serines</span>, this
  parameter would be set to something like the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;M *
  15.995 S $ 79.966&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The AA
  motif can include multiple residues, and the peptide termini are represented
  by opening &#8220;(<span class=GramE>&#8220; and</span> closing
  &#8220;)&#8221; parentheses for the N and C termini, respectively. For
  example, an N terminal <span class=SpellE>deamidation</span> of glutamine may
  be specified by the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;(Q ^
  -17&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If the last
  residue in the motif is not the residue intended to be modifiable, then use
  an exclamation mark to indicate that the residue preceding the mark is the
  modifiable residue. Using the previous example, &#8220;(Q! ^ -17&#8221; is an
  equivalent way to specify it. Another example would be specifying the <span
  class=SpellE>demidation</span> of <span class=SpellE>asparagine</span> when
  it is N terminal to a <span class=SpellE>glycine</span>, which might look
  like:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;N!G
  @ -17&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Another
  possibility is to specify a block of interchangeable residues in the motif,
  which is supported by the &#8220;[<span class=GramE>&#8220; and</span>
  &#8220;]&#8221; brackets. For example, to specify a potential <span
  class=SpellE>phosphorylation</span> on any serine, <span class=SpellE>threonine</span>,
  or tyrosine, use the string:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;[STY]
  * 79.966&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The
  &#8220;{<span class=GramE>&#8220; and</span> &#8220;}&#8221; brackets work in
  the opposite way as the &#8220;[&#8220; and &#8220;]&#8221; brackets, i.e.
  &#8220;{STY} * 79.966&#8221; specifies a potential <span class=SpellE>phosphorylation</span>
  on every residue EXCEPT serine, <span class=SpellE>threonine</span>, or
  tyrosine. Both kinds of brackets can be combined with the exclamation mark,
  in which case the exclamation mark should come after the block (because the
  block counts as a single residue). Using the previous example, &#8220;[STY]!
  * 79.966&#8221; is an equivalent way to specify it.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Using the
  negative multi-residue brackets is the best way to indicate the &#8220;any
  residue except&#8221; concept, and it works on single residues as well. For
  example, to specify a mod on lysine except when it is at the C terminus of a
  peptide, use something like the string &#8220;K<span class=GramE>!{</span>)}
  # 144&#8221;. Another example is specifying the cleavage-blocking <span
  class=SpellE>homoserine</span> mod in a <span class=SpellE>CnBr</span> digest
  when a serine or <span class=SpellE>threonine</span> is C terminal to a <span
  class=SpellE>methionine</span>:<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>&#8220;M![ST]
  * -29.99&#8221;<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Note that
  it is not currently possible to specify (for example) the
  non-cleavage-blocking <span class=SpellE>homoserine</span> <span
  class=SpellE>lactone</span> mod in a <span class=SpellE>CnBr</span> digest,
  because the motif would extend outside of the peptide sequence itself. In the
  future a string like &#8220;M!){ST} * -17&#8221; might work for that, but for
  now, if &#8220;(<span class=GramE>&#8220; is</span> used it must be the first
  character in the motif, and likewise if &#8220;)&#8221; is used it must be
  the last character in a motif. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:18'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MaxDynamicMods</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 2)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter sets the maximum number of modified residues that may be in any
  candidate sequence.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:19'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MaxResultRank</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 5)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter sets the maximum rank of peptide-spectrum-matches to report for
  each spectrum. A rank is all PSMs that score the same (common for isobaric
  residues and ambiguous modification localization). <span class=SpellE>MyriMatch</span>
  may report extra ranks in order to ensure that the top target match and top
  decoy match from each digestion specificity (full, semi, <span class=GramE>non</span>)
  is reported. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:20'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>CleavageRules</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(string, &#8220;Trypsin/P&#8221;)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  important parameter allows the user to control the way peptides are generated
  from the protein database. It can be used to configure the search on <span
  class=SpellE>tryptic</span> peptides only, on non-<span class=SpellE>tryptics</span>,
  or anything in between. It can even be used to test multiple residue motifs
  at a potential cleavage site. This parameter describes which amino acids are
  valid on the N and C termini of a digestion site. The parameter is specified
  in <a
  href="http://www.ebi.ac.uk/ontology-lookup/browse.do?ontName=MS&amp;termId=MS:1001180&amp;termName=cleavage%20agent%20regular%20%25expression">PSI-MS
  regular expression syntax</a> (a limited Perl regular expression syntax). <span
  class=SpellE>MyriMatch</span> can recognize the following protease names and
  automatically use the corresponding regular expression for this parameter.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-spacerun:yes'>&nbsp;</span><o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-spacerun:yes'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span><b style='mso-bidi-font-weight:normal'>Protease names:<o:p></o:p></b></span></p>
  <p class=MsoListParagraphCxSpFirst style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Trypsin/P&#8221;
  (allows for cut after K or R)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Trypsin&#8221;
  (normal trypsin cut, disallows cutting when the site is before a <span
  class=SpellE>proline</span>)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&quot;<span
  class=SpellE>Chymotrypsin</span>&#8221; (allows cut after F,Y,W,L. Disallows
  cutting before <span class=SpellE>proline</span>)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&quot;<span
  class=SpellE>TrypChymo</span>&#8221; (combines &#8220;Trypsin/P&#8221; and
  &#8220;<span class=SpellE>Chymotrypsin</span>&#8221; cleavage rules)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Lys-C&#8221;<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Lys-C/P&#8221;
  (Lys-C, disallowing cutting before <span class=SpellE>proline</span>)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;Asp-N&#8221;<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;<span
  class=SpellE>PepsinA</span>&#8221; (Cuts right after F, L)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;<span
  class=SpellE>CNBr</span>&#8221; (<span class=SpellE>Cyanogen</span> bromide)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;<span
  class=SpellE>Formic_acid</span>&#8221; (Formic acid)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpMiddle style='margin-left:.85in;mso-add-space:
  auto;text-indent:-.25in;mso-list:l2 level1 lfo6'><![if !supportLists]><span
  style='font-family:"Arial","sans-serif";mso-fareast-font-family:Arial'><span
  style='mso-list:Ignore'>-<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
  </span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>&#8220;<span
  class=SpellE>NoEnzyme</span>&#8221; (not supported; use the proper enzyme and
  set <span class=SpellE>MinTerminiCleavages</span> to 0)<o:p></o:p></span></p>
  <p class=MsoListParagraphCxSpLast style='margin-left:.85in;mso-add-space:
  auto'><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>A complete
  list of supported protease names can be found <a
  href="http://www.ebi.ac.uk/ontology-lookup/browse.do?ontName=MS&amp;termId=MS:1001045&amp;termName=cleavage%20agent%20name">here</a>.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>Note:</span></b><span
  style='font-family:"Arial","sans-serif"'> <span class=SpellE>CleavageRules</span>
  can also work with <span class=GramE>an earlier</span> but deprecated regular
  expression syntax. <b style='mso-bidi-font-weight:normal'>We highly
  discourage users from using the old syntax.</b> Briefly, the old syntax is a
  space-delimited list of cleavage rules, where each cleavage rule itself is a
  space-delimited pair of strings. The first string of the cleavage rule
  specifies the residue or residues that must be N-terminal to a potential
  cleavage site. The second string specifies the residue or residues that must
  be C-terminal to the site. Either string in the pair can contain multiple
  sequences of one or more residues, separated by the &#8216;|&#8217;
  character. A &#8216;.&#8217; character is a wildcard that will accept
  anything. Additionally, the &#8216;[<span class=GramE>&#8216; and</span>
  &#8216;]&#8217; characters refer to the N and C termini of a protein.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Now that
  you are thoroughly confused, here are examples of single cleavage rules:<o:p></o:p></span></p>
  <p class=MsoNormal><span class=GramE><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>R .</span></b></span><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if the N-terminal residue is R<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>R|<span class=GramE>K .</span></span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if the N-terminal residue is R <i
  style='mso-bidi-font-style:normal'>or</i> K<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>[ .</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage at the N terminus of a protein<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>. ]</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage at the C terminus of a protein<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>The <b
  style='mso-bidi-font-weight:normal'>&#8220;.&#8221;</b> wildcard is important
  because it allows the cleavage routine to work very quickly. However, if you
  wanted to leave out <span class=SpellE>proline</span> from a <span
  class=SpellE>tryptic</span> digest, you would have to explicitly declare the
  valid residues for both sides of a cleavage site:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>R|K
  A|C|D|E|F|G|H|I|K|L|M|N|Q|R|S|T|V|W|Y</span></b><span style='font-family:
  "Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Remember
  that this parameter is a <i style='mso-bidi-font-style:normal'>list</i> of
  cleavage rules; a real <span class=SpellE>tryptic</span> digest can be
  declared with two cleavage rules:<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:"Arial","sans-serif"'>[|R|K
  . <span class=GramE>. ]</span></span></b><span style='font-family:"Arial","sans-serif"'><span
  style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if it is at the N terminus of a protein,<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>or
  if the N-terminal residue is R or K; a site is valid for cleavage<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:2'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>if
  it is at the C terminus of a protein<o:p></o:p></span></p>
  <p class=MsoNormal style='margin-left:91.8pt;text-indent:-91.8pt'><span
  style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Also note
  that a cleavage rule can have a residue string of more than one residue,
  allowing for multiple-residue cleavage motifs:<o:p></o:p></span></p>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:"Arial","sans-serif"'>[M<span class=GramE>|[</span> .</span></b><span
  style='font-family:"Arial","sans-serif"'><span style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>a
  site is valid for cleavage if it is at the N terminus of a protein,<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>or
  if the N-terminal sequence of residues is [M (i.e. the M must<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><span
  style='mso-tab-count:3'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>be
  at the N terminus of a protein)<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:21'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MinTerminiCleavages</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 2)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>By default,
  when generating peptides from the protein database, a peptide must start and
  end at a valid cleavage site. Setting this parameter to 0 or 1 will reduce
  that requirement, so that <span class=GramE>neither terminus or</span> only
  one terminus of the peptide must match one of the cleavage rules specified in
  the <span class=SpellE>CleavageRules</span> parameter. This parameter is
  useful to turn a <span class=SpellE>tryptic</span> digest into a semi-<span
  class=SpellE>tryptic</span> digest.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:22'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MaxMissedCleavages</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, -1)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>By default,
  when generating peptides from the protein database, a peptide may contain any
  number of missed cleavages. A missed cleavage is a site within the peptide
  that matches one of the cleavage rules (refer to <span class=SpellE>CleavageRules</span>).
  Settings this parameter to some other number will stop generating peptides
  from a sequence if it contains more than the specified number of missed
  cleavages.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:23'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MinPeptideMass</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 0 <span class=SpellE>Da</span>)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  preprocessing the experimental spectra, any spectrum with a precursor mass
  that is less than the specified mass will be disqualified. This parameter is
  useful to eliminate inherently unidentifiable spectra from an input data set.
  A setting of 500 for example, will eliminate most 3-residue matches and clean
  up the output file quite a lot.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:24'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MaxPeptideMass</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 10000 <span class=SpellE>Da</span>)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  preprocessing the experimental spectra, any spectrum with a precursor mass
  that exceeds the specified mass will be disqualified.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:25'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MinPeptideLength</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 5)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  digesting proteins, any peptide which does not meet or exceed the specified
  length will be disqualified.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:26'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>MaxPeptideLength</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 75)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  digesting proteins, any peptide which exceeds this specified length will be
  disqualified.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:27'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>StatusUpdateFrequency</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 5 seconds)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Preprocessing
  spectra and scoring candidates may take a long time. A measure of progress
  through the protein database will be given on intervals that are specified by
  this parameter, measured in seconds.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:28'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>UseSmartPlusThreeModel</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(<span class=SpellE>boolean</span>, true)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Once a
  candidate sequence has been generated from the protein database, <span
  class=SpellE>MyriMatch</span> determines which spectra will be compared to
  the sequence. For each unique charge state of those spectra, a set of
  theoretical fragment ions is generated by one of several different
  algorithms.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>For +1 and
  +2 precursors, a +1 b and y ion is always predicted at each peptide bond.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>For +3 and
  higher precursors, the fragment ions predicted depend on the way this
  parameter is set. When this parameter is true, then for each peptide bond, an
  internal calculation is done to estimate the <span class=SpellE>basicity</span>
  of the b and y fragment sequence. The <span class=GramE>precursors</span>
  protons are distributed to those ions based on that calculation, with the
  more basic sequence generally getting more of the protons. For example, when
  this parameter is true, each peptide bond of a +3 precursor will either
  generate a +2 b<sub>i</sub> and a +1 <span class=SpellE>y<sub>i</sub></span>
  ion, or a +1 b<sub>i</sub> and a +2 <span class=SpellE>y<sub>i</sub></span>
  ion. For a +4 precursor, depending on <span class=SpellE>basicity</span>, a
  peptide bond breakage may result in a +1 b<sub>i</sub> and a +3 <span
  class=SpellE>y<sub>i</sub></span> ion, a +2 b<sub>i</sub> and a +2 <span
  class=SpellE>y<sub>i</sub></span> ion, or a +3 b<sub>i</sub> and a +1 <span
  class=SpellE>y<sub>i</sub></span> ion. When this parameter is false, however,
  ALL possible charge distributions for the fragment ions are generated for
  every peptide bond. So a +3 sequence of length 10 will always have
  theoretical +1 y5, +2 y5, +1 b5, and +2 b5 ions.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:29'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>ProteinSamplingTime</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 15, seconds)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Before
  beginning sequence candidate generation and scoring, <span class=SpellE>MyriMatch</span>
  will do a random sampling of the protein database to get an estimate of the
  number of comparisons that will be done by the job. The longer it is allowed
  to sample the database the more accurate it will be. Of course, if there are
  fewer proteins in the database than the sample size, all proteins will be
  used in the sampling and the number of comparisons will be exact.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:30'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>ComputeXCorr</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(<span class=SpellE>boolean</span>,
  true)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If true, a <span
  class=SpellE>Sequest</span>-like cross correlation (<span class=SpellE>xcorr</span>)
  score will be calculated for the top ranking hits in each spectrum&#8217;s
  result set.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:31'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>NumIntensityClasses</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 3)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>Before
  scoring any candidates, experimental spectra have their peaks stratified into
  the number of intensity classes specified by this parameter. Spectra that are
  very dense in peaks will likely benefit from more intensity classes in order
  to best take advantage of the variation in peak intensities. Spectra that are
  very sparse will not see much benefit from using many intensity classes.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:32;height:12.1pt'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in;height:12.1pt'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>ClassSizeMultiplier</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(real, 2)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in;
  height:12.1pt'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>When
  stratifying peaks into a specified, fixed number of intensity classes, this
  parameter controls the size of each class relative to the class above it
  (where the peaks are more intense). At default values, if the best class, A,
  has 1 peak in it, then class B will have 2 peaks in it and class C will have
  4 peaks.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:33'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>NumBatches</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(integer, 50)<o:p></o:p></span></p>
  </td>
  <td width=977 valign=top style='width:732.65pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>This
  parameter sets a number of batches per node to strive for when using the
  MPI-based parallelization features. Setting this too low means that some
  nodes will finish before others (idle processor time), while setting it too
  high means more overhead in network transmission as each batch is smaller. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:34;mso-yfti-lastrow:yes'>
  <td width=245 style='width:183.55pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span class=SpellE><span
  style='font-family:"Arial","sans-serif"'>UseMultipleProcessors</span></span><span
  style='font-family:"Arial","sans-serif"'><o:p></o:p></span></p>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-family:"Arial","sans-serif"'>(<span class=SpellE>boolean</span>,
  true)<o:p></o:p></span></p>
  </td>
  <td width=977 style='width:732.65pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-family:"Arial","sans-serif"'>If true,
  each process will use all the processing units available on the system it is
  running on.<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal><span style='font-size:14.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:14.0pt;font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
font-family:"Arial","sans-serif"'>Int<a name=InterpretingResults></a>erpreting
results<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:37.5pt;text-indent:-19.5pt;mso-list:l3 level1 lfo8;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-size:14.0pt;
mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>a)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>S<a
name=InterpretingResultsSearchOutput></a>earch-time output of <span
class=SpellE>MyriMatch</span> serves several purposes. The majority of the
output will usually be progress information, telling the user which part of the
job that <span class=SpellE>MyriMatch</span> is currently working on, and in
some cases how far along into that part the job is. There will be periodic
updates when <span class=SpellE>MyriMatch</span> is preprocessing spectra and
when it is generating candidates from the database and comparing those
candidates against the spectra. In a multi-process (MPI) job, there will also
be progress information on bulk transfers of data over the network.
Additionally, <span class=SpellE>MyriMatch</span> will display statistics on
the spectra that remain after preprocessing, specifically the average number of
peaks in a spectrum before and after preprocessing. Also provided is the
average number of the percentage of peaks that were filtered out by the
preprocessing step. Finally, in the case of an MPI job, when the database
search is complete each node that took part in the search will display
statistics detailing the work that node did. The lines will be like:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='font-family:"Arial","sans-serif"'>Process #1 (<span
class=SpellE>foohost</span>) stats: &lt;<span class=SpellE>numBatches</span>&gt;
/ &lt;<span class=SpellE>numProteins</span>&gt; / &lt;<span class=SpellE>numCandidatesGenerated</span>&gt;
/ &lt;<span class=SpellE>numCandidatesSearched</span>&gt; / &lt;<span
class=SpellE>numComparisonsDone</span>&gt;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:37.5pt;text-indent:-19.5pt;mso-list:l3 level1 lfo8;
tab-stops:list 37.5pt'><![if !supportLists]><span style='font-size:14.0pt;
mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>b)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-family:"Arial","sans-serif"'>O<a
name=InterpretingResultsSQTFile></a>ne <span class=SpellE>pepXML</span> (a
database search output format originally developed at the SPC) file is produced
for every input spectra file that a <span class=SpellE>MyriMatch</span> job
searches. The file contains an entry for each spectrum kept during the search
(i.e. only the spectra that were not obviously <span class=GramE>junk</span>)
that was compared to any candidate. Only the best <span class=SpellE>MaxResults</span>
(a <span class=SpellE>config</span> parameter, defaulting to 5) matches for
each spectrum are kept and output in the file. The <span class=SpellE>pepXML</span>
format is quite versatile, but does have a few limitations. It is not possible
to fully encode <span class=SpellE>MyriMatch&#8217;s</span> <span class=SpellE>CleavageRules</span>
variable if multi-residue motifs are used (in the rule &#8220;[<span
class=GramE>|[</span>M|K|R .&#8221; the &#8220;[M&#8221; part is a
multi-residue motif), and it is not possible to fully encode Static/<span
class=SpellE>DynamicMods</span> if multi-residue motifs are used (in the mod
&#8220;(Q ^ -17&#8221; the &#8220;(Q&#8221; part is a multi-residue motif). We
have proposed extensions to <span class=SpellE>pepXML</span> to support these
motif-oriented capabilities.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:37.5pt;text-indent:-19.5pt;mso-list:l3 level1 lfo8;
tab-stops:list 37.5pt'><a name=InterpretingResultsValidation></a><![if !supportLists]><span
style='font-size:14.0pt;mso-bidi-font-size:12.0pt;font-family:"Arial","sans-serif";
mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>c)<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp; </span></span></span><![endif]><span
style='font-family:"Arial","sans-serif"'>To validate results generated by <span
class=SpellE>MyriMatch</span> (and also several other popular search engines),
users can pass <span class=SpellE>pepXML</span> files to <span class=SpellE>IDPickerQonvert</span>
(in preparation for analyzing the results in the <span class=SpellE>IDPicker</span>
suite). This approach only works when the protein database that was searched
included distracter (decoy) proteins with a common prefix in their name that is
not a prefix in any of the valid proteins. Most often, the database that was
searched will include proteins in their forward and reverse orders, with the
reversed protein having a prefix added to the name like &#8220;rev_&#8221;.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-family:"Arial","sans-serif"'><o:p>&nbsp;</o:p></span></p>

</div>

</body>

</html>
