<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:st1="urn:schemas-microsoft-com:office:smarttags"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta http-equiv=Content-Type content="text/html; charset=utf-8">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 11">
<meta name=Originator content="Microsoft Word 11">
<link rel=File-List href="index_files/filelist.xml">
<title>IDPicker</title>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="City"/>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="place"/>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="PlaceType"/>
<o:SmartTagType namespaceuri="urn:schemas-microsoft-com:office:smarttags"
 name="PlaceName"/>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Matt Chambers</o:Author>
  <o:LastAuthor>Matt Chambers</o:LastAuthor>
  <o:Revision>6</o:Revision>
  <o:TotalTime>3467</o:TotalTime>
  <o:Created>2007-07-24T19:07:00Z</o:Created>
  <o:LastSaved>2007-10-01T15:42:00Z</o:LastSaved>
  <o:Pages>1</o:Pages>
  <o:Words>2880</o:Words>
  <o:Characters>16416</o:Characters>
  <o:Company>MSRC</o:Company>
  <o:Lines>136</o:Lines>
  <o:Paragraphs>38</o:Paragraphs>
  <o:CharactersWithSpaces>19258</o:CharactersWithSpaces>
  <o:Version>11.8132</o:Version>
 </o:DocumentProperties>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:PunctuationKerning/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:Compatibility>
   <w:BreakWrappedTables/>
   <w:SnapToGridInCell/>
   <w:WrapTextWithPunct/>
   <w:UseAsianBreakRules/>
   <w:DontGrowAutofit/>
  </w:Compatibility>
  <w:BrowserLevel>MicrosoftInternetExplorer4</w:BrowserLevel>
 </w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" LatentStyleCount="156">
 </w:LatentStyles>
</xml><![endif]--><!--[if !mso]><object
 classid="clsid:38481807-CA0E-42D2-BF39-B33AF135CC4D" id=ieooui></object>
<style>
st1\:*{behavior:url(#ieooui) }
</style>
<![endif]-->
<style>
<!--
 /* Font Definitions */
 @font-face
	{font-family:Wingdings;
	panose-1:5 0 0 0 0 0 0 0 0 0;
	mso-font-charset:2;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:0 268435456 0 0 -2147483648 0;}
@font-face
	{font-family:Tahoma;
	panose-1:2 11 6 4 3 5 4 4 2 4;
	mso-font-charset:0;
	mso-generic-font-family:swiss;
	mso-font-pitch:variable;
	mso-font-signature:1627421319 -2147483648 8 0 66047 0;}
 /* Style Definitions */
 p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";}
p.MsoCommentText, li.MsoCommentText, div.MsoCommentText
	{mso-style-noshow:yes;
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";}
span.MsoCommentReference
	{mso-style-noshow:yes;
	mso-ansi-font-size:8.0pt;
	mso-bidi-font-size:8.0pt;}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{color:purple;
	text-decoration:underline;
	text-underline:single;}
p.MsoCommentSubject, li.MsoCommentSubject, div.MsoCommentSubject
	{mso-style-noshow:yes;
	mso-style-parent:"Comment Text";
	mso-style-next:"Comment Text";
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-fareast-font-family:"Times New Roman";
	font-weight:bold;}
p.MsoAcetate, li.MsoAcetate, div.MsoAcetate
	{mso-style-noshow:yes;
	margin:0in;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:8.0pt;
	font-family:Tahoma;
	mso-fareast-font-family:"Times New Roman";}
@page Section1
	{size:8.5in 11.0in;
	margin:1.0in 1.25in 1.0in 1.25in;
	mso-header-margin:.5in;
	mso-footer-margin:.5in;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
 /* List Definitions */
 @list l0
	{mso-list-id:59913793;
	mso-list-type:hybrid;
	mso-list-template-ids:-1923173740 1549957072 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l0:level1
	{mso-level-start-at:2;
	mso-level-number-format:alpha-lower;
	mso-level-text:"%1\)";
	mso-level-tab-stop:36.75pt;
	mso-level-number-position:left;
	margin-left:36.75pt;
	text-indent:-18.75pt;
	mso-ansi-font-size:14.0pt;
	mso-ansi-font-weight:normal;}
@list l0:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l0:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1
	{mso-list-id:180971617;
	mso-list-type:hybrid;
	mso-list-template-ids:-972648950 -1767603584 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l1:level1
	{mso-level-number-format:roman-upper;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.5in;
	mso-ansi-font-size:14.0pt;
	mso-bidi-font-size:14.0pt;}
@list l1:level2
	{mso-level-number-format:alpha-lower;
	mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l1:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2
	{mso-list-id:309867048;
	mso-list-type:hybrid;
	mso-list-template-ids:-1361028172 -647586132 67698713 67698715 67698703 67698713 67698715 67698703 67698713 67698715;}
@list l2:level1
	{mso-level-number-format:alpha-lower;
	mso-level-text:"%1\)";
	mso-level-tab-stop:.5in;
	mso-level-number-position:left;
	text-indent:-.25in;
	mso-ansi-font-size:14.0pt;
	mso-ansi-font-weight:normal;}
@list l2:level2
	{mso-level-tab-stop:1.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l2:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-p sition:left;
	text-indent:-.25in;}
@list l3
	{mso-list-id:408774144;
	mso-list-type:hybrid;
	mso-list-template-ids:-1672313078 -324887614 -1466167636 484594088 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l3:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:>;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.25in;
	font-family:Arial;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
@list l3:level2
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:|;
	mso-level-tab-stop:1.25in;
	mso-level-number-position:left;
	margin-left:1.25in;
	text-indent:-.25in;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";}
@list l3:level3
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:Ø;
	mso-level-tab-stop:1.75in;
	mso-level-number-position:left;
	margin-left:1.75in;
	text-indent:-.25in;
	font-family:Wingdings;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:Arial;}
@list l3:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l3:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4
	{mso-list-id:657267439;
	mso-list-type:hybrid;
	mso-list-template-ids:268219800 -324887614 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l4:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:>;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.25in;
	font-family:Arial;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
@list l4:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:1.25in;
	mso-level-number-position:left;
	margin-left:1.25in;
	text-indent:-.25in;
	font-family:"Courier New";}
@list l4:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l4:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5
	{mso-list-id:692878021;
	mso-list-type:hybrid;
	mso-list-template-ids:1167754248 -324887614 -1466167636 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l5:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:>;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.25in;
	font-family:Arial;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
@list l5:level2
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:|;
	mso-level-tab-stop:1.25in;
	mso-level-number-position:left;
	margin-left:1.25in;
	text-indent:-.25in;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";}
@list l5:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l5:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6
	{mso-list-id:1566255503;
	mso-list-type:hybrid;
	mso-list-template-ids:579107944 -324887614 -1466167636 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l6:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:>;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.25in;
	font-family:Arial;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
@list l6:level2
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:|;
	mso-level-tab-stop:1.25in;
	mso-level-number-position:left;
	margin-left:1.25in;
	text-indent:-.25in;
	font-family:Arial;
	mso-bidi-font-family:"Times New Roman";}
@list l6:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l6:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7
	{mso-list-id:1914049347;
	mso-list-type:hybrid;
	mso-list-template-ids:1372123818 -324887614 67698691 67698693 67698689 67698691 67698693 67698689 67698691 67698693;}
@list l7:level1
	{mso-level-start-at:0;
	mso-level-number-format:bullet;
	mso-level-text:>;
	mso-level-tab-stop:.75in;
	mso-level-number-position:left;
	margin-left:.75in;
	text-indent:-.25in;
	font-family:Arial;
	mso-fareast-font-family:"Times New Roman";
	mso-bidi-font-family:"Times New Roman";}
@list l7:level2
	{mso-level-number-format:bullet;
	mso-level-text:o;
	mso-level-tab-stop:1.25in;
	mso-level-number-position:left;
	margin-left:1.25in;
	text-indent:-.25in;
	font-family:"Courier New";}
@list l7:level3
	{mso-level-tab-stop:1.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level4
	{mso-level-tab-stop:2.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level5
	{mso-level-tab-stop:2.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level6
	{mso-level-tab-stop:3.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level7
	{mso-level-tab-stop:3.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level8
	{mso-level-tab-stop:4.0in;
	mso-level-number-position:left;
	text-indent:-.25in;}
@list l7:level9
	{mso-level-tab-stop:4.5in;
	mso-level-number-position:left;
	text-indent:-.25in;}
ol
	{margin-bottom:0in;}
ul
	{margin-bottom:0in;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
 table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-parent:"";
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ansi-language:#0400;
	mso-fareast-language:#0400;
	mso-bidi-language:#0400;}
table.MsoTableGrid
	{mso-style-name:"Table Grid";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
 	border:solid windowtext 1.0pt;
	mso-border-alt:solid windowtext .5pt;
	mso-padding-alt:0in 5.4pt 0in 5.4pt;
	mso-border-insideh:.5pt solid windowtext;
	mso-border-insidev:.5pt solid windowtext;
	mso-para-margin:0in;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ansi-language:#0400;
	mso-fareast-language:#0400;
	mso-bidi-language:#0400;}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="13314"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body lang=EN-US link=blue vlink=purple style='tab-interval:.5in'>

<div class=Section1>

<p class=MsoNormal><span style='font-size:36.0pt;font-family:Arial'>IDPicker
v1.5</span><span style='font-size:14.0pt;font-family:Arial'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:18.0pt;mso-bidi-font-size:36.0pt;
font-family:Arial'>Accurate, transparent, and parsimonious protein assembly for
MS/MS search results<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-size:18.0pt;mso-bidi-font-size:36.0pt;
font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-size:14.0pt;mso-bidi-font-size:36.0pt;
font-family:Arial'>Table of Contents:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>I.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:Arial'><a href="#Overview">Overview</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>II.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;mso-bidi-font-size:
36.0pt;font-family:Arial'><a href="#Usage">Usage</a><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:
Ignore'>a.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:Arial'><a href="#UsageConvert">IDPickerQonvert</a> (converting
scores from raw peptide identifications into Q values, a.k.a. FDR)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:
Ignore'>b.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:Arial'><a href="#UsageAssemble">IDPickerAssemble</a> (assembling
and grouping peptide identifications into a single file)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:1.0in;text-indent:-.25in;mso-list:l1 level2 lfo2;
tab-stops:list 1.0in'><![if !supportLists]><span style='mso-bidi-font-size:
36.0pt;font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:
Ignore'>c.<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='mso-bidi-font-size:36.0pt;
font-family:Arial'><a href="#UsageRepo t">IDPickerReport</a> (applying
parsimony and HTML reporting to assembled peptide identifications)<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.5in;mso-list:l1 level1 lfo2;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:14.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>III.<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><span style='font-size:14.0pt;font-family:Arial'><a
href="#Glossary">Glossary</a><o:p></o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:Arial'>Over<a name=Overview></a>view<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-family:Arial'>A
comprehensive introduction to IDPicker and its resulting output is available in
<a href="overview-1-2.pdf">this whitepaper</a>.</span><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><a name=Usage></a><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:Arial'>Usage<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l2 level1 lfo4;
tab-stops:list .5in'><a name=UsageConvert></a><![if !supportLists]><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:Arial;mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>a)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='mso-bidi-font-size:36.0pt;font-family:Arial'>Importing raw peptide
identifications into the IDPicker pipeline<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='mso-bidi-font-size:36.0pt;font-family:Arial'>IDPicker currently supports
reading raw peptide identifications via the <st1:place w:st="on"><st1:PlaceName
 w:st="on">Seattle</st1:PlaceName> <st1:PlaceName w:st="on">Proteome</st1:PlaceName>
 <st1:PlaceType w:st="on">Center</st1:PlaceType></st1:place>’s pepXML
format.<span style='mso-spacerun:yes'>  </span>Many search engines have output
that can be converted to this format using various SPC tools.<span
style='mso-spacerun:yes'>  </span>But the pepXML format is not convenient to
store the protein assembly data that is generated and used in the IDPicker
pipeline.<span style='mso-spacerun:yes'>  </span>Instead, raw peptide identifications
are explicitly converted to their corresponding false discovery rates (FDR)
based on validation via a target-decoy search strategy.<span
style='mso-spacerun:yes'>  </span>During the conversion process, the FASTA
protein database (which must be the same between all files given as input to
Qonvert) which was used for the search is read in order to combine the proteins
that map with confidently identified peptides to their respective descriptions
from the FASTA database.<span style='mso-spacerun:yes'>  </span>The database is
also used to find an offset into the protein for each peptide identification.<span
style='mso-spacerun:yes'>  </span>If a peptide maps to more than one protein in
the database, the offset is calculated using the first protein only.<span
style='mso-spacerun:yes'>  </span>Typically, some amount of FDR filtering is
done at this step to increase processing efficiency and reduce file size, but
it is not required.<span style='mso-spacerun:yes'>  </span>Basic usage of
IDPickerQonvert is easy, with most of the settings defaulting to typical
values:<o:p></o:p></span></p>

<table class=MsoTableGrid border=0 cellspacing=0 cellpadding=0
 style='margin-left:.5in;border-collapse:collapse;mso-yfti-tbllook:480;
 mso-padding-alt:0in 5.4pt 0in 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=204 valign=top style='width:153.0pt;padding:0in 5.4pt 0in 5.4pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='mso-bidi-font-size:36.0pt;font-family:Arial'>idpQonvert<o:p></o:p></span></b></p>
  </td>
  <td width=987 valign=top style='width:740.25pt;padding:0in 5.4pt 0in 5.4pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-size:10.0pt;font-family:Arial'>[&lt;pepXML filemask&gt;
  [&lt;another pepXML filemask&gt;] ...] [-b &lt;filepath to list of input
  pepXML filemasks, one per line&gt;] [-dump] [-cfg &lt;config filepath&gt;]<o:p></o:p></span></b></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'>Input pepXML files can be specified as filemasks
(i.e. supporting wildcards) on the command line, or they can be read from a
batch file which lists each input filemask on a separate line.<span
style='mso-spacerun:yes'>  </span>The default filtering and validation
parameters can be overridden by using the “-Name &lt;value&gt;” notation on the
command line, e.g. “-MaxAbsoluteFDR 0.05”; these overrides can occur anywhere
on the command line.<span style='mso-spacerun:yes'>  </span>The parameters can
also be set via a configuration file: idpQonvert will look for idpQonvert.cfg
in the working directory and use the settings there if it exists, or you can
pass a filepath to a custom configuration file with the -cfg flag.<span
style='mso-spacerun:yes'>  </span>The runtime parameters specific to
IDPickerQonvert are as follows:<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<table>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;mso-border-alt:
  solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Name<o:p></o:p></span></b></p>
  </td>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Type
  (Default)<o:p></o:p></span></b></p>
  </td>
  <td width=924 style='width:693.0pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Description<o:p></o:p></span></b></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>MaxResultRank<o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>integer (1)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum rank that a search result can have to be passed
  through to the “qonverted” file.<span style='mso-spacerun:yes'> 
  </span>Things get really messy when one starts accepting more than the
  highest ranked hit, so the default value here is usually best.<span
  style='mso-spacerun:yes'>  </span>Note: when peptide candidates get equal
  scores, they all belong to the same rank, so it is possible to have more than
  one peptide in a rank 1 result.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>MaxAbsoluteFDR</span><span
  style='font-family:Arial'><o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>real (0.25)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum FDR that the “xcorr”-type score of each peptide
  identification must meet (e.g. “0.25” for 75% confidence or 25% FDR).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'><a name="test">MaxRelativeFDR</a></span><span
  style='font-family:Arial'><o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>real (1.0)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum confidence that the “deltcn”-type score of each
  peptide identification must meet (e.g. “0.25” for 75% confidence or 25%
  FDR).<span style='mso-spacerun:yes'>  </span>Note: the two FDR filters have
  an AND relationship - a peptide must pass both filters to be accepted.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td width=132 valign=top style='width:99.0pt;border:solid windowtext 1.0pt;
  border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='margin-right:3.1pt;text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'><a name='ProteinDatabase'>ProteinDatabase</a><o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>string (automatic)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>By
  default, IDPickerQonverter will try to use the database found in the input
  pepXML files, but if none of those can be found, this parameter will need to
  be set to a valid filepath to the same database which was used in the input
  data.<span style='mso-spacerun:yes'>  </span>If this parameter is set, it
  overrides the databases stored in the pepXML files.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td width=132 valign=top style='width:99.0pt;border:solid windowtext 1.0pt;
  border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='margin-right:3.1pt;text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>DecoyRatio<o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>real (1.0)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>If the
  number of target and decoy proteins in the database is not equal, this
  parameter must be set to describe the ratio of target proteins to decoy
  proteins in the database (equal to # of target proteins divided by # of decoy
  proteins).<span style='mso-spacerun:yes'>  </span>This is not done
  automatically for reasons of efficiency.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6;mso-yfti-lastrow:yes'>
  <td width=132 valign=top style='width:99.0pt;border:solid windowtext 1.0pt;
  border-top:none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='margin-right:3.1pt;text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'><a name='DecoyPrefix'>DecoyPrefix</a><o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>string (“rev_”)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>IDPickerQonvert
  determines if a protein accession ID is a decoy or not by a certain unique
  prefix.<span style='mso-spacerun:yes'>  </span>This parameter sets that
  prefix.<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'>Example usage (with files source1.pepXML,
source2.pepXML, source3.pepXML, all searched against database
“c:\proteindb-with-reverse.fasta”):<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.25in;mso-list:l7 level1 lfo6;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>&gt;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>idpQonvert source1.pepXML
source2.pepXML source3.pepXML<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l6 level2 lfo8;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading protein database from
filepath: &quot;c:\proteindb-with-reverse.fasta&quot;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l6 level2 lfo8;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading pepXML file from filepath:
&quot;source1. pepXML&quot;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l6 level2 lfo8;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Writing validated identifications to
filepath: &quot;source1.xml&quot;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l6 level2 lfo8;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading pepXML file from filepath:
&quot;source2. pepXML&quot;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l6 level2 lfo8;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Writing validated identifications to
filepath: &quot;source2.xml&quot;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l6 level2 lfo8;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading pepXML file from filepath:
&quot;source3. pepXML&quot;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l6 level2 lfo8;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Writing validated identifications to
filepath: &quot;source3.xml&quot;<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:36.75pt;text-indent:-18.75pt;mso-list:
l0 level1 lfo10;tab-stops:list 36.75pt'><a name=UsageAssemble></a><![if !supportLists]><span
style='font-size:14.0pt;mso-bidi-font-size:12.0pt;font-family:Arial;mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>b)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-family:Arial'><a name='GroupsExplanation'>Assembling and grouping identifications into an
experimental hierarchy</a></span></b><span style='font-family:Arial'><o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='font-family:Arial'>IDPicker supports protein assembly and analysis with
arbitrarily complex hierarchies.<span style='mso-spacerun:yes'>  </span>This
important information is not lost when analyzing the data.<span
style='mso-spacerun:yes'>  </span>A few relatively simple use cases could be an
IDPicker analysis on a single LC-MS run, a group of LC-MS runs (like a MudPIT),
or a group of groups of LC-MS runs (a group of MudPITs).<span
style='mso-spacerun:yes'>  </span>A more complex use case could be a
multi-level hierarchy across multiple instrument types, search engines,
replicate types, etc.<span style='mso-spacerun:yes'>  </span>This information
is injected into the analysis during this step of the pipeline (while at the
same time consolidating identification data from multiple sources into a single
file).<span style='mso-spacerun:yes'>  </span>The final confidence filtering
level is usually used at this stage, but that need not always be the case.<span
style='mso-spacerun:yes'>  </span></span><span style='mso-bidi-font-size:36.0pt;
font-family:Arial'>Basic usage of IDPickerAssemble is easy, with most of the
settings defaulting to typical values:</span><span style='font-family:Arial'><o:p></o:p></span></p>

<table class=MsoTableGrid border=0 cellspacing=0 cellpadding=0
 style='margin-left:.5in;border-collapse:collapse;mso-yfti-tbllook:480;
 mso-padding-alt:0in 5.4pt 0in 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=204 valign=top style='width:153.0pt;padding:0in 5.4pt 0in 5.4pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:Arial'>idpAssemble<o:p></o:p></span></b></p>
  </td>
  <td width=987 valign=top style='width:740.25pt;padding:0in 5.4pt 0in 5.4pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-size:10.0pt;font-family:Arial'>&lt;output filepath&gt;
  [&lt;source group name&gt; &lt;source filemask&gt; [&lt;another group
  name&gt; &lt;another</span></b> <b style='mso-bidi-font-weight:normal'><span
  style='font-size:10.0pt;font-family:Arial'>source filemask&gt;] ...] [-b
  &lt;filepath to list of group/filemask pairs, one per line&gt;] [-dump] [-cfg
  &lt;config filepath&gt;]<o:p></o:p></span></b></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Arial'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'>The first required argument is the filepath to which
the assembled identifications will be written, e.g.
“c:\assembled-ids.xml”.<span style='mso-spacerun:yes'>  </span>The other
required parameters are given as group/filemask pairs, where the filemask represents
one or more qonverted files and the group describes where those files belong in
the experimental hierarchy.<span style='mso-spacerun:yes'>  </span>Groups are
described by a POSIX-like path, with “/” being the root group and any other
group path, e.g. “/some/sub/groups”, creates subgroups within the hierarchy
(similar to a computer filesystem).<span style='mso-spacerun:yes'>  </span>The
group names can contain spaces, but the path will need to be quoted for it to
be parsed correctly.<span style='mso-spacerun:yes'>  </span>Like in IDPickerQonvert,
the default filtering and validation parameters can be overridden by using the
“-Name &lt;value&gt;” notation on the command line, e.g. “-MaxAbsoluteFDR
0.05”; these overrides can occur anywhere on the command line.<span
style='mso-spacerun:yes'>  </span>The parameters can also be set via a
configuration file: idpAssemble will look for idpAssemble.cfg in the working
directory and use the settings there if it exists, or you can pass a filepath
to a custom configuration file with the -cfg flag.<span
style='mso-spacerun:yes'>  </span>The runtime parameters specific to
IDPickerAssemble are as follows:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<table>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;mso-border-alt:
  solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Name<o:p></o:p></span></b></p>
  </td>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Type
  (Default)<o:p></o:p></span></b></p>
  </td>
  <td width=924 style='width:693.0pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Description<o:p></o:p></span></b></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>MaxResultRank<o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>integer (1)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum rank that a search result can have to be passed
  through to the assembled file.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>MaxAbsoluteFDR</span><span
  style='font-family:Arial'><o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>real (0.05)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum FDR that the “xcorr”-type score of each peptide
  identification must meet (e.g. “0.25” for 75% confidence or 25% FDR).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3;mso-yfti-lastrow:yes'>
  <td width=132 style='width:99.0pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>MaxRelativeFDR</span><span
  style='font-family:Arial'><o:p></o:p></span></p>
  </td>
  <td width=132 style='width:99.0pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>real (1.0)<o:p></o:p></span></p>
  </td>
  <td width=924 valign=top style='width:693.0pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum confidence that the “deltcn”-type score of each peptide
  identification must meet (e.g. “0.25” for 75% confidence or 25% FDR).<span
  style='mso-spacerun:yes'>  </span>Note: the two FDR filters have an AND
  relationship - a peptide must pass both filters to be accepted.<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal><span style='font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Arial'><span style='mso-tab-count:
1'>            </span>Example usage (with files source1.xml, source2.xml,
source3.xml):<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.25in;mso-list:l5 level1 lfo12;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>&gt;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>idpAssemble combined-sources.xml /1
source1.xml /2 source2.xml /3 source3.xml -MaxAbsoluteFDR 0.25<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Initializing workspace... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
source1.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
source2.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
source3.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l5 level2 lfo12;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Writing assembled peptides XML to
filepath: combined-sources.xml<o:p></o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Arial'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><span style='font-family:Arial'>Another
example (with a more complicated hierarchy with multiple sources from multiple
instrument types):<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.25in;mso-list:l4 level1 lfo14;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>&gt;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>idpAssemble combined-sources.xml
“/low accuracy/ltq” ltq*.xml “/low accuracy/lcq” lcq*.xml “/high accuracy/orbi”
orbi*.xml -cfg multi_instrument.cfg<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Initializing workspace... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
ltq_source1.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
ltq_source2.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
lcq_source1.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
lcq_source2.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
orbi_source1.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Reading peptides from filepath: .\
orbi_source2.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Parsing XML... done.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.25in;text-indent:-.25in;mso-list:l3 level2 lfo16;
tab-stops:list 1.25in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>|<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>Writing assembled peptides XML to
filepath: combined-sources.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Arial'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Arial'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:-.25in;mso-list:l0 level1 lfo10;
tab-stops:list 36.75pt'><a name=UsageReport></a><![if !supportLists]><span
style='font-size:14.0pt;mso-bidi-font-size:12.0pt;font-family:Arial;mso-fareast-font-family:
Arial'><span style='mso-list:Ignore'>c)<span style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-family:Arial'>Generating an IDPicker HTML report<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.5in;text-indent:.25in'><span
style='font-family:Arial'>The final stage of the IDPicker analysis pipeline
applies the final filters to a file assembled in the previous stage and creates
an HTML report in a new directory (the name of the output directory is one of
the command line parameters).<span style='mso-spacerun:yes'>  </span></span><span
style='mso-bidi-font-size:36.0pt;font-family:Arial'>Basic usage of IDPickerReport
is easy, with most of the settings defaulting to typical values:</span><b
style='mso-bidi-font-weight:normal'><span style='font-family:Arial'><o:p></o:p></span></b></p>

<table class=MsoTableGrid border=0 cellspacing=0 cellpadding=0
 style='margin-left:.5in;border-collapse:collapse;mso-yfti-tbllook:480;
 mso-padding-alt:0in 5.4pt 0in 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td width=204 valign=top style='width:153.0pt;padding:0in 5.4pt 0in 5.4pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-family:Arial'>idpReport<o:p></o:p></span></b></p>
  </td>
  <td width=987 valign=top style='width:740.25pt;padding:0in 5.4pt 0in 5.4pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-size:10.0pt;font-family:Arial'>&lt;output name&gt; &lt;input
  peptides XML filemask&gt; [&lt;another filemask&gt; ...] [-dump] [-cfg
  &lt;config filepath&gt;]<o:p></o:p></span></b></p>
  </td>
 </tr>
</table>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.5in'><span style='mso-bidi-font-size:
36.0pt;font-family:Arial'>The first required argument is a path to a directory (which
may or may not exist yet) to which the files for the HTML report will be
written, e.g. “reports/TestReport2000”.<span style='mso-spacerun:yes'> 
</span>The other required parameters are simply filemasks representing one or
more qonverted IDPicker files.<span style='mso-spacerun:yes'>  </span>Like in
IDPickerQonvert, the default filtering and validation parameters can be
overridden by using the “-Name &lt;value&gt;” notation on the command line,
e.g. “-MaxAbsoluteFDR 0.05”; these overrides can occur anywhere on the command
line.<span style='mso-spacerun:yes'>  </span>The parameters can also be set via
a configuration file: idpReport will look for idpReport.cfg in the working
directory and use the settings there if it exists, or you can pass a filepath
to a custom configuration file with the -cfg flag.<span
style='mso-spacerun:yes'>  </span>The runtime parameters specific to
IDPickerReport are as follows:<o:p></o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<table>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;mso-border-alt:
  solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Name<o:p></o:p></span></b></p>
  </td>
  <td width=125 style='width:93.9pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Type
  (Default)<o:p></o:p></span></b></p>
  </td>
  <td width=886 style='width:664.2pt;border:solid windowtext 1.0pt;border-left:
  none;mso-border-left-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><b
  style='mso-bidi-font-weight:normal'><span style='font-family:Arial'>Description<o:p></o:p></span></b></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>MaxResultRank<o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>integer (1)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum rank that a search result can have to be passed
  through to the assembled file.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'><a name='MaxAbsoluteFDR'>MaxAbsoluteFDR</a></span><span
  style='font-family:Arial'><o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>real (0.05)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This parameter
  sets the maximum FDR that the “xcorr”-type score of each peptide
  identification must meet (e.g. “0.25” for 75% confidence or 25% FDR).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>MaxRelativeFDR<o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>real (1.0)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the maximum confidence that the “deltcn”-type score of each
  peptide identification must meet (e.g. “0.25” for 75% confidence or 25%
  FDR).<span style='mso-spacerun:yes'>  </span>Note: the two FDR filters have
  an AND relationship - a peptide must pass both filters to be accepted.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'><a name='MinDistinctPeptides'>MinDistinctPeptides</a><o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>integer (2)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A distinct
  peptide is a subsequence of a protein’s amino acid sequence, which has been
  identified (and passed filters) at least once based on the given input, but
  may be identified more than once or thousands of times.<span
  style='mso-spacerun:yes'>  </span>A thousand identifications of the same
  subsequence still counts as one distinct peptide, and this parameter sets the
  minimum number of distinct peptides that a protein must be linked to in order
  to be included in the final report.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'><a name='MaxAmbiguousIds'>MaxAmbiguousIds</a><o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>integer (2)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>An ambiguous
  identification comes from a result that has multiple peptides with equal
  scores, i.e. belonging to the same rank.<span style='mso-spacerun:yes'> 
  </span>Such identifications can severely distort the efficacy of IDPicker’s
  connected component (clustering) analysis, so this filter can be set to
  reject them outright to prevent it.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'><a name='MostParsimoniousAnalysis'>MostParsimoniousAnalysis</a><o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>Boolean (true)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter controls whether the final protein list is filtered by the minimum
  covering set, i.e. the minimum set of proteins that is necessary to describe
  all identified peptides.<span style='mso-spacerun:yes'>  </span>This kind of
  filter is known as “parsimony” or “Occam’s Razor.”<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>GenerateBipartiteGraphs<o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>Boolean (false)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A
  sometimes useful feature of IDPickerReport is the ability to generate GraphViz
  representations of the connected components (clusters) in the workspace.<span
  style='mso-spacerun:yes'>  </span>Enabling this feature requires that
  GraphViz’s “dot” program be accessible via the default system path.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>RawSourceHostURL<o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>string (“http://localhost/”)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>When
  IDPicker reports are generated with the intent to be viewed from a web
  server, this setting describes the host URL to use for the server-side
  interactive content in the report (e.g. the on-the-fly spectrum viewer via
  the generateSpectrumSvg CGI program).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:9'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>RawSourceExtension<o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>string (“.xml”)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the extension that a spectrum source file is expected to be
  (generateSpectrumSvg requires the mzData format).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:10;mso-yfti-lastrow:yes'>
  <td width=177 style='width:132.9pt;border:solid windowtext 1.0pt;border-top:
  none;mso-border-top-alt:solid windowtext .5pt;mso-border-alt:solid windowtext .5pt;
  padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>RawSourcePath<o:p></o:p></span></p>
  </td>
  <td width=125 style='width:93.9pt;border-top:none;border-left:none;
  border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal align=center style='text-align:center'><span
  style='font-size:10.0pt;font-family:Arial'>string (“”)<o:p></o:p></span></p>
  </td>
  <td width=886 valign=top style='width:664.2pt;border-top:none;border-left:
  none;border-bottom:solid windowtext 1.0pt;border-right:solid windowtext 1.0pt;
  mso-border-top-alt:solid windowtext .5pt;mso-border-left-alt:solid windowtext .5pt;
  mso-border-alt:solid windowtext .5pt;padding:.15in .1in .15in .1in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>This
  parameter sets the path to the spectrum source files on the web server.<span
  style='mso-spacerun:yes'>  </span>If it is left at the default empty string,
  no server-side interactive content is generated in the HTML report.<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><span
style='mso-tab-count:1'>            </span>Example usage (with file
combined-sources.xml):<o:p></o:p></span></p>

<p class=MsoNormal style='margin-left:.75in;text-indent:-.25in;mso-list:l3 level1 lfo16;
tab-stops:list .75in'><![if !supportLists]><span style='font-size:10.0pt;
font-family:Arial;mso-fareast-font-family:Arial'><span style='mso-list:Ignore'>&gt;<span
style='font:7.0pt "Times New Roman"'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</span></span></span><![endif]><b style='mso-bidi-font-weight:normal'><span
style='font-size:10.0pt;font-family:Arial'>idpReport combined-analysis
combined-sources.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span><st1:place w:st="on"><st1:City w:st="on">Reading</st1:City></st1:place>
and parsing peptides XML from filepath: .\combined-sources.xml<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>| <span
style='mso-tab-count:1'>          </span><span
style='mso-spacerun:yes'>       </span>...parsed 20368873 of 20368873 bytes<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished reading and parsing peptides
XML; 15.4061514 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Filtering out results with more than
2 ambiguous ids...<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span><span
style='mso-spacerun:yes'>        </span>...removing result 84 of 84<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished filtering by maximum
ambiguous ids; 0.1874988 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Filtering out proteins with less than
2 distinct peptides...<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span><span
style='mso-spacerun:yes'>        </span>...testing protein 8115 of 8115<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished filtering by minimum
distinct peptides; 6.3437094 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Assembling protein groups...<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span><span
style='mso-spacerun:yes'>        </span>...adding protein 8115 of 8115<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished assembling protein groups;
1.3906161 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Assembling peptide groups...<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span><span
style='mso-spacerun:yes'>        </span>...adding result 14217 of 14217<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished assembling peptide groups;
0.4843719 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Assembling clusters...<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span><span
style='mso-spacerun:yes'>        </span>...linking peptide group 2 of 2 to
protein group 4648 of 4648<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished assembling clusters;
0.6093711 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Assembling minimum covering set for
clusters...<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Assembling minimum covering set for
cluster 2548 of 2548 (1 protein groups, 2 results)<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished assembling minimum covering
sets; 0.2656233 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Filtering workspace by minimum
covering set...<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span><span
style='mso-spacerun:yes'>        </span>...linking peptide group 2 of 2 to
protein group 2916 of 2916<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Finished filtering by minimum
covering set; 1.8437382 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Assembling source groups... finished;
0 seconds elapsed.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Creating output directory for cluster
summaries: combined-analysis<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Writing overall summary to filepath:
combined-analysis-summary.html<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Writing data processing details to
filepath: combined-analysis-processing.html<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Writing source groups summary to
filepath: combined-analysis-groups.html<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Writing sequences per protein table
to filepath: combined-analysis-sequences-per-protein-by-group.html<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Writing spectra per protein table to
filepath: combined-analysis-spectra-per-protein-by-group.html<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Writing spectra per peptide table to
filepath: combined-analysis-spectra-per-peptide-by-group.html<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:1.0in'><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>|<span
style='mso-tab-count:1'>           </span>Writing summary for cluster 2554 of
2554 to filepath: combined-analysis-cluster2554.html<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='margin-left:.25in'><a name=Glossary></a><span
style='font-size:14.0pt;mso-bidi-font-size:36.0pt;font-family:Arial'>Glossary<b
style='mso-bidi-font-weight:normal'> </b></span><b style='mso-bidi-font-weight:
normal'><span style='font-size:10.0pt;font-family:Arial'>(a.k.a. what do all
these ambiguous terms actually mean?!)<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.25in;text-indent:.25in'><b
style='mso-bidi-font-weight:normal'><span style='font-size:10.0pt;font-family:
Arial'>Click on a term to look it up in the Wikipedia.<o:p></o:p></span></b></p>

<p class=MsoNormal style='margin-left:.25in'><span style='font-size:14.0pt;
mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

<table class=MsoTableGrid border=0 cellspacing=0 cellpadding=0 width=1191
 style='width:893.25pt;margin-left:.5in;border-collapse:collapse;mso-yfti-tbllook:
 480;mso-padding-alt:0in 5.4pt 0in 5.4pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>/<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>The “root”
  source group.<span style='mso-spacerun:yes'>  </span>This group always
  contains the analysis results for all other source groups combined.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a href="http://en.wikipedia.org/wiki/Bipartite_graph">Bipartite
  Graph</a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A visual way
  of representing the relationships between protein groups and peptide groups.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>CID<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A unique
  cluster identification number for a given analysis.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a
  href="http://en.wikipedia.org/wiki/Connected_component_%28graph_theory%29">Cluster</a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A set of
  interrelated protein groups and peptide groups.<span
  style='mso-spacerun:yes'>  </span>A cluster cannot be interrelated with any
  other cluster.<span style='mso-spacerun:yes'>  </span>In graph theory, this
  concept is called a “connected component.”<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4;height:.5in'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:.5in'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a href="http://en.wikipedia.org/wiki/False_discovery_rate">False
  Discovery Rate (FDR)</a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:.5in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A
  statistical estimate of the probability that a given result is a false
  one.<span style='mso-spacerun:yes'>  </span>The absolute FDR is based on the
  absolute scores that the search engines supply (e.g. XCorr, MVH, Hyperscore).<span
  style='mso-spacerun:yes'>  </span>The relative FDR is based on the
  corresponding relative scores (e.g. DeltCn, Expectation Value).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5;height:21.15pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:21.15pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>GID<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:21.15pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A unique
  global identification number for either a protein group or a peptide group.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>GLID<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A numeric
  or alphabetic symbol used as a cluster-local identifier for a protein group
  or peptide group.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>Identification
  (id)<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A match
  between a spectrum and a peptide with assigned false discovery rates.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a href="http://en.wikipedia.org/wiki/Set_cover_problem">Minimum
  Covering Set (MCS)</a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>The
  smallest set of protein groups necessary to explain the existence of all
  peptide groups in a cluster.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:9;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a href="http://en.wikipedia.org/wiki/peptide">Peptide </a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A short
  chain of amino acids that can be matched with a spectrum to form an
  identification.<span style='mso-spacerun:yes'>  </span>This usage of
  “peptide” implies <i style='mso-bidi-font-style:normal'>distinctness</i>, i.e.
  that it may have been identified to many different spectra in the input data,
  but it only counts as one peptide.<span style='mso-spacerun:yes'>  </span>A <i
  style='mso-bidi-font-style:normal'>unique</i> peptide, on the other hand, is
  one that belongs to only one protein in the FASTA database.<o:p></o:p></span></p>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:10;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>Peptide
  Group (metapeptide)<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A group
  of results that share the same set of proteins.<span
  style='mso-spacerun:yes'>  </span>These groups are formed to make the
  analysis more presentable.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:11;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a href="http://en.wikipedia.org/wiki/protein">Protein</a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A unique accession
  string that identifies a large chain of amino acids with biological meaning
  in a FASTA database.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:12;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>Protein
  Group (metaprotein)<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A group
  of proteins that share the same set of results.<span
  style='mso-spacerun:yes'>  </span>They are indiscernible from each other
  based on available evidence.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:13;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>Rank<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A number
  assigned to a result to establish its relative ordering to other results for
  a given spectrum.<span style='mso-spacerun:yes'>  </span>Lower ranks are
  better.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:14;height:.5in'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:.5in'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>Result<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:.5in'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>One or
  more identifications that matched to a spectrum with the same score and thus produced
  the same FDRs.<span style='mso-spacerun:yes'>  </span>When a result has more
  than one identification, all of them are presented together to emphasize the
  ambiguity of the result.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:15;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a href="http://en.wikipedia.org/wiki/peptide">Sequence</a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A
  sequence simply refers to a peptide (see Peptide).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:16;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>SID<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A
  sequence identification number for distinguishing the different members of a
  given protein or peptide group.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:17;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b><span style='font-size:10.0pt;font-family:Arial'>Source
  Group<o:p></o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A
  canonical and hierarchical assignment of one or more input files to cause them
  to be analyzed separately (as well as with other groups).<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:18;height:32.85pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:32.85pt'>
  <p class=MsoNormal><u><span style='font-size:10.0pt;font-family:Arial;
  color:blue'><a href="http://en.wikipedia.org/wiki/Mass_spectrum">Spectrum
  (Spectra)</a><o:p></o:p></span></u></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:32.85pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'>A
  centroided list of mass-to-charge peaks from a tandem mass spectrum with an
  assigned charge state.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:19;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-size:10.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:20;mso-yfti-lastrow:yes;height:24.0pt'>
  <td width=216 valign=top style='width:2.25in;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
  style='font-size:10.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></b></p>
  </td>
  <td width=975 valign=top style='width:731.25pt;padding:0in 5.4pt 0in 5.4pt;
  height:24.0pt'>
  <p class=MsoNormal><span style='font-size:10.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>
  </td>
 </tr>
</table>

<p class=MsoNormal><span style='mso-bidi-font-size:36.0pt;font-family:Arial'><o:p>&nbsp;</o:p></span></p>

</div>

</body>

</html>
