<html>

<head>
    <link rel="stylesheet" type="text/css" href="..\..\SkylineStyles.css">
</head>

<body>
    <h1 class="document-title">Skyline DDA Search for MS1 Filtering</h1>
    <p>The Skyline Targeted Mass Spectrometry Environment provides informative visual displays of the raw mass spectrometer data you import into your
        Skyline documents. These displays allow you to manipulate the data by performing tasks such as refining the peptides and transitions you are
        measuring, and tuning integration boundaries. Originally developed to analyze quantitative assays from selected reaction monitoring (SRM –
        also referred to as multiple reaction monitoring or MRM) mass spectrometry, Skyline has been expanded to extract time-intensity chromatograms
        from MS1 spectra for use in peptide quantification experiments involving mass spec runs with data dependent MS/MS.</p>
    <p>The Skyline MS1 Full-Scan Filtering supports importing data sets from discovery-type proteomics experiments where mass spectrometers were
        operated in data dependent acquisition (DDA) mode. After importing the raw data, new and previously existing Skyline features facilitate
        quantifying peptide precursor MS1 signal across many replicate acquisitions. This mode may also be used for visualizing and better
        understanding quantitative outputs from other “Label Free” quantitative tools, because of the exceptional data visualization plots in Skyline.
    </p>
    <p>This tutorial will cover the following areas critical to making effective use of Skyline MS1 Filtering when you want Skyline to perform peptide
        spectrum matching for the MS/MS spectra in your DDA data:</p>
    <ul>
        <li>Setting up a Skyline document for MS1 filtering</li>
        <li>Performing a DDA search on raw data to find potential quantitative targets</li>
    </ul>
    <p>Skyline aims to provide a vendor-neutral platform for targeted mass spectrometry investigation. It can import raw data for MS1 filtering from
        the instrument vendors Agilent, Bruker, SCIEX, Shimadzu, Thermo-Scientific and Waters, making the expertise you gain here transferrable to any
        mass spectrometry lab with an instrument from one of these vendors.</p>
    <h1>Getting Started</h1>
    <p>To start this tutorial, download the following ZIP file:</p>
    <p><a href="https://skyline.ms/tutorials/DdaSearchMs1Filtering.zip">https://skyline.ms/tutorials/DdaSearchMs1Filtering.zip</a></p>
    <p>Extract the files in it to a folder on your computer, like:</p>
    <p>C:\Users\brendanx\Documents</p>
    <p>This will create a new folder:</p>
    <p>C:\Users\brendanx\Documents\DdaSearchMS1Filtering </p>
    <p>If you have been using Skyline prior to starting this tutorial, it is a good idea to revert Skyline to its default settings. To do so: </p>
    <ul>
        <li>Start Skyline.</li>
    </ul>
    <ul class="keep-next">
        <li>On the <strong>Start Page, </strong>click <strong>Blank Document </strong>which looks like this: </li>
    </ul>
    <p><img src="image-0.png" /></p>
    <ul>
        <li>On the <strong>Settings</strong> menu, click <strong>Default. </strong></li>
        <li>Click <strong>No </strong>on the form asking if you want to save the current settings.</li>
    </ul>
    <p>The document settings in this instance of Skyline have now been reset to the default.</p>
    <p>Since this tutorial covers a proteomics topic, you can choose the proteomics interface by doing the following:</p>
    <ul>
        <li>Click the user interface control in the upper right-hand corner of the Skyline window, and click <strong>Proteomics interface
            </strong>which looks like this: </li>
    </ul>
    <p><img src="image-1.png" /></p>
    <p>Skyline is operating in proteomics mode which is displayed by the protein icon <img src="image-2.png" /> in the upper right-hand corner of the
        Skyline window.</p>
    <p>You could start editing this blank document in a number of ways, but for this tutorial you will use a sequential set of forms called a wizard
        that will walk you through the steps of searching mass spectrometer data dependent acquisition (DDA) data files, setting up targets, and
        importing chromatograms from those files.</p>
    <p>Before starting the DDA search, you need to change the internal standard Skyline uses by default:</p>
    <ul>
        <li>On the <strong>Settings</strong> menu, click <strong>Peptide Settings</strong>.</li>
        <li>Click on the <strong>Modifications</strong> tab.</li>
        <li>Set the <strong>Internal standard type</strong> to “<strong>none</strong>”</li>
    </ul>
    <p class="keep-next">The Peptide Settings form should now look like:</p>
    <p><img src="image-3.png" /></p>
    <ul>
        <li>Click the <strong>OK</strong> button in the <strong>Peptide Settings</strong> form.</li>
    </ul>
    <h1>Searching DDA Files to Load Peptides into a Skyline Document</h1>
    <p>You can use the <strong>Run Peptide Search</strong> wizard to run a peptide search on the MS/MS spectra in DDA data files.</p>
    <p>First, do the following to save your new document:</p>
    <ul>
        <li>Click the <strong>Save</strong> button on the toolbar (Ctrl-S).</li>
        <li>Navigate to the DdaSearchMS1Filtering folder you created for this tutorial.</li>
        <li>In the <strong>File name</strong> field, enter ‘DdaSearchMS1FilteringTutorial.sky’.</li>
        <li>Click the <strong>Save</strong> button.</li>
    </ul>
    <p>Now, initiate the <strong>Run Peptide Search</strong> wizard as follows:</p>
    <ul>
        <li>On the <strong>File</strong> menu, choose <strong>Search</strong> and click <strong>Run Peptide Search</strong>.</li>
    </ul>
    <p class="keep-next">Skyline should present a form that looks like this:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-4.png" /></p>
    <p>This form can be used to search for peptides in raw data (e.g. RAW, WIFF, *.d, mzML, mz5, mzXML). The mz5 files for this tutorial are
        centroided to make them faster to download than the original profile Thermo RAW files produced by the mass spectrometer.</p>
    <p>Add the included DDA mz5 files to the search by doing the following:</p>
    <ul>
        <li>Click the <strong>Add Files</strong> button.</li>
        <li>Select all the mz5 files in the DdaSearchMS1Filtering folder you created for this tutorial.</li>
        <li>Click the <strong>Open</strong> button.</li>
    </ul>
    <p class="keep-next">The wizard form should now look like this:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-5.png" /></p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p class="keep-next">A form should appear asking you how to handle the prefix shared by the three mz5 files: </p>
    <p><img src="image-6.png" /></p>
    <ul>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <p>The wizard will advance to the <strong>Add Modifications</strong> page, where it lists all of the amino acid modifications in the document that
        you might want to include in the DDA search. It is important to distinguish here between fixed and variable modifications: fixed (sometimes
        called static) modifications are always applied to the specified amino acids. For example, Carbamidomethyl C is usually treated as a fixed
        modification because all cysteines in the data are expected to be alkylated. Oxidation M is almost always treated as a variable modification
        because oxidation is hit-or-miss depending on sample handling. Skyline’s search always treats isotopic labels as variable, but you can change
        whether other modifications are treated as fixed or variable by clicking the <strong>Edit modifications</strong> button.</p>
    <p class="keep-next">You can also add modifications to the document from this page. Since the document was reset to defaults, the list starts with
        only Carbamidomethyl (C):</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-7.png" /></p>
    <p>These data are SILAC labelled, so you will need to add heavy label modifications here. To add them, do the following:</p>
    <ul>
        <li>Click the <strong>Edit modifications</strong> button.</li>
        <li>Click the <strong>Edit heavy modifications</strong> menu option.</li>
        <li>Click the <strong>Add</strong> button in the <strong>Edit Isotope Modifications</strong> form.</li>
        <li>In the <strong>Name</strong> field of the <strong>Edit Isotope Modification</strong> form, enter “Label:13C(6)15N(2) (C-term K)”.</li>
        <li>Click the down arrow at the right side of the <strong>Name</strong> field and click the item with the same name. This will populate the
            specificity and composition fields from Unimod.</li>
    </ul>
    <p class="keep-next">The <strong>Edit Isotope Modification</strong> form should now look like this:</p>
    <p><img src="image-8.png" /></p>
    <ul>
        <li>Click the <strong>OK</strong> button</li>
    </ul>
    <p>Add a second isotope modification by doing the following:</p>
    <ul>
        <li>Click the <strong>Add</strong> button on the <strong>Edit Isotope Modifications</strong> form.</li>
        <li>From the <strong>Name</strong> dropdown list of the <strong>Edit Isotope Modification</strong> form, choose “Label:13C(6)15N(4) (C-term
            R)”.</li>
    </ul>
    <p>The <strong>13C</strong> and <strong>15N</strong> checkboxes are checked automatically to tell Skyline to use <sup>13</sup>C for all carbon
        atoms and <sup>15</sup>N for all nitrogen atoms present in an Arginine molecule, for a total mass shift of 10 Daltons (6x <sup>13</sup>C + 4x
        <sup>15</sup>N).</p>
    <p class="keep-next">The <strong>Edit Isotope Modification</strong> form should now look like this:</p>
    <p><img src="image-9.png" /></p>
    <p>Skyline automatically calculates both the monoisotopic and average masses, approximately 8 Daltons for Lysine (K) and 10 Daltons for Arginine
        (R) that will result from using 13C and 15N in these amino acid residues. To finish adding the heavy modifications:</p>
    <ul>
        <li>Click the <strong>OK</strong> button in the <strong>Edit Isotope Modification</strong> form.</li>
        <li>Click the <strong>OK</strong> button in the <strong>Edit Isotope Modifications</strong> form.</li>
        <li>Check the checkboxes in the <strong>Add Modifications </strong>list for the “Label:13C(6)15N(2) (C-term K)” and
            <strong>“</strong>Label:13C(6)15N(4) (C-term R)<strong>”</strong> modifications you just created.</li>
    </ul>
    <p>Now you will add Oxidation (M) as a structural modification:</p>
    <ul>
        <li>Click the <strong>Edit modifications</strong> button.</li>
        <li>Click the <strong>Edit structural modifications</strong> menu option.</li>
        <li>Click the <strong>Add</strong> button in the <strong>Edit Structural Modifications</strong> form.</li>
        <li>In the <strong>Name</strong> field of the <strong>Edit Structural Modification</strong> form, enter “Oxidation (M)”.</li>
        <li>Click the down arrow at the right side of the <strong>Name</strong> field and click the item with the same name. This will populate the
            specificity and composition fields from Unimod.</li>
        <li>Click the <strong>OK</strong> button in the <strong>Edit Structural Modification</strong> form.</li>
        <li>Click the <strong>OK</strong> button in the <strong>Edit Structural Modifications</strong> form.</li>
        <li>Check the checkbox in the <strong>Add Modifications </strong>list for the “Oxidation (M)” modification you just created.</li>
        <li>Also make sure the checkbox for “Carbamidomethyl (C)” is checked, as it should be because you chose the default settings.</li>
    </ul>
    <p class="keep-next">At this point the <strong>Add Modifications</strong> page should look like:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-10.png" /></p>
    <ul>
        <li>Click the<strong> Next </strong>button.</li>
    </ul>
    <p>The wizard will advance to the <strong>Configure Full-Scan Settings</strong> page.</p>
    <ul>
        <li>In the <strong>Mass Accuracy</strong> field, enter “20”.</li>
    </ul>
    <p class="keep-next">The other fields on this page should default to values you can use for this tutorial, and the wizard should look like this:
    </p>
    <p><img alt="A screenshot of a computer screen

Description automatically generated" src="image-11.png" /></p>
    <ul>
        <li>See the <a href="https://skyline.ms/_webdav/home/software/Skyline/@files/tutorials/MS1Filtering-20_1.pdf#page=9">MS1 Full Scan Filtering
                tutorial</a> (page 9) to learn more about these settings.</li>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>This should bring you to the <strong>Import FASTA</strong> page in the wizard. For this tutorial you will use a human protein FASTA with the
        sequences from Sigma-Aldrich’s Universal Proteomics Standard (UPS) prepended at the top (so that Skyline uses those accession numbers for any
        peptides that are shared between UPS and non-UPS proteins). To select the FASTA:</p>
    <ul>
        <li>Click the <strong>Browse</strong> button.</li>
        <li>Select the ‘2014_01_HUMAN_UPS.fasta’ file in the folder you created for this tutorial.</li>
        <li>Click the <strong>Open</strong> button in the <strong>Open FASTA</strong> form.</li>
    </ul>
    <p class="keep-next">The wizard should now look like this:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-12.png" /></p>
    <ul>
        <li>Click the <strong>Next</strong> button.</li>
    </ul>
    <p>The wizard will advance to the <strong>Adjust Search Settings</strong> page. Here you can set the most important parameters for the DDA search.
        For this tutorial, do the following:</p>
    <ul>
        <li>In the <strong>MS1 tolerance</strong> field, enter “5”. (Note that when you leave the text box the form will assume you mean ppm and set
            the unit box accordingly.)</li>
        <li>In the <strong>MS2 tolerance</strong> field, enter “10”.</li>
    </ul>
    <p class="keep-next">The form should now look like this:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-13.png" /></p>
    <p>The <strong>Max q-value</strong> field sets a false discovery rate (FDR) limit allowed for the peptide matches in the search results. In this
        tutorial, you will use a 1% FDR (maximum q value of 0.01), which is the default and the most common limit used in the field of proteomics.</p>
    <ul>
        <li>Click the <strong>Next</strong> button to start the search.</li>
    </ul>
    <p class="keep-next">The <strong>DDA Search</strong> page will show you the progress of the search. You may also cancel the search by clicking the
        <strong>Cancel Search</strong> button on this page.</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-14.png" /></p>
    <p>After the search is finished:</p>
    <ul>
        <li>Click the <strong>Finish</strong> button.</li>
    </ul>
    <p class="keep-next">Skyline will start to build a spectral library from the search results. After the library build is completed, a message box
        will pop up to warn you that there were some spectra that could be interpreted as multiple different peptides with equal confidence, and that
        those peptides will be ignored:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-15.png" /></p>
    <ul>
        <li>Click the <strong>OK</strong> button to dismiss the message.</li>
    </ul>
    <p class="keep-next">Skyline will then start to import the library into your document. When that is finished it will ask you to set criteria for
        including proteins in the document:</p>
    <p><img alt="A screenshot of a computer screen

Description automatically generated" src="image-16.png" /></p>
    <p>Skyline can merge proteins that match the same set of peptides into protein groups and find a minimal list of proteins that explain all the
        peptides. You can read more about protein grouping and parsimonious protein inference in this documentation page: <a
            href="https://skyline.ms/wiki/home/software/Skyline/page.view?name=Skyline%20Protein%20Association%2022.1">Skyline Protein
            Association</a>. For this tutorial, leave the checkboxes for creating protein groups and finding a minimal list of proteins unchecked.</p>
    <ul>
        <li>Select <strong>Assigned to first protein </strong>from the <strong>Shared peptides are: </strong>dropdown list<strong>.</strong></li>
        <li>Click the <strong>OK</strong> button.</li>
    </ul>
    <h1>Configuring Skyline to View the Imported Data</h1>
    <p>After the proteins are imported into the document, you will see the main Skyline window with the UPS proteins at the top of the
        <strong>Targets</strong> view. You should see <strong>5,219 </strong>proteins there (counted in the status bar). Skyline will also begin
        extracting chromatograms with progress displayed in an <strong>Importing Results</strong> form. You can wait several minutes for chromatogram
        extraction to complete or you can continue with all but the last step below during the import.</p>
    <ul>
        <li>Click on the [+] icon next to the third protein P01112ups|RASH_HUMAN_UPS.</li>
        <li>Click on the [+] icon next to that protein’s third peptide R.TGEGFL<strong>C</strong>VFAINNTK.S.</li>
        <li>Click on that peptide’s first precursor <a id="OLE_LINK1"></a><a id="OLE_LINK2"></a>835.9140++ and the chromatogram for that precursor and
            the MS/MS spectrum for that peptide will appear. (Note that the bold, underlined residue “<strong>C</strong>” in the peptide sequence
            indicates a carbamidomethyl cysteine). </li>
        <li>If you do not see the MS/MS spectrum, on the <strong>View</strong> menu, click <strong>Library Match</strong>.</li>
        <li>If you do not see as many annotated peaks in the <strong>Library Match</strong> view as in the image below, on the <strong>View</strong>
            menu, choose<strong> Ion Types</strong> and check <strong>B</strong> and <strong>Y</strong>.</li>
        <li>If you do not see the entire chromatogram for the peptide, on the <strong>View</strong> menu, choose <strong>Auto-Zoom</strong> and click
            <strong>None</strong> (Shift-F11).</li>
        <li>Right-click the chromatogram graph, choose <strong>Peptide ID Times</strong>, and click <strong>Aligned</strong> if it is unchecked.</li>
    </ul>
    <p class="keep-next">Your Skyline window should look something like this:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-17.png" /></p>
    <p>The document is now fully configured for MS1 Filtering with three DDA runs imported. You will see that the chromatogram in this view is
        approximately 10 minutes in length (133 to 143 minutes), due to the <strong>Use only scans within [5] minutes of MS/MS IDs</strong> setting
        chosen in the import wizard. Note that when the Skyline document is set up for MS1 Filtering, in the place where you would see product ion
        transitions (e.g. y-ions) for triple quadrupole SRM experiments, you will now see the different precursor isotope peaks, such as for peptide
        TGEGFL<strong>C</strong>VFAINNTK: precursor – 835.914++, precursor [M+1] - 836.4154++, and precursor [M+2] – 836.9164++.</p>
    <p>To configure a few other features that will be helpful in general, especially to visualize certain MS1 Filtering data, perform the following
        steps:</p>
    <ul>
        <li>On the <strong>Settings</strong> menu, make sure <strong>Integrate All </strong>is checked.</li>
    </ul>
    <p>This tells Skyline to treat all chromatograms in a peak group (here precursor ions M, M+1 and M+2) as integrating together, regardless of
        whether peaks appear to be co-eluting with the largest peak. It no longer impacts the integrated peak areas as it once did.</p>
    <ul>
        <li>On the <strong>View</strong> menu, choose <strong>Peak Areas</strong> and click <strong>Replicate Comparison</strong>.</li>
    </ul>
    <p class="keep-next">A window with a graph like this should appear:</p>
    <p><img alt="Chart, bar chart

Description automatically generated" src="image-18.png" /></p>
    <p>You can dock the <strong>Peak Areas</strong> window to your desired location by doing the following:</p>
    <ul>
        <li>Click and hold down the left mouse button, and then drag it until the mouse cursor is above the <strong>Library Match</strong> view.</li>
        <li>When a set of 5 icons in the shape of a cross appears, move the mouse into the lower icon and release the left mouse button to divide the
            space on the right edge of the Skyline window between the <strong>Peak Areas</strong> and <strong>Library Match</strong> views.</li>
        <li>Move the <strong>Library Match</strong> view below the <strong>Targets</strong> view by the same process.</li>
        <li>On the <strong>View</strong> menu, choose <strong>Auto-Zoom</strong> and click <strong>Best Peak</strong> (F11).</li>
        <li>On the <strong>View</strong> menu, choose <strong>Arrange Graphs</strong> and click <strong>Column</strong>.</li>
        <li>Right-click a chromatogram graph, and click <strong>Legend</strong> to turn off the legend in the chromatogram graphs.</li>
    </ul>
    <p class="keep-next">Now your Skyline window should look something like this:</p>
    <p><img alt="A screenshot of a computer

Description automatically generated" src="image-19.png" /></p>
    <p>From this point you can learn more about working with DDA data by consulting the <a
            href="https://skyline.ms/_webdav/home/software/Skyline/@files/tutorials/MS1Filtering-20_1.pdf">MS1 Full Scan Filtering tutorial</a>.</p>
</body>

</html>